\documentclass[11pt, oneside]{article}

%
% Packages
%

\usepackage{geometry}
\geometry{letterpaper}
\usepackage[parfill]{parskip}      		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\usepackage{booktabs}
\usepackage{topcapt}
\usepackage[labelfont =  {sf, bf}, textfont = sf, format = plain]{caption}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{color}
\usepackage{array}
\usepackage{gensymb}

% Make helvetica the default sans-serif font
\renewcommand\sfdefault{phv}

% Package command for citing R packages
\newcommand{\pkg}[1]{{\fontseries{b}\selectfont #1}} 

%
% Commenting
%

\newcommand{\cdm}[1]{{ \color{magenta} [{\bf{CDM:}} {\em#1}]}} % Chris' comments are in magenta.
\newcommand{\ala}[1]{{ \color{blue} [{\bf{ALA:}} {\em#1}]}} % Amy's comments are in blue.

%
% Where to find files
%

<<Front Matter>>=

  # The path to the ms directory may need to be changed for different users
  # Manuscript files
  pathMS <- "~/Google Drive/CardLocalAdaptation/ms" # Chris' MacBook
  pathMS <- "~/Documents/CardAdapt/ms" # Chris' Linux laptop
  
  # Data files for paper
  pathDat <- paste(pathMS, "/Data", sep = "")

  # Figures for paper
  pathFig <- paste(pathMS, "/Figures", sep = "")

  # Tables for paper
  pathTab <- paste(pathMS, "/Tables", sep = "")

  # Small function to make statistical significance asterisks
  sigStar <- function(pvalue)
  {
    if (pvalue >= 0.05) return("")
    if (pvalue < 0.05 & pvalue >= 0.01) return("*")
    if (pvalue < 0.01 & pvalue >= 0.001) return("**")
    if (pvalue < 0.001) return("***")
  }

  # Color palette
  # Primary color:

   primShade0 <- rgb(1, 0, 0.286)
   primShade1 <- rgb(1, 0.643, 0.745)
   primShade2 <- rgb(1, 0.439, 0.6)
   primShade3 <- rgb(0.765, 0, 0.22)
   primShade4 <- rgb(0.588, 0, 0.169)

  # Secondary color (1):

  sec1shade0 <- rgb(0.008, 0.957, 1)
  sec1shade1 <- rgb(0.643, 0.984, 1)
  sec1shade2 <- rgb(0.443, 0.976, 1)
  sec1shade3 <- rgb(0, 0.49, 0.514)
  sec1shade4 <- rgb(0, 0.376, 0.396)

  # Secondary color (2):

  sec2shade0 <- rgb(0.988, 1, 0)
  sec2shade1 <- rgb(0.996, 1, 0.643)
  sec2shade2 <- rgb(0.992, 1, 0.439)
  sec2shade3 <- rgb(0.835, 0.847, 0)
  sec2shade4 <- rgb(0.643, 0.651, 0)

@

%
% Title and authors
%

\title{(working title) Local Physiological Adaptation in \textit{Mimulus cardinalis}}
\author{Christopher D. Muir and Amy L. Angert}
%\date{}							% Activate to display a given date or no date

%
% Start document
%

\begin{document}
\maketitle
\listoffigures
\listoftables

\section*{Abstract}

% write/analyze
% proposal for summer
  % local fitness set concave
  % range-wide fitness set convex
  % interaction between shape of fitness set and gene flow?
% evo-phys proposal

\section*{Introduction}

Local adaptation is one of the most ubiquitous observations in nature: organisms perform well in their natal environment, but poorly outside it. Local adaptation within species most often involves genetic differences in ecologically relevant traits caused by selection to different environments, leading to predictable trait-environment correlations that persist in a common garden. Although local adaptation is well-supported by data from reciprocal transplant experiments, trait-environment associations, and ecological genetics/genomics, predicting which traits respond to selection exceeds our understanding. In particular, the role of physiological traits is poorly understood, even though physiology explains adaptation to the abiotic environment, which we need to understand as a baseline for anticipating how organisms will respond to climate change. A related problem is identifying the exact selective agents, such as abiotic factors like soil water availability or biotic interactions [etc.]... Here, we simultaneously investigate which traits and causal selective agents explain range-wide patterns of local physiological adaptation to climate in \textit{Mimulus}, a model genus for local adaptation studies.

% niche-tracking versus niche-shifting
One surprising pattern is that populations adapt to different environments by adjusting their life history to stay within the same fundamental physiological niche rather than have the niche itself evolve. For example, mosquitoes adjust diapause length with latitude rather than evolve altered ...\cdm{what other examples could we include here?} This suggests that in many situations it is evolutionarily easier to change life history rather than fundamental physiological tolerance.

% Selective agents
We also know little about the selective agents, such as climatic factors or biotic interactions, that are most important in driving adaptive divergence. In certain situations, such as pollinator shifts and edaphic adaptation, it is straightforward to identify putative selective agents and test predicted mechanisms. However, it is more challenging for cases in which traits vary with less specific factors like mean annual precipitation or latitude. For example, body size clines (larger size at higher latitude) have evolved repeatedly on three contients in \textit{Drosophila subobscura}, strongly indicating selection. But which one or more of the many climatic factors that covary with latitude actually controls selection on body size? Selection might be even more complex for species that live over multiple seasons, meaning that environmental variation rather than the average environment may be a selective agent.

% Motivate using Mimulus
Linking physiological traits to potentially complex patterns of local adaptation requires integrating multiple lines of evidence from comparative, experimental, genomic studies under both lab and field conditions. Many classic and contemporary studies of local adaptation have been conducted using species from genus \textit{Mimulus} because of its natural history, easy propogation, and genetic/genomic resources. ... Conspicuous absence of phys mechanism

% Final paragraph
1. `intrinsic' physiological variation
2. variation in stress response
3. what ecological factors explain divergence (i.e. putative selective agents underlying local adaptation)

Conclude that observations are best explained by stabilizing selection on the fundamental niche, but spatially varying divergent selection on coordinated life history and physiological traits.

\section*{Methods}

\subsection*{Population Selection}

We used 16 populations from throughout the range of \textit{M. cardinalis} (Table 1). Seeds were collected in the field \cdm{Amy, can you explain seed collection methods?}.

<<focClim, echo = FALSE>>=
# Read in climate data for 16 focal populations to use in making Table 1
# need to make this available from ms directory
load('~/Google Drive/CardLocalAdaptation/Data/Climate/Robjects/focClim.RData') # focClim

@

\begin{table}[ht]
   \centering
   \topcaption{16 Focal populations}
   \begin{tabular}{@{} lllllll @{}}
      \toprule
  Name& Region  & Demo?& Pop gen?& Latitude  & Longtiude  & Elevation (mas) \\
      \midrule
	HAU & South Margin		& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "HAU"], 3)}	& 
    \Sexpr{round(focClim$Longitude[focClim$ID == "HAU"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "HAU"]}   \\
	CTC	& South Margin		& yes	& no	& \Sexpr{round(focClim$Latitude[focClim$ID == "CTC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "CTC"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "CTC"]}   \\
	CUR	& South Margin		& yes	& yes & \Sexpr{round(focClim$Latitude[focClim$ID == "CUR"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "CUR"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "CUR"]}   \\
	GRP & South Margin		& no	& no	& \Sexpr{round(focClim$Latitude[focClim$ID == "GRP"], 3)} &
    \Sexpr{round(focClim$Longitude[focClim$ID == "GRP"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "GRP"]}   \\
	WWC &	Transverse		  & yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "WWC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "WWC"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "WWC"]}   \\
	MIL	& Transverse	  	& yes	& no	& \Sexpr{round(focClim$Latitude[focClim$ID == "MIL"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "MIL"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "MIL"]}   \\
	WFM	& Transverse	  	& yes	& no	& \Sexpr{round(focClim$Latitude[focClim$ID == "WFM"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "WFM"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "WFM"]}   \\
	NMT	& South Sierras		& yes	& yes?& \Sexpr{round(focClim$Latitude[focClim$ID == "NMT"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "NMT"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "NMT"]}   \\
	PRD	& South Sierras		& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "PRD"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "PRD"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "PRD"]}   \\
	RWD	& South Sierras		& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "RWD"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "RWD"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "RWD"]}   \\
	WNA	& Central Sierras	& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "WNA"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "WNA"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "WNA"]}   \\
	RBW	& Central Sierras	& yes	& no	& \Sexpr{round(focClim$Latitude[focClim$ID == "RBW"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "RBW"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "RBW"]}   \\
	MYU	& North Sierras		& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "MYU"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "MYU"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "MYU"]}   \\
	LIJ	& North Sierras		& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "LIJ"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "LIJ"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "LIJ"]}   \\
	DPC	& North Coast	  	& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "DPC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "DPC"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "DPC"]}   \\
	RCC	& North Margin		& yes	& yes	& \Sexpr{round(focClim$Latitude[focClim$ID == "RCC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$ID == "RCC"], 3)}	& \Sexpr{focClim$Elevation[focClim$ID == "RCC"]}   \\
	\bottomrule
   \end{tabular}
\end{table}

\subsection*{Plant propagation}

On 14 April, 2014, 3-5 seeds per family were sown directly on sand (Quikrete Play Sand, Georgia, USA) watered to field capacity in RLC4 Ray Leach cone-tainers placed in RL98 98-well trays (Stuewe \& Sons, Inc., Oregon, USA). We used pure sand both to facilitate root-washing and because \textit{M. cardinalis} typically grows in sandy, riparian soils (A. Angert, pers. obs.). Two jumbo-sized cotton balls at the bottom of cone-tainers prevented sand from washing out. Cone-tainers were continuously bottom-watered during germination by placing them in medium-sized flow trays (FLOWTMD, Stuewe \& Sons, Inc., Oregon, USA) filled part way with water, placed on benches in greenhouses at the University British Columbia campus in Vancouver, Canada (49\degree 15' N, 123\degree 15' W). Misters thoroughly wetted the top of the sand every two hours during the day. Most seeds germinated between 1 and 2 weeks, but we allowed 3 weeks before transferring seedlings to growth chambers. Germination was recorded daily from one to two weeks after sowing, and every few days thereafter. On 5 May (21 days after sowing), seedlings were transferred to one of two MODEL Growth Chambers (Conviron, Manitoba, Canada). We thinned seedlings to one plant per cone-tainer, leaving the center-most, largest plant. 702 of 768 (91.4\%) had plants that could be used in the experiment. We allowed one week at constant, non stressful conditions (day: 20\celsius, night: 16\celsius) for plants to acclimate to growth chambers before starting treatments. The initial size of seedlings, measured as the length of the first true leaves, did not differ between populations, families, or treatments (Table S\#).

\paragraph*{Monitoring environment}

  	Cross check with Poorter paper \\
		Light (PAR sensors), temp/humidity (HOBOs), soil moisture
    
\subsection*{Treatments}

We used 2 watering treatments on two populations on two soils. The watering treatments are: well-watered, drought.

\subsubsection*{Temperature}

We simulated typical growing season (June and July) air temperatures at the two most thermally divergent focal sites in our study, Whitewater Canyon (High Temp) and Little Jameson (Low Temp). We downloaded daily interpolated mean, minimum, and maximum air temperature from 13 years (2000-2012) at both sites from ClimateWNA \citep{Wang_etal_2012}. Daily temperatures from ClimateWNA are usually highly correlated with the air temperature recorded from data loggers in the field at these sites (A. Angert, unpub. data). Hence, the ClimateWNA temperature profiles are likely to similar to actual thermal regimes experienced by \textit{M. cardinalis} in nature. To create realistic temperature regimes, we calculated the mean temperature trend from June to July using LOESS \citep{Cleveland_etal_1992}. The residuals were highly autocorrelated at both sites (warmer than average days are typically followed by more warm days) and there was strong correlation ($r = 0.65$) between sites (warm days in WWC were also warm in LIJ). The `VARselect' function in the \pkg{vars} package for R \citep{Pfaff_2008} indicated that a lag two Vector Autoregression (VAR(2)) model best captured the within-site autocorrelation as well as between-site correlation in residuals. We fit and simulated from the VAR(2) using the package \pkg{dse} \citep{Gilbert_2014} in R. Simulated data closely resembled the autocorrelation and between-site correlation of the actual data. From simulated mean temperature, we next selected minimum and maximum daily temperatures. Mean, min, and max temperature were highly correlated at both sites. We chose min and max temperatures using site-specific fitted linear models between mean, max, and min temperature, with additional variation given by normally-distributed random deviates with variance equal to the residual variance of the linear models. For each day, the nighttime (22:00 - 6:00) chamber temperature was set to the simulated minimum temperature. During the middle of the day, chamber temperature was set to the simulated maximum temperature, with a variable period of transition between min and max so that the average temperature was equal the simulated mean temperature.

\subsubsection*{Water}

      April 14 germinate
      Before May 20 (36 days): 7.5 min (complete flood) every 2 hours
      between May 20 to June 10 (57 days), down to 30 seconds, 1x per day
			Wet: daily/constant irrigation with cooled water
			Dry: start at week \#, gradually decrease watering frequency/level

			\cdm{Are there any data from California about soil water content through the season?}

<<ExperimentalDesign>>=
library(scales) # for color transparency
exptlDes <- read.csv(paste(pathDat, "/ExperimentalDesign.csv", sep = ""))

pdf(paste(pathFig, "/ExperimentalDesignBase.pdf", sep = ""), 7, 5)
par(mai = rep(1, 4))
plot(0, 0, type = "n", xlim = c(0, max(exptlDes$DayN)), ylim = c(-1, 41), 
		 axes = F, frame.plot = T, cex.lab = 1.5, xlab = "Key treatments and measurements", 
     ylab = expression(paste("Temperature (", phantom()*degree, "C)")))

# Temperature axis
axis(2, at = seq(0, 40, 10), lwd = 0, lwd.ticks = 1, las = 1,
	labels = c(expression(0*degree), expression(10*degree), expression(20*degree), 
						 expression(30*degree), expression(40*degree)))
# Axis of time on top
mtext("Days after sowing", side = 3, line = 3.5, cex = 1.5)
mtext("(Date for climate simulation)", side = 3, line = 2.25, cex = 1.5)
axis(3, at = seq(0, 90, 30), lwd = 0, lwd.ticks = 1)
axis(3, at = c(17, 47, 78), lwd = 0, lwd.ticks = 1, line = -0.5, 
     labels = c("(June)", "(July)", "(August)"))

# Indicate sowing, beginning of temperature and drought treatments
abline(v = 0, col = "grey", lwd = 2, lty = 2)
mtext("Sow", side = 1, at = 0, line = 0)
abline(v = 17, col = "grey", lwd = 2, lty = 2)
mtext("Temp\nTreatment", side = 1, at = 17, line = 1)
abline(v = 37, col = "grey", lwd = 2, lty = 2)
mtext("Drought\nTreatment", side = 1, at = 37, line = 1)

# Polygon of Hot treatment, min and max
with(exptlDes[18:94, ], polygon(c(17:93, 93:17), c(Tmin_hot, rev(Tmax_hot)),
  border = NA, col = alpha(primShade0, 0.25)))

# Polygon of Cool treatment, min and max
with(exptlDes[18:94, ], polygon(c(17:93, 93:17), c(Tmin_cool, rev(Tmax_cool)),
  border = NA, col = alpha(sec1shade0, 0.25)))

# Lines with average temperature
with(exptlDes[18:94, ], points(17:93, Tavg_hot, lwd = 2, type = "l", 
	col = primShade3))
with(exptlDes[18:94, ], points(17:93, Tavg_cool, lwd = 2, type = "l", 
	col = sec1shade3))

# Rug indicating measurment timing
rect(6, -2, 17, 0, border = "white", col = "grey")
with(exptlDes, text(11.5, -1, labels = "Germ", col = "white", font = 3, 
  cex = 0.9))

with(exptlDes, rect(min(which(LERmeas)), -2, max(which(LERmeas)), 0, 
	border = "white", col = "grey"))
with(exptlDes, text(min((which(LERmeas)) + max(which(LERmeas))) / 2, -1, 
	labels = "LER", col = "white", font = 3, cex = 0.9))

with(exptlDes, rect(min(which(SERmeas)), 0, max(which(SERmeas)), 2,
	border = "white", col = "grey"))
with(exptlDes, text(min((which(SERmeas)) + max(which(SERmeas))) / 2, 1, 
	labels = "SER", col = "white", font = 3, cex = 0.9))

with(exptlDes, rect(min(which(Harvest)), -2, max(which(Harvest)), 0, 
	border = "white", col = "grey"))
with(exptlDes, text(min((which(Harvest)) + max(which(Harvest))) / 2, -1, 
	labels = "Harvest", col = "white", font = 3, cex = 0.9))

with(exptlDes, text(min((which(PhotoMeas)) + max(which(PhotoMeas))) / 2, 5, 
	labels = "Photo", col = "grey", font = 3, cex = 0.9))
with(exptlDes, rect(min(which(PhotoMeas)), 2, max(which(PhotoMeas)), 4, 
	border = "white", col = "grey"))
#with(exptlDes, text(min((which(PhotoMeas)) + max(which(PhotoMeas))) / 2, 3, 
#	labels = "Photo", col = "white", font = 3, cex = 0.9))

dev.off()

# Legend
pdf(paste(pathFig, "/ExperimentalDesignLegend.pdf", sep = ""), 1.5, 2)
par(mai = rep(0, 4))
plot(0, 0, type = "n", xlim = c(0, 1.5), ylim = c(0, 1), axes = F)

polygon(c(0, 0.5, 1, 1, 0.5, 0), c(0.8, 1, 1, 0.55, 0.55, 0.35), border = NA, 
				col = alpha(primShade0, 0.25))
polygon(c(0, 0.5, 1, 1, 0.5, 0), c(0.6, 0.45, 0.45, 0, 0, 0.15), border = NA, 
				col = alpha(sec1shade0, 0.25))
points(c(0, 0.5, 1), c(0.575, 0.775, 0.775), lwd = 2, type = "l", col = primShade3)
points(c(0, 0.5, 1), c(0.375, 0.225, 0.225), lwd = 2, type = "l", col = sec1shade3)
text(0.5, 0.9, labels = "Hot", col = primShade3)
text(0.5, 0.1, labels = "Cool", col = sec1shade3)
text(1.25, 1, labels = "Max", col = alpha(primShade0, 0.25))
text(1.25, 0.55, labels = "Min", col = alpha(primShade0, 0.25))
text(1.25, 0.775, labels = "Mean", col = primShade3)

text(1.25, 0, labels = "Min", col = alpha(sec1shade0, 0.25))
text(1.25, 0.45, labels = "Max", col = alpha(sec1shade0, 0.25))
text(1.25, 0.225, labels = "Mean", col = sec1shade3)

dev.off()

@

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_ExptlDes.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Experimental Design]{CAPTION}
	\label{fig:Fig_ExptlDes}
\end{figure}

\subsection*{Measurements: Growth and photosynthesis}

\begin{table}[ht]
   \centering
   \topcaption{Traits}
   \begin{tabular}{@{} ll @{}}
      \toprule
  Trait & Units \\
      \midrule
  Day of germination  & day \\
  Leaf expansion rate  &  mm day$^{-1}$  \\
  Shoot elongation rate  &  mm or cm day$^{-1}$  \\
  Harvest dry mass &  g  \\
  Photosynthetic rate &  $\mu$mol CO$_2$ m$^{-2}$ s$^{-1}$\\
  Mortality &  Probability?  \\
	    \bottomrule
   \end{tabular}
\end{table}

\paragraph{Day of germination} We tested for population variation in germination rate, measured as Days to Germination, using a lognormal survival model fit using the survreg function in the R package \pkg{survival} version 2.38 \citep{Therneau_2015}. The model was fit with Population as a fixed effect and Family as random effect using a $\Gamma$ frailty function. The signifcance of the Population effect was determined using analysis of deviance.

<<Germination rate>>=

  library(survival)
  pathMS <- "~/Google Drive/CardLocalAdaptation/ms"
  pathDat <- paste(pathMS, "/Data", sep = "")

  master <- read.csv(paste(pathDat, "/MasterDatasheet_out.csv", sep = ""), 
    row.names = 1)

  # Create Surv object
	germ <- with(subset(master, !is.na(master$MinGermDay)), Surv(MinGermDay, MaxGermDay,	
		type = "interval2"))

	# lognormal produced best fit to data
	fitGerm <- survreg(germ ~ PopID + frailty(Family, sparse = F), 
		data = subset(master, !is.na(master$MinGermDay)), dist = "lognormal")
	suppressWarnings(anova(fitGerm)) # significant family and population effects
	
@

\paragraph{Growth rate: leaf expansion and shoot elongation}

We censused longest leaf length $>1$ mm on 10 days (twice per week) between 12 May and 12 June (28 to 59 days after sowing). We ceased measuring leaf length once it appeared to asymptote and growth shifted to shoot elongation.  We also censused plant height on 7 days (twice per week) between 29 May and 20 June (45 to 67 days after sowing). Both leaf expansion and shoot elongation were modeled as a second-order polynomials of time with individual coefficients (separate for leaf and shoot growth) using empirical Bayes' estimates from linear mixed-effects models fit using the R package \pkg{lme4} version 1.1-7 \citep{Bates_etal_2014}.

<<Leaf expansion rate>>=

  library(lme4)
  library(lmerTest)

  # New (blank) datasheet on which to add data
  newdata <- read.csv(paste(pathDat, "/MasterDatasheet_in.csv", sep = ""))

  # Read in LER data
  LERdata <- read.csv(paste(pathDat, "/LERdata.csv", sep = ""))

  #
	# Empirical Bayes Approach: get coefficients for every plant
	#

	# suppressWarnings(mmLER <- lmer(log(LLL) ~ poly(DayN, 2, raw = T) + 
    # (poly(DayN, 2, raw = T)|indiv), data = LERdata))
  # save(mmLER, file = paste(pathDat, "/mmLER.RData", sep = ""))
  load(file = paste(pathDat, "/mmLER.RData", sep = ""))

  # Extract coefficients and add to datasheet
	Y <- t(unlist(fixef(mmLER)) + t(ranef(mmLER)$indiv))[match(newdata$indiv, 
		rownames(ranef(mmLER)$indiv)), ]
	colnames(Y) <- c("b0_LER", "b1_LER", "b2_LER")
  newdata <- cbind(newdata, Y)

  # Model-based LER
	# Timeframe for Cool treatment: Day 0 - 31
	# Timeframe for Hot treatment: Day 0 - 24
	newdata$LERstart <- exp(newdata$b0_LER)
	newdata$LERend <- exp(newdata$b0_LER + 
					newdata$b1_LER * ifelse(newdata$TempTrt == "Cool", 31, 24) + 
					newdata$b2_LER * ifelse(newdata$TempTrt == "Cool", 31 ^ 2, 24 ^ 2))
	newdata$LER_AbsGrowth <- (newdata$LERend - newdata$LERstart) / 
		ifelse(newdata$TempTrt == "Cool", 31, 24)

  # Mixed model ANOVA with family as random factor fit using step down procedure
	mm <- lmer(LER_AbsGrowth ~ AvgGermDay + PopID * TempTrt * WaterTrt + (1|Family),
		data = master)
	fitLER <- step(mm, reduce.random = F) # includes pop, temp, h2o, population x water
	# fitLER <- step(mm, ddf = "Kenward-Roger", reduce.random = F) # same result using Satterthwaite ddf

@


<<Shoot elongation rate>>=

  # Read in SER data
  SERdata <- read.csv(paste(pathDat, "/SERdata.csv", sep = ""))

  #
  # Empirical Bayes Approach: get coefficients for every plant
	#

	# suppressWarnings(mmSER <- lmer(log(height + 1) ~ poly(DayN, 2, raw = T) + 
    # (poly(DayN, 2, raw = T)|indiv), data = SERdata))
  # save(mmSER, file = paste(pathDat, "/mmSER.RData", sep = ""))
  load(file = paste(pathDat, "/mmSER.RData", sep = ""))

  # Extract coefficients and add to datasheet
  Y <- t(unlist(fixef(mmSER)) + t(ranef(mmSER)$indiv))[match(newdata$indiv, 
		rownames(ranef(mmSER)$indiv)), ]
	colnames(Y) <- c("b0_SER", "b1_SER", "b2_SER")
  newdata <- cbind(newdata, Y)

  # Model-based SER
	# Timeframe for Cool treatment: Day 0 - 31
	# Timeframe for Hot treatment: Day 0 - 24
	newdata$SERstart <- exp(newdata$b0_SER)
	newdata$SERend <- exp(newdata$b0_SER + 
					newdata$b1_SER * ifelse(newdata$TempTrt == "Cool", 31, 24) + 
					newdata$b2_SER * ifelse(newdata$TempTrt == "Cool", 31 ^ 2, 24 ^ 2))
	newdata$SER_AbsGrowth <- (newdata$SERend - newdata$SERstart) / 
		ifelse(newdata$TempTrt == "Cool", 31, 24)

  mm <- lmer(SER_AbsGrowth ~ AvgGermDay + PopID * TempTrt * WaterTrt + 
		(1|Family), data = master)
	fitSER <- step(mm, reduce.random = F) #includes pop, temp, h2o, temp x water
@

\paragraph{Photosynthesis}
During the week of 10 to 16 June (57 to 63 days after sowing), we measured daytime photosynthetic rate and stomatal conductance on a subset of \Sexpr{length(which(!is.na(master$Photo)))} plants evenly spread between treatments and families within populations. The youngest, fully-expanded leaf acclimated for 3 minutes to reach steady state in a 6 cm$^2$ chamber of a LI-COR 6400XT Portable Photosynthesis System (LI-COR Biosciences, Lincoln, Nebraska). All measurements were made at ambient light (400 $\mu$mol m$^{-2}$ s$^{-1}$), temperature, and moderate relative humidity. During this period, we suspended normal day-to-day temperature fluctuations and set daytime temperatures to its average for that period (Cool: 26.5$\degree$; Hot: 36.1$\degree$ \cdm{get exact temps used}) so that all plants within a temperature treatment were measured under the same conditions.

<<Photosynthesis>>=

  # Photosynthetic rate
  mm <- lmer(Photo ~ PopID * TempTrt * WaterTrt + (1|Family),
    data = subset(master, !is.na(master$Photo)))
  fitPhoto <- step(mm, reduce.random = F)
  
  # Intrinsic photosynthetic rate
  # = residuals of Conductance - Photosynthesis regression
  # Removed one outlier (may change if data are rearranged!)
	mm <- lmer(resPhoto ~ PopID * TempTrt * WaterTrt + (1|Family),
    data = subset(master, !is.na(master$Photo))[-52, ])
	fitIntrnPhoto <- step(mm, reduce.random = F)

@

\paragraph{Mortality}
We assayed mortality during twice-weekly growth measurements. We could not get GLMM with Family effects to converge, so we used GLM with a quasibinomial error structure and assessed signifiance using Type 2 Analysis of Deviance with the R package \pkg{car}. 

<<Mortality>>=

  library(car)

  fitMort <- glm(DiedFromStress ~ PopID * TempTrt * WaterTrt,
    data = subset(master, !is.na(master$DiedFromStress)), family = "quasibinomial")

  Anova(fitMort, type = 2)
@


\paragraph{Biomass (show correlation between growth rate and biomass?)}


\subsection*{Intrinsic variation and plasticity}

\cdm{which traits: germination, growth rate, biomass, biomass allometry, leaf traits, photosynthesis?, mortality?}
We tested for Population, Treatment, and Population $\times$ Treatment interactions using mixed model ANOVA with Family included as a random factor. Models were fit by restricted maximum likelihood using lmer from the R package \pkg{lme4} \citep{Bates_etal_2014}. Significant fixed effect terms were selected using a step-wise backward elimination procedure implemented with the step function in the R package \pkg{lmerTest} version 2.0-11 \citep{Kuznetsova_etal_2014}. Denominator degrees of freedom for $F$-tests were estimated using Satterthwaite's approximation. Significant Population effect indicate intrinsic trait differences; significant Population $\times$ Treatment effects indicate population differences in plasticity. For growth rate, we also accounted for differences in germination rate by including day of germination as a factor.

\subsection*{Principal components of germination, growth, and photosynthesis}

<<Principal Components>>=

  popMeans <- data.frame(
    germ = c(coef(fitGerm)[1], coef(fitGerm)[1] + coef(fitGerm)[2:16]),
    LER = fitLER$lsmeans.table[1:16, "Estimate"],
    SER = fitSER$lsmeans.table[1:16, "Estimate"],
    photo = fitIntrnPhoto$lsmeans.table[1:16, "Estimate"])

  traitPC <- prcomp(popMeans, center = T, scale. = T)
  focClim$PC1 <- traitPC$x[, "PC1"]
  
  # Percent variance explained by first principal component
  pve <- paste(round(100 * traitPC$sdev[1] ^ 2 / sum(traitPC$sdev ^ 2), 1), "%")
  
  # Plot latitude versus PC1
  pdf(paste(pathFig, "/Figure_PC1vLat.pdf", sep = ""), 5.5, 5)
  par(mai = c(1, 1.5, 0.25, 0.25))
  plot(focClim$Lat, focClim$PC1, xlim = c(32, 44), ylim = c(-4, 4), axes = F,
    frame.plot = T, type = "n", xlab = "", ylab = "")
  title(xlab = "Latitude", cex.lab = 1.5)
  title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
    list(x = pve)))
  axis(1, at = seq(32, 44, 4), labels = c(expression(32*degree), 
		expression(36*degree), expression(40*degree), expression(44*degree)), 
		lwd = 0, lwd.ticks = 1) 
  axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
  fitLat <- lm(PC1 ~ Latitude, data = focClim)
  
	# Label y-axis from slow to fast growth
	mtext(c("Slow\nGrowth", "Fast\nGrowth"), 2, at = c(-4, 4), cex = 1.25, 
				font = 3, las = 1, line = 4, adj = 0.5)

  # polygon of confidence intervals
  x <- seq(min(focClim$Latitude), max(focClim$Latitude), l = 1e3)
  u <- predict(fitLat, new = data.frame(Latitude = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitLat) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "grey80", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  lines(range(focClim$Lat), rev(range(predict(fitLat))), lwd = 2)
  points(focClim$Lat, focClim$PC1, pch = 21, col = "black", bg = primShade0, cex = 1.5,
    lwd = 2)
  dev.off()

  # plot axes could be range maps with areas highlighted (x-axis) and short to tall plants (y-axis)
  #plot(focClim$Lat, traitPC$x[, "PC1"])
  #cor.test(focClim$Lat, traitPC$x[, "PC1"])
  #cor.test(popMeans[, 4], traitPC$x[, "PC1"])
  #plot(focClim$bio16_cva, traitPC$x[, "PC1"])
@


\subsection*{Selective agents and environmental correlates}

As with previous studies in \textit{Mimulus} and many other species [CITES], latitude \textit{per se} predicted trait variation more than many obvious environmental variables (e.g. temperature or precipitation) that covary with latitude.

<<VSURF>>=

library(VSURF)
  # Load VSURF results of climatic variables associated with latitude at cardinalis occurence points
  # Note that occurences have been randomly thinned to account for variation in sampling effort
  load(paste(pathDat, "/clim2lat_fit_t1.RData", sep = "")) # fit_t1
  vsurfLat <- fit_t1

	# Load climVars
	load(paste(pathDat, "/ExploratoryCimateVariables.RData", sep = ""))

  # Figure of variable importance for latitude
  pdf(paste(pathFig, "/Figure_clim2lat_imp.pdf", sep = ""), 4, 5)
  par(mar = c(5, 6, 1, 1))
  plot(0, 0, type = "n", xlim = c(0, ceiling(vsurfLat$ord.imp$x)[1]), 
		ylim = c(1, length(vsurfLat$varselect.pred)), axes = F, frame.plot = T, 
		xlab = "", ylab = "")
	title(xlab = "Climate-Latitude\nImportance", line = 4, cex.lab = 1.5)
  for (i in 1:length(vsurfLat$varselect.pred)) 
  {
     lines(x = c(-1, length(vsurfLat$varselect.pred)), y = c(i, i), col = "grey")
  }
  points(vsurfLat$ord.imp$x[1:length(vsurfLat$varselect.pred)], 
    length(vsurfLat$varselect.pred):1, pch = 21, col = "black", bg = "white", cex = 1.5)
	# mtext(climVars[vsurfLat$varselect.pred], side = 2, 
    # at = length(vsurfLat$varselect.pred):1, las = 1, line = 1)
	axis(1, lwd = 0, lwd.ticks = 1)
	dev.off()
  
  # Variable selection using random forests (VSURF)
  linmod.PC1 <- as.formula(paste("PC1 ~ ", paste(climVars, collapse = " + "), sep = ""))
  set.seed(69647)
  vsurfPC1 <- VSURF.parallel(linmod.PC1, data = focClim[, c("PC1", climVars)], 
    ncores = 3)

  # Plot most important climatic predictor versus PC1
  pdf(paste(pathFig, "/Figure_bio16cva_v_Lat.pdf", sep = ""), 5, 5)
  par(mar = c(5, 5, 1, 1))
  plot(focClim$bio16_cva, focClim$PC1, xlim = c(0.25, 0.55), ylim = c(-4, 4), axes = F,
    frame.plot = T, type = "n", xlab = "", ylab = "")
  title(xlab = "Precipitation of Wettest Quarter", cex.lab = 1.5)
  title(xlab = "(Coefficient of variation)", cex.lab = 1, line = 4)
  title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
    list(x = pve)))
  axis(1, lwd = 0, lwd.ticks = 1) 
  axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
  fitBio16cva <- lm(PC1 ~ bio16_cva, data = focClim)
  
  # polygon of confidence intervals
  x <- seq(min(focClim$bio16_cva), max(focClim$bio16_cva), l = 1e3)
  u <- predict(fitBio16cva, new = data.frame(bio16_cva = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitBio16cva) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "grey80", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  lines(range(focClim$bio16_cva), range(predict(fitBio16cva)), lwd = 2)
  points(focClim$bio16_cva, focClim$PC1, pch = 21, col = "black", bg = primShade0, 
    cex = 1.5, lwd = 2)
  dev.off()

  # Overlap between latitude predictors and phenotype predictors
  imp2 <- climVars[vsurfPC1$varselect.pred]
  nImpPC1 <- length(imp2)

  vsurfLat$varselect.pred
  imp1 <- climVars[vsurfLat$varselect.pred]
  nImpLat <- length(imp1)

  # Figure of variable importance for traits
	pdf(paste(pathFig, "/Figure_clim2pc1_imp.pdf", sep = ""), 4, 5)
	par(mar = c(5, 1, 1, 6))
	
	plot(0, 0, type = "n", xlim = c(0, 0.3), ylim = c(1, nImpLat), 
		axes = F, frame.plot = T, xlab = "", ylab = "")
	title(xlab = "Climate-Trait PC1\nImportance", line = 4, cex.lab = 1.5)
	for (i in (nImpLat:(nImpLat - nImpPC1 + 1))) 
	{
		lines(x = c(-100, 100), y = c(i, i), col = "grey")
	}
	points(vsurfPC1$ord.imp$x[1:nImpPC1], nImpLat:(nImpLat - nImpPC1 + 1), 
		pch = 21, col = "black", bg = "white", cex = 1.5)
	# mtext(climVars[vsurfPC1$varselect.pred], side = 4, las = 1, line = 1, 
  #  at = nImpLat:(nImpLat - nImpPC1 + 1), 
	#	font = ifelse(vsurfPC1$varselect.pred %in% vsurfLat$varselect.pred,	2, 1))
	axis(1, lwd = 0, lwd.ticks = 1, at = seq(0, 0.3, 0.1))
  dev.off()
@


(this also goes into the goal of trying to figure out ecological selective agent driving trait variation). Why does latitude predict trait variation, often better than any of the climatic variables that vary with latitude? Some hypotheses:
\begin{itemize}
  \item{H1: Latitude encompasses multivariate climatic dimension.}
  \item{H2: Latitude is substitute for average selective environment. i.e. populations are connected by gene flow, such that alleles at a given site must do well there and in surrounding areas.}
\end{itemize}

Predictions:
\begin{itemize}
  \item{P1: Climatic axis in high dimensional climate space should correlate with latitude (across card sites especially, not just any site [using pseudoabsences]). Focal site position in high-dimensional climatic space should correlated with traits as well or better than latitude.}
  \item{P2: Climate (averaged over nearby sites) varies with latitude and the nearby average climate predicts trait values as well or better than latitude.}
  \item{P1+2: These are not mutually exclusive and both could be imortant. i.e. average selective environment in high-dimensional climate space predicts trait values as well or better than latitude.}
\end{itemize}

Methods specific to latitude vs. trait correlation:
We used Random Forest regression \citep{Liaw_Wiener_2002} to identify putative climatic factors underlying trait-latitude associations in \textit{M. cardinalis}. We used a comprehensive set of \cdm{358 in total before thinning} recent (since 2000) known \textit{M. cardinalis} occurences recorded from herbarium records and an exhaustive field survey in 2010-11 \citep{Angert_ENM}. For each occurence, we used a 90m digital elevation model from HydroSHEDS \citep{Lehner_etal_2006} to extract elevation. Monthly interpolated climate layers were calculated using ClimateWNA \citep{Wang_etal_2012}, which accurately downscales climate data specifically for the rugged topography of western North America. For each occurence, we calculated bioclimatic variables using the biovars function in the R package \pkg{dismo} \citep{Hijmans_etal_2014}. In total, we included X environmental variables from ClimateWNA and 17? bioclimatic variables (Table S\#). For each variable, we calculated both a 30-year normal by averging annual values between 1981 and 2010 and 30-year coefficient of variation, a standardized metric of interannual climatic variation). Temperatures were converted to Kelvin to be on an interval scale appropriate for calculating the coefficient of variation. 

Selection at given site might not reflect simply the local environment, but rather the average environment in a surrounding region, since alleles will experience selection across populations linked by migration. We therefore calculated spatially averaged \cdm{maybe spatially-integrated is better term?} climate means and coefficients of variation. We sampled climate at 1000 random points (at 90-m resolition) within a 10 km buffer around focal populations. Since \textit{M. cardinalis} is found exclusively in riparian areas, we only selected points along streams using the National Hydrogeoraphy Dataset \citep{NHD}. Climatic means and CVs were weighted by the predicted suitability of each point, using ENM models from \cite{Angert_ENM}.

To identify climatic variables that were closely associated with latitude in \textit{M. cardinalis}, we used a "Variable Selection Using Random Forest" (VSURF) algorithm in the R \pkg{VSURF} version 0.8.2 \citep{Genuer_etal_2014}. In this algorithm, climatic variables are regressed against latitude, and the most important variables were selected after a three-stage selection process that progressively excludes unimportant variables.

Predictions: looked for climatic variables (means or CVs) that were strong predictors of both population latitudes (from all card pops) and trait variation in 16 focal popultions. If spatial average environment is important, then expect greater predictive power and closer association with latitude-predictors than point averages.

Evaluated fit on test set (correlation between predicted versus actual latitude) and on pseudoabsence set.
, Despite this common, and long-standing pattern, only certain traits In particular, adaptation to the local abiotic environment is generally thought to be the major cause of natural selection. Physiology, the study of organismal function, connects fitness to the abiotic environment, yet we actually know little about the physiological basis of adaptation to different environments within a species. To address this question, we looked at physiological variation across a broad latitudual gradient within \textit{Mimulus cardinalis}...

% Methods for spatial-average selection
For 16 focal populations, I took the average climate within a 10 km buffer, clipped to streams based on the National Hydrological Database \citep{NHD} weighted by climatic suitability determined [multimodel average/ensemble?] from ecological niche modeling \citep{Angert_ENM}.

\begin{table}[ht]
   \centering
   \topcaption{Climatic variables}
   \begin{tabular}{@{} ll @{}}
      \toprule
  Climate variable & Abbreviation \\
      \midrule
  % climateWNA VARIABLES:
	DD\_0	& degree-days below 0\celsius (chilling degree-days)  \\
	DD5	  & degree-days above 5\celsius (growing degree-days)   \\
	DD\_18	& degree-days below 18\celsius (heating degree-days)  \\
	DD18	& degree-days above 18\celsius (cooling degree-days)  \\
	NFFD	& number of frost-free days                           \\
	PAS	  & precipitation as snow (mm) between August in previous year and July in current  \\
	Eref	& Hargreaves reference evaporation (mm) \\
	CMD	  & Hargreaves climativ moisture deficit (mm) \\
	MAR   &	mean annual solar radiation (NOTE: removing because too many missing values) \\
	RH    &	mean annual relative humidity \\
  % BIOCLIM VARIABLES:
  bio1	&	Annual Mean Temperature	\\
	bio2	&	Mean Diurnal Range (Mean of monthly (max temp - min temp))	\\
	bio3	&	Isothermality (bio2/bio7) (* 100)	\\
	bio4	&	Temperature Seasonality (standard deviation *100)	\\
	bio5	&	Max Temperature of Warmest Month	\\
	bio6	&	Min Temperature of Coldest Month	\\
	bio7	&	Temperature Annual Range (bio5-bio6)	\\
	bio8	&	Mean Temperature of Wettest Quarter	\\
	bio9	&	Mean Temperature of Driest Quarter	\\
	bio10	&	Mean Temperature of Warmest Quarter	\\
	bio11	&	Mean Temperature of Coldest Quarter	\\
	bio12	&	Annual Precipitation	\\
	bio15	&	Precipitation Seasonality (Coefficient of Variation)	\\
	bio16	&	Precipitation of Wettest Quarter	\\
	bio17	&	Precipitation of Driest Quarter	\\
	bio18	&	Precipitation of Warmest Quarter	\\
	bio19	&	Precipitation of Coldest Quarter	\\
	    \bottomrule
   \end{tabular}
\end{table}

%--------------------------------------------------
% Results
%--------------------------------------------------

\section*{Results}

\subsection*{Coordinated latitudinal clines in germination, growth, and photosynthesis}

Genetic differences between populations affecting originsmal function and performance point to traits that may be involved in local adaptation. Using a common garden design, we identified strong genetic differences in germination, growth, and photosynthetic rate among populations of \textit{M. cardinalis} (Table~\ref{table:Table_AnovaSummary}). A single principal component captured \Sexpr{pve}$\%$ of the trait variation among populations, defining an axis of variation from fast to slow growth (Fig~\ref{fig:Fig_PC1vLat}). As we explain below, intrinsic differences between populations in terms of plant function (photosynthesis) and performance (growth) contrasted greatly with little variation in plasticity.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Table of Population, Treatment, Population x Treatment interactions for varies traits
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{table}[ht]
   \centering
   \topcaption{Population, treatment effects}
   \begin{tabular}{@{} llllllll @{}}
      \toprule
                                   & Germination & Leaf expansion & Stem elongation & Photosynthesis & Intrin. Photo & Mortality \\
                                   & survreg     & lmer           & lmer            & lmer           & lmer          & glm \\
      \midrule
  Population                       &
    \Sexpr{sigStar(anova(fitGerm)["Pr(>Chi)"]["PopID",])}  &
    \Sexpr{sigStar(fitLER$anova.table["PopID", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["PopID", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID", "Pr(>Chisq)"])} & \\
  Temperature                      &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["TempTrt", "Pr(>F)"])}    &
    \Sexpr{sigStar(fitSER$anova.table["TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["TempTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(Anova(fitMort, type = 2)["TempTrt", "Pr(>Chisq)"])} & \\
  Water                            &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["WaterTrt", "Pr(>F)"])}   &
    \Sexpr{sigStar(fitSER$anova.table["WaterTrt", "Pr(>F)"])}   &
    \Sexpr{sigStar(fitPhoto$anova.table["WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["WaterTrt", "Pr(>F)"])} &     
    \Sexpr{sigStar(Anova(fitMort, type = 2)["WaterTrt", "Pr(>Chisq)"])} & \\
  Pop $\times$ Temp                &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["PopID:TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID:TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID:TempTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["PopID:TempTrt", "Pr(>F)"])} &     
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID:TempTrt", "Pr(>Chisq)"])} & \\
  Pop $\times$ Water               &
     NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["PopID:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID:WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["PopID:WaterTrt", "Pr(>F)"])} &     
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID:WaterTrt", "Pr(>Chisq)"])} & \\
  Temp $\times$ Water              &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} &     
    \Sexpr{sigStar(Anova(fitMort, type = 2)["TempTrt:WaterTrt", "Pr(>Chisq)"])} & \\
  Pop $\times$ Temp $\times$ Water &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(fitIntrnPhoto$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} &     
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID:TempTrt:WaterTrt", "Pr(>Chisq)"])} & \\
      \bottomrule
   \end{tabular}
   \label{table:Table_AnovaSummary}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of latitude versus PC1 (growth axis)
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_PC1vLat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations grow faster]{CAPTION}
	\label{fig:Fig_PC1vLat}
\end{figure}

\subsection*{Remarkably little evidence for variation in plasticity}

Genotype $\times$ environment interactions are also a common signature of local adaptation [CITE]. For example, populations from more stressful environments may tradeoff reduced growth rate under benign conditions for the ability to tolerate stress by maintaining positive growth rates and/or surviving through adverse conditions. We found remarkably little evidence for this pattern in \textit{M. cardinalis}. There was only one statistically significant Population $\times$ Treatment interaction (Table~\ref{table:Table_AnovaSummary}), but this idiosyncratic result was weak and would not have survived correction for multiple testing. Otherwise, populations responded remarkably similarly to treatments: faster growth in the hot treatment, slower growth in the dry treatment, and high mortality in the hot, dry treatment (Table~\ref{table:Table_AnovaSummary}). Note that interactions were calculated after factoring out intrinsic trait differences, necessarily reducing statistical power to detect significant interactions relative to main effects. However, the fact that the Population and Treatment effects were highly significant ($P \ll 0.001$ in most cases) suggests that statistical power alone cannot explain why we failed to detect Population $\times$ Treatment interactions.

\subsection*{Latitude and phenotypic clines are associated with more variable precipitation}

Latitude explains variation
Presumably selective agent is not latitude, but something related to it
Exploratory approach to identify putative selective agent to test in future studies
We used a machine learning approach...identified interannual variation in precipitation (Fig~\ref{fig:Fig_ClimVarImp})
Obviously, cannot identify alternative variables that were not included in analysis.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of climatic variable importance
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_ClimVarImp.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Interannual variation in precipitation is closely correlated with latitude and trait variation]{CAPTION}
	\label{fig:Fig_ClimVarImp}
\end{figure}

%--------------------------------------------------
% Discussion
%--------------------------------------------------

\section*{Discussion}

1. How does local adaptation occur? It's complex (gene flow, shape of tradeoffs, demography) and although there is abundant evidence for local adaptaiton, we do not a general sense for when local adaptation succeeds or fails. For this, we need a system in which there are axes of variation and constraint, traits with known physiological function?, and selective agents

2. Why does local adaptation succeed? shallow environmental gradients lead to convex tradeoffs not heavily constrained by gene flow.

3. Selective agent - unpredictability. Explain

4. Why does local adaptation fail? concave, high gene flow, demography. Qualify that our metrics of plasticity are coarse, but at least suggest that GxE is much weaker than G for the environmental gradients evaluated here.

5. Future work and broad conclusions. Supports general conclusions that traits related to timing and growth are evolutionarily labile whereas those related to the fundamental niche are constrained (cites like \cite{Emery_etal_2012, Emery_Ackerly_2014}).

% stuff about environmental grain
How organisms adapt depends on the grain of the environment, ..., and gene flow between populations

Possibilities:
1. no trait variation: no environmental variation, high gene flow, generalist (convex)
2. variation in mean:
3. variation in plasticity: convex?,

our data are most consistent with:
- concave tradeoffs, fine-grained (coarse+concave would predict alt morphs), high gene flow

%--------------------------------------------------
% References
%--------------------------------------------------

\setlength{\bibsep}{6pt}
\bigskip

\bibliography{refs}
\bibliographystyle{evolution}

%--------------------------------------------------
% Supplementary Information
%--------------------------------------------------

% DIC table for initial size (LLL on 5/12)
\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[ANOVA table, leaf expansion rate]{Initial size of seedlings did not vary among Populations, Families, or Treatments. We used a censored Gaussian model of initial size at the outset of the experiment (longest leaf length of the first true leaves). The model was censored because we could not accurately measure leaves less than 0.25 mm with digital callipers (217 of 702, 30.9\%, were too small). We fit models using a Bayesian MCMC method implemented using the MCMCglmm function with default priors in the R package \pkg{MCMCglmm} version 2.17 \citep{Hadfield_2010}. We estimated the posterior distribution from 1000 samples of an MCMC chain run for $10 ^ 5$ steps after a $10^4$ step burn-in. We step-wise backward elimination procedure to find the best-supported model according to Deviance Information Criterion (DIC).}
	\begin{center}
	\begin{tabular}{>{\everypar{\hangindent1cm}{}\raggedright}p{6cm}lc}
	\toprule

	\input{Tables/Table_InitialSize.txt}

	\bottomrule
	\end{tabular}
	\end{center}
\end{table}

% ANOVA table for LER (need to label tables)
\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[ANOVA table, leaf expansion rate]{CAPTION}
	\begin{center}
	\begin{tabular}{lcccccc}
	\toprule

	\input{Tables/Table_lllGrowthAnova.txt}

	\bottomrule
	\end{tabular}
	\end{center}
\end{table}

% ANOVA table for SER (need to label tables)
\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[ANOVA table, stem expansion rate]{CAPTION}
	\begin{center}
	\begin{tabular}{lcccccc}
	\toprule

	\input{Tables/Table_heightGrowthAnova.txt}

	\bottomrule
	\end{tabular}
	\end{center}
\end{table}

% Figures showing latitudinal clines with specific traits

\begin{figure}[h!]
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/Figure_DoG_Lat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations germinate sooner.]{CAPTION}
	\label{fig:Fig_DoG}
\end{figure}

\begin{figure}[h!]
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/Figure_LLL_Lat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations grow faster (leaf expansion rate).]{CAPTION}
	\label{fig:Fig_LLL}
\end{figure}

\begin{figure}
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/Figure_Height_Lat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations grow faster (stem elongation rate).]{CAPTION}
	\label{fig:Fig_height}
\end{figure}

\end{document}





	\subsection{Structure}

	Leaf traits method: \\
	Day 1: \\
	- select focal leaf (e.g. second or third leaf); \\
	- place in H20, dark, cool overnight. \\
	Day 2: \\
	- wet weight; \\
	- scan; \\
	- place in drying oven. \\

		\begin{enumerate}
			\item{Stomatal density, guard cell length, ratio from leaf opposite focal leaf (above}
			\item{Leaf size/shape (from scans)}
			\item{Leaf thickness/SLA}
			\item{Colorimetry (from scans, spec)}
			\item{stem anthocyanin (qualitative)}
		\end{enumerate}

	OPEN QUESTIONS: \\
	save wet samples in preservative (e.g. for leaf venation?) \\
	chlorophyll content? \\
	branching architecture (anything in plant protocols?)

	\subsection{Function}

		\subsubsection{during experiment}

		\begin{enumerate}
			\item{Instantaneous photosynthetic rate, $g_\text{s}$, water-use efficiency.}
				\subitem{3 min per plant per LICOR}
				\subitem{Do as many plants as possible in 1-2 weeks}
				\subitem{Before and after drought?}
			\item{Respiration}
			\item{Whole plant water-use (gravimetric)}
				\subitem{}
			\item{Leaf temperature (subset)}
			\item{Temperature response curve (subset)}
				\subitem{see protocol here: http://prometheuswiki.publish.csiro.au/tiki-index.php?page=Temperature+response+of+photosynthesis+using+a+Li-Cor+6400}
			\item{Light response curve}
				\subitem{Sampling? Only in cool, wet?}
				\subitem{At what temperature?}

		\end{enumerate}

		\subsubsection{after experiment}

		\begin{enumerate}
			\item{CN content}
			\item{carbon isotope discrimination}
		\end{enumerate}


	\subsection{Development time and Life history}

	\begin{enumerate}
		\item{Flowering time: census daily}
		\item{Starch accumulation: collect (subset) of plants}
		\item{Rhizome formation: collect on subset of plants harvested for dry mass}
	\end{enumerate}

	\subsection{Floral (2nd/3rd/etc flower?)}

	\begin{enumerate}
		\item{Flower size}
		\item{Herkogamy}
	\end{enumerate}

\section{Performance}

	Possibilities: \\
	1. (sub-)weekly census of plant height. $\approx$ 0.5 min per plant, so allow 1 day \\

	$$RGR_\text{height} = \frac{\text{ln} H_2 - \text{ln} H_1}{t_2 - t_1}$$

	2. early development (pre-treatment) leaf-area growth with camera \\
	3. harvest dry biomass before and after treatments (get whole plant leaf area using scanner at same time to ground truth camera)

	$$RGR_\text{mass} = \frac{\text{ln} M_2 - \text{ln} M_1}{t_2 - t_1}$$

\section{Determining ecological gradients that matter}

Performance tradeoffs
Isolation by ecology

\section{Tests of divergent selection}

	\subsection{Trait-environment correlation}

	We will account for relatedness by using genetic distance matrix in linear models.

	\subsection{$P_\text{ST}$ -- $F_\text{ST}$}

	Contrast trait classes with Fst distribution

	\subsection{Isolation by ecology}

	Is \textbf{G} better predicted by \textbf{E} than geographic distance? Use Gideon's method?

	\subsection{Performance tradeoffs between treatments}


\section{Ecological strategy, physiological mechanism, and tradeoff}

	\subsection{Drought escape}
		Pattern: \\
			Flowering time shorter in drier/S populations \\
			More between population variation in flowering time more variable than predicted by Fst \\
			... \\

	\subsection{Drought avoidance}
		Pattern: \\
			Maybe this should really be dehydration avoidance? \\
			Populations from drier/S avoid dehydration by limiting water loss; more variable than predicted by Fst \\

			Mechanism \\
			closing stomata earlier during soil drying

		Mechanisms: \\
			lower water-use \\

	\subsection{Drought tolerance}
		Pattern: \\
			drier/S populations have greater tolerance to low tissue water content

		Mechanisms: \\
			greater WUE \\
			ability to photosynthesize during drought \\

\section{Extension: incipient speciation as a byproduct or divergence?}

Collect pollen to measure viability.


\end{document}
