\documentclass[11pt, oneside]{article}

%
% Packages
%

\usepackage{geometry}
\geometry{letterpaper}
\usepackage[parfill]{parskip}      		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\usepackage{booktabs}
\usepackage{topcapt}
\usepackage[labelfont =  {sf, bf}, textfont = sf, format = plain]{caption}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{color}
\usepackage{array}
\usepackage{gensymb}
\usepackage{setspace}
\newcommand{\stretchy}{1.5}
\usepackage{lineno}
\usepackage{textcomp}
\usepackage{cleveref}
\usepackage{subcaption}

% Make helvetica the default sans-serif font
% \renewcommand\sfdefault{phv}

% Package command for citing R packages
\newcommand{\pkg}[1]{{\fontseries{b}\selectfont #1}} 

% Setup

<<Setup, echo=FALSE, results=hide>>=

  # Manuscript files
  pathMS <- getwd() # for sweave
  # pathMS <- "ms" # for debugging
  
  # Data files for paper
  pathDat <- paste(pathMS, "/Data", sep = "")

  # Figures for paper
  pathFig <- paste(pathMS, "/Figures", sep = "")

  # Tables for paper
  pathTab <- paste(pathMS, "/Tables", sep = "")

  # Source custom functions
  source("functions.R") # for sweave
  # source(paste(pathMS, "/functions.R", sep = "")) # for debugging
  
  # Color palette
  # Primary color:

  primShade0 <- rgb(1, 0, 0.286)
  primShade1 <- rgb(1, 0.643, 0.745)
  primShade2 <- rgb(1, 0.439, 0.6)
  primShade3 <- rgb(0.765, 0, 0.22)
  primShade4 <- rgb(0.588, 0, 0.169)

  # Secondary color (1):

  sec1shade0 <- rgb(0.008, 0.957, 1)
  sec1shade1 <- rgb(0.643, 0.984, 1)
  sec1shade2 <- rgb(0.443, 0.976, 1)
  sec1shade3 <- rgb(0, 0.49, 0.514)
  sec1shade4 <- rgb(0, 0.376, 0.396)

  # Secondary color (2):

  sec2shade0 <- rgb(0.988, 1, 0)
  sec2shade1 <- rgb(0.996, 1, 0.643)
  sec2shade2 <- rgb(0.992, 1, 0.439)
  sec2shade3 <- rgb(0.835, 0.847, 0)
  sec2shade4 <- rgb(0.643, 0.651, 0)

@

%
% Title and authors
%

\title{Grow with the flow: a latitudinal cline in physiology is associated with more variable precipitation in \textit{Erythranthe cardinalis}}
\author{Christopher D. Muir$^{1,*}$ and Amy L. Angert$^1$}
\date{}

%
% Start document
%

\begin{document}
\SweaveOpts{concordance=TRUE}
\maketitle

$^1$ Biodiversity Research Centre, University of British Columbia, Vancouver, BC, Canada \\
$^*$corresponding author: Chris Muir, cdmuir@biodiversity.ubc.ca \\

Running Head: Latitudinal cline and climate in \textit{Erythranthe} \\

Key words: local adaptation, cline, photosynthesis, growth rate, monkeyflower, \textit{Mimulus} \\

Data will be archived on Dryad upon publication. \\

%--------------------------------------------------
% Acknowledgements
%--------------------------------------------------

\section*{Acknowledgements}
Erin Warkman and Lisa Lin helped collect data. CDM was supported by a Biodiversity Postdoctoral Fellowship funded by the NSERC CREATE program. ALA was supported by an NSERC Discovery Grant and a grant from the National Science Foundation (DEB 0950171). Two anonymous referees provided constructive comments on earlier versions of this manuscript.

\clearpage

\setstretch{\stretchy}

\begin{center}
{\LARGE  \centering Grow with the flow: a latitudinal cline in physiology is associated with more variable precipitation in \textit{Erythranthe cardinalis}}
\end{center}

\section*{Abstract}

Local adaptation is commonly observed in nature: organisms perform well in their natal environment, but poorly outside it. Correlations between traits and latitude, or latitudinal clines, are among the most common pieces of evidence for local adaptation, but identifying the traits under selection and the selective agents is challenging. Here, we investigated a latitudinal cline in growth and photosynthesis across 16 populations of the perennial herb \textit{Erythranthe cardinalis} (Phrymaceae). Using machine learning methods, we identify interannual variation in precipitation as a likely selective agent: Southern populations from more variable environments had higher photosynthetic rates and grew faster. We hypothesize that selection may favor a more annualized life history -- grow now rather than save for next year -- in environments where severe droughts occur more often. Thus our study provides insight into how species may adapt if Mediterranean climates become more variable due to climate change.

\section*{Introduction}

% Toggle these commands to get version for doing word count
\linenumbers
% \pagenumbering{gobble}
% \raggedright

% Local adaptation and clines
Local adaptation has been documented within numerous species; populations generally have higher fitness in their native environment, but perform poorly outside it \citep{Schluter_2000, Leimu_Fischer_2008, Hereford_2009}. However, the prevalance of local adaptation remains difficult to assess because researchers rarely test for local adaptation unless there are obvious phenotypic or environmental differences (but see \citeauthor{Hereford_Winn_2008} \citeyear{Hereford_Winn_2008}). When local adaptation occurs, it frequently leads to clines in both phenotypes and allele frequencies when selection varies over environmental gradients \citep{Huxley_1938, Endler_1977, Barton_1999}. Phenotypic differences between populations along a cline often have a genetic basis and can be studied in a common garden \citep{Turesson_1922, Clausen_etal_1940, Hiesey_etal_1942}. Despite a long history of studying local adaptation and clines, it remains challenging to identify exactly which traits are under selection and which differ for nonadaptive reasons. In particular, the role that physiological differences play in local adaptation is poorly understood, despite the fact that physiology is frequently assumed to explain adaptation to the abiotic environment. A related problem is identifying which of the myriad and often covarying aspects of the environment causes spatially varying selective pressures. 

% Intrinsic versus plastic variation 
When populations are locally adapted, reaction norms for fitness will cross, such that local genotypes have higher fitness than foreign genotypes and rank orders change across environments \citep{Kawecki_Ebert_2004}. The traits that underlie local adaptation, however, need not mirror this pattern. Populations can have fixed genetic differences conferring trait values that are adaptive at home but neutral or maladaptive away. Alternatively, the ability to plastically respond to a particular environment or the magnitude of response to an environment could be adaptive. We distinguish between these patterns of adaptive trait differences by referring to `genetic variation' in trait means and `genetic variation in plasticity', respectively. Genetic variation in plasticity is synonymous with genotype-by-environment interactions, or simply (G$\times$E). Genetic variation in trait means and plasticity are both involved in adaptation. For example, genetic variation in photoperiod responses \citep{Blackman_etal_2011} and developmental rate \citep{Stinchcombe_etal_2004} allow organisms to properly time their life history with the local environment. Conversely, sun and shade plants do not have intrinsically higher or lower rates of carbon assimilation, but rather, genetic variation in plasticity cause sun plants to assimilate more under high light and shade plants under low light \citep{Givnish_1988}. In plants especially, we know little about the prevalence and adaptive significance of variation in fundamental physiological traits like photosynthesis and their impact on plant performance \citep{Flood_etal_2011}.

% Selective agents (trait-enviro assoc)
A basic approach to identify candidate traits underlying local adaptation is to find associations between traits and environments. Either genetic variation in trait means and/or plasticity should vary clinally along environmental gradients. Indeed, clines in ecologically important traits are widespread in nature \citep{Endler_1977} and often adaptive, but in most cases the selective agent is unknown. For example, in \textit{Drosophila} numerous latitudinal clines exist for traits like thermal tolerance \citep{Hoffmann_etal_2002}, body size (\cite{Coyne_Beecham_1987} and references therein), and life history \citep{Schmidt_etal_2005}. Some \textit{Drosophila} clines have evolved multiple times (\cite{Oakeshott_etal_1982, Huey_etal_2000}, see also \cite{Bradshaw_Holzapfel_2001}) or shifted in response to climate change \citep{Umina_etal_2005}, evincing climatic adaptation. Similarly, plant species exhibit latitudinal clines in traits like flowering time \citep{Stinchcombe_etal_2004}, cyanogenesis \citep{Kooyers_Olsen_2012}, leaf morphology \citep{Hopkins_etal_2008, Stock_etal_2014}, and drought resistance \citep{Kooyers_etal_2015} that likely relate to climatic variation. 

Despite the fact that latitudinal clines have been studied for a long time, latitude \textit{per se} cannot be a selective agent. Latitude may be strongly correlated with one or two key climatic variables, such as temperature, precipitation, or growing degree-days. Latitude may also correlate with the strength of biotic interactions \citep{Schemske_etal_2009} or other nonclimatic aspects of the environment, though as we explain below, we do not yet have compelling data that these are important in our study system. Hence, we focus on whether latitude could be an effective proxy for an underlying climatic driver, in which case we would expect a yet stronger relationship between traits and the key climatic variable(s) driving selection. Alternatively, latitude may be more strongly related to traits than any single climatic variable for at least two reasons. First, latitude may be correlated with several climatic agents of selection that are individually weak, but add up to a strong latitudinal cline. Alternatively, gene flow among neighbouring populations could smooth out local climatic effects, since alleles will experience selection across populations linked by migration \citep{Slatkin_1978, Paul_etal_2011, Hadfield_2016}. We refer to this as the `climatic neighborhood'. For example, in mountainous regions average temperature at a given latitude varies widely, but in aggregate, a lower latitude set of populations will experience warmer climate than a higher latitude one. Thus, any particular low latitude population would be warm-adapted, even if it was located in a cooler (e.g. high elevation) site. Because many climatic factors vary latitudinally, and which climatic factors vary latitudinally changes over the earth's surface (e.g. coastal vs. continental), dissecting the evolution of latitudinal clines across many species will help identify generalities, such as whether thermal tolerance maxima or seasonal timing is more important \citep{Bradshaw_Holzapfel_2008}, and whether local or regional climate shapes selective pressures.

In this study, we investigated two major questions: 1) whether genetic variation in physiological trait means or plasticity corresponds with latitude; and 2) what climatic factor(s) could plausibly be responsible for latitudinal clines. Within question 2, we tested three hypotheses outlined in the previous paragraph: latitudinal clines are explained by a single dominant climatic factor, multiple climatic factors, or the climatic neighborhood experienced by nearby population connected through gene flow. These hypotheses are not mutually exclusive since, for example, single or multiple factors in a climatic neighborhood may lead to latitudinal clines. We focused on climate because climate often determines and where species are found and also can exert strong selection on populations within species, though we acknowledge that other abiotic and biotic factors could also contribute to selection and the overall pattern of local adaptation. There is also a compelling need to know how populations are (or are not) locally adapted to climate so as to predict how they will respond to climate change \citep{Aitken_Whitlock_2013}. 

We examined these questions in \textit{Erythranthe cardinalis} (formerly \textit{Mimulus cardinalis} [\citeauthor{Nesom_2014} \citeyear{Nesom_2014}]) because linking physiological traits to potentially complex patterns of local adaptation requires integrating multiple lines of evidence from comparative, experimental, and genomic studies under both lab and field conditions. Many classic and contemporary studies of local adaptation use \textit{Mimulus sensu lato} species because of their natural history, easy propagation, and genetic/genomic resources \citep{Clausen_etal_1940, Hiesey_etal_1971, Bradshaw_Schemske_2003, Wu_etal_2008, Lowry_Willis_2010, Wright_etal_2013}. Yet, there is a deficiency of links between local adaptation and physiological mechanisms \citep{Angert_2006, Angert_etal_2008, Wu_etal_2010, Wright_etal_2013}. We measured genetic variation in trait means and plasticity in response to temperature and drought among 16 populations distributed over 10.7\textdegree of latitude. We found a latitudinal cline of trait means in photosynthesis and growth, but little evidence for variation in plasticity. Interannual variation in precipitation and temperature are associated with this axis of variation, suggesting that climatic variance rather than mean may be an important driver of local adaptation in \textit{E. cardinalis}. The climatic neighborhoods around populations explained trait variation better than local climate, indicating that latitudinal clines may be common because latitude integrates effects of selection on populations connected through gene flow. We place these findings in the context of life history theory and consider future directions in the Discussion. 

\section*{Material and Methods}

Data and annotated source code to reproduce these analyses and manuscript are available on GitHub (https://github.com/cdmuir/card-cline).

<<Summary Statistics, echo = FALSE, results=hide, warning=FALSE>>=

  master <- read.csv(paste(pathDat, "/MasterDatasheet_out.csv", sep = ""), 
    row.names = 1)

  # Number of families per population
  famTab <- tapply(master$Family, list(master$PopID), function(X) length(unique(X)))
  
  # Individuals per population per treatment
  indPerPop <- tapply(master$indiv, list(master$TempTrt, master$WaterTrt, 
                                         master$PopID), length)
  
@

\subsection*{Population Selection}

\textit{E. cardinalis} is a perennial forb native to the Western US (California and Oregon). It is predomintantly outcrossing, self-compatible, and pollinated primarily by hummingbirds. We used 16 populations from throughout the range of \textit{E. cardinalis} (Table~\ref{table:Table_FocalPops}). These populations were intentionally chosen to span much of the climatic range of the species based on all known occurrences (see below). Seeds were collected in the field from mature, undehisced fruit left open for 2-4 weeks to dry, then stored at room temperature. To control for maternal effects, we grew a large number of field-derived seeds in the greenhouse and generated seed families for this experiment by haphazardly crossing individuals from the same population. We selected seed families to maximize the number of field-derived individuals represented. Thus, we used seeds from \Sexpr{nlevels(master$Family)} greenhouse-derived seed families, \Sexpr{min(famTab)}--\Sexpr{max(famTab)} (mean = \Sexpr{round(mean(famTab), 1)}, median = \Sexpr{median(famTab)}) families per population.

<<focClim, echo = FALSE, results=hide, warning=FALSE>>=

# Read in climate data for 16 focal populations to use in making Table 1
load(paste(pathDat, "/focClim.RData", sep = ""))

@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Table of focal population
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{table}[ht]
   \centering
   \topcaption[Focal Populations]{Latitude, longitude, and elevation (mas = meters above seal level) of 16 focal populations used in this study.}
   \begin{tabular}{@{} llll @{}}
      \toprule
  Name  & Latitude  & Longtiude  & Elevation (mas) \\
      \midrule
	Hauser Creek & \Sexpr{round(focClim$Latitude[focClim$PopID == "HAU"], 3)}	& 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "HAU"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "HAU"]}   \\
	Cottonwood Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "CTC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "CTC"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "CTC"]}   \\
	Sweetwater River	& \Sexpr{round(focClim$Latitude[focClim$PopID == "CUR"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "CUR"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "CUR"]}   \\
	Grade Road Palomar & \Sexpr{round(focClim$Latitude[focClim$PopID == "GRP"], 3)} &
    \Sexpr{round(focClim$Longitude[focClim$PopID == "GRP"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "GRP"]}   \\
	Whitewater Canyon & \Sexpr{round(focClim$Latitude[focClim$PopID == "WWC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "WWC"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "WWC"]}   \\
	Mill Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "MIL"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "MIL"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "MIL"]}   \\
	West Fork Mojave River	& \Sexpr{round(focClim$Latitude[focClim$PopID == "WFM"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "WFM"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "WFM"]}   \\
	North Fork Middle Tule River	& \Sexpr{round(focClim$Latitude[focClim$PopID == "NMT"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "NMT"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "NMT"]}   \\
	Paradise Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "PRD"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "PRD"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "PRD"]}   \\
	Redwood Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "RWD"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "RWD"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "RWD"]}   \\
	Wawona  & \Sexpr{round(focClim$Latitude[focClim$PopID == "WNA"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "WNA"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "WNA"]}   \\
	Rainbow Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "RBW"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "RBW"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "RBW"]}   \\
	Middle Yuba River	& \Sexpr{round(focClim$Latitude[focClim$PopID == "MYU"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "MYU"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "MYU"]}   \\
	Little Jamison Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "LIJ"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "LIJ"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "LIJ"]}   \\
	Deep Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "DPC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "DPC"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "DPC"]}   \\
	Rock Creek	& \Sexpr{round(focClim$Latitude[focClim$PopID == "RCC"], 3)} & 
    \Sexpr{round(focClim$Longitude[focClim$PopID == "RCC"], 3)}	& \Sexpr{focClim$Elevation[focClim$PopID == "RCC"]}   \\
	\bottomrule
	\end{tabular}
	\label{table:Table_FocalPops}
\end{table}

\subsection*{Plant propagation}

On 14 April, 2014, 3-5 seeds per family were sown directly on sand (Quikrete Play Sand, Georgia, USA) watered to field capacity in RLC4 Ray Leach cone-tainers placed in RL98 98-well trays (Stuewe \& Sons, Inc., Oregon, USA). We used pure sand because \textit{E. cardinalis} typically grows in sandy, riparian soils (A. Angert, pers. obs.). Two jumbo-sized cotton balls at the bottom of cone-tainers prevented sand from washing out. Cone-tainers sat in medium-sized flow trays (FLOWTMD, Stuewe \& Sons, Inc., Oregon, USA) to continuously bottom-water plants during germination in greenhouses at the University British Columbia campus in Vancouver, Canada (49\degree 15' N, 123\degree 15' W). Misters thoroughly wetted the top of the sand every two hours during the day. Most seeds germinated between 1 and 2 weeks, but we allowed 3 weeks before transferring seedlings to growth chambers. We recorded germination daily between one to two weeks after sowing, and every 2-3 days thereafter. On 5 May (21 days after sowing), we transferred seedlings to one of two growth chambers (Model E-15 Conviron, Manitoba, Canada). We thinned seedlings to one plant per cone-tainer, leaving the center-most plant. \Sexpr{nrow(master)} of 768 (\Sexpr{round(100 * nrow(master) / 768, 1)}\%) had plants that could be used in the experiment. We allowed one week at constant, non stressful conditions (day: 20\celsius, night: 16\celsius) for plants to acclimate to growth chambers before starting treatments. The initial size of seedlings, measured as the length of the first true leaves, did not differ between populations, families, or treatments (Table~\ref{table:TableS_InitialSize}).
    
\subsection*{Temperature and drought treatments}

We imposed four treatments, a fully-factorial cross of two temperature levels and two watering levels. The temperature levels closely simulated an average growing season at the thermal extremes of the species range, which we designate as Hot and Cool treatments. Watering levels contrasted a perennial and seasonal stream, which we refer to as Well-watered and Drought treatments. A detailed description of treatments is provided in the Supplemental Materials and Methods and summarized in Fig~\ref{fig:Fig_ExptlDes}. Because growth chambers cannot be subdivided, one chamber was assigned to the Hot treatment level and another to the Cool treatment level. Within each chamber, there were two Well-watered blocks and two Drought blocks. The photosynthetically active radiation in both chambers was approximately 400 $\mu$mol quanta m$^{-2}$ s$^{-1}$ and set to a 16:8 light:dark cycle to simulate summer growing conditions. The growth chambers did not control humidity, but because of watering and high plant transpiration rates, the relative humidity was quite high in both temperature levels (data not shown). Lower humidity would have made the drought more severe, but low soil moisture is stressful in and of itself. The total number of plants in each treatment was: $n_{\text{cool,dry}} = \Sexpr{sum(indPerPop["Cool", "Dry", ])}$; $n_{\text{cool,ww}} = \Sexpr{sum(indPerPop["Cool", "Wet", ])}$; $n_{\text{hot,dry}} = \Sexpr{sum(indPerPop["Hot", "Dry", ])}$; $n_{\text{hot,ww}} = \Sexpr{sum(indPerPop["Hot", "Wet", ])}$. Each population had \Sexpr{min(indPerPop)}--\Sexpr{max(indPerPop)} individuals per treatment level (mean = \Sexpr{round(mean(indPerPop), 1)}, median = \Sexpr{median(indPerPop)}).

<<ExperimentalDesign, echo=FALSE, results=hide, warning=FALSE>>=
library(scales) # for color transparency
exptlDes <- read.csv(paste(pathDat, "/ExperimentalDesign.csv", sep = ""))

pdf(paste(pathFig, "/Figure_ExptlDes.pdf", sep = ""), 9, 5)
layout(matrix(c(rep(1, 7), rep(2, 2)), nrow = 1))
par(mai = rep(1, 4))
plot(0, 0, type = "n", xlim = c(0, max(exptlDes$DayN)), ylim = c(-1, 41), 
		 axes = F, frame.plot = T, xlab = "", ylab = "")
mtext("Key treatments and measurements", 1, cex = 1.5, line = 5, las = 1)

# Temperature axis
axis(2, at = seq(0, 40, 10), lwd = 0, lwd.ticks = 1, las = 1, cex.axis = 1.5,
	labels = c(expression(0*degree), expression(10*degree), expression(20*degree),
	           expression(30*degree), expression(40*degree)))
mtext(expression(paste("Temperature (", phantom()*degree, "C)")), 2, 
	cex = 1.5, line = 4, las = 3)

# Watering axis
axis(4, at = c(0, 40), lwd = 0, lwd.ticks = 1, las = 1, labels = c(1, 12), 
     cex.axis = 1.5)
mtext(expression(paste("Watering Freq (no. Day", phantom()^-1, ")")), 4,
      cex = 1.5, line = 5, las = 3)

# Axis of time on top
mtext("Days after sowing", side = 3, line = 5, cex = 1.5)
mtext("(Date for climate simulation)", side = 3, line = 3, cex = 1.5)
axis(3, at = seq(0, 90, 30), lwd = 0, lwd.ticks = 1, cex.axis = 1.5)
axis(3, at = c(17, 47, 78), lwd = 0, lwd.ticks = 1, line = -0.5, cex.axis = 1.5, 
     labels = c("(June)", "(July)", "(August)"))

# Indicate sowing, beginning of temperature and drought treatments
abline(v = 0, col = "grey", lwd = 2, lty = 2)
mtext("Sow", side = 1, at = 0, line = 1)
abline(v = 17, col = "grey", lwd = 2, lty = 2)
mtext("Temp\nTreatment", side = 1, at = 17, line = 2)
abline(v = 37, col = "grey", lwd = 2, lty = 2)
mtext("Drought\nTreatment", side = 1, at = 37, line = 2)

# Polygon of Hot treatment, min and max
with(exptlDes[18:94, ], polygon(c(17:93, 93:17), c(Tmin_hot, rev(Tmax_hot)),
  border = NA, col = alpha(primShade0, 0.25)))

# Polygon of Cool treatment, min and max
with(exptlDes[18:94, ], polygon(c(17:93, 93:17), c(Tmin_cool, rev(Tmax_cool)),
  border = NA, col = alpha(sec1shade0, 0.25)))

# Lines with average temperature
with(exptlDes[18:94, ], points(17:93, Tavg_hot, lwd = 2, type = "l", 
	col = primShade3))
with(exptlDes[18:94, ], points(17:93, Tavg_cool, lwd = 2, type = "l", 
	col = sec1shade3))

# Rug indicating measurment timing
rect(6, -2, 17, 0, border = "white", col = "grey")
with(exptlDes, text(11.5, -1, labels = "Germ", col = "black", font = 3, 
  cex = 0.9))

with(exptlDes, rect(min(which(LERmeas)), -2, max(which(LERmeas)), 0, 
	border = "white", col = "grey"))
with(exptlDes, text(min((which(LERmeas)) + max(which(LERmeas))) / 2, -1, 
	labels = "LER", col = "black", font = 3, cex = 0.9))

with(exptlDes, rect(min(which(SERmeas)), 0, max(which(SERmeas)), 2,
	border = "white", col = "grey"))
with(exptlDes, text(min((which(SERmeas)) + max(which(SERmeas))) / 2 - 5, 1, 
	labels = "SER", col = "black", font = 3, cex = 0.9))

with(exptlDes, rect(min(which(Harvest)), -2, max(which(Harvest)), 0, 
	border = "white", col = "grey"))
with(exptlDes, text(min((which(Harvest)) + max(which(Harvest))) / 2, -1, 
	labels = "Harvest", col = "black", font = 3, cex = 0.9))

with(exptlDes, rect(min(which(PhotoMeas)), 2, max(which(PhotoMeas)), 4, 
	border = "white", col = "grey"))
with(exptlDes, text(min((which(PhotoMeas)) + max(which(PhotoMeas))) / 2, 5, 
	labels = "Photo", col = "black", font = 3, cex = 0.9))

# Line of Drought treatment
# Rescale watering frequency to vary from 0 to 40
exptlDes$WaterFreq_dry <- exptlDes$WaterFreq_dry * (40 / 11) - 40 / 11

with(exptlDes, points(DayN, WaterFreq_dry, type = "l", lwd = 3, col = "black",
                      lty = 2))
lines(range(exptlDes$DayN), rep(40, 2), lwd = 3)

# Legend
# pdf(paste(pathFig, "/ExperimentalDesignLegend.pdf", sep = ""), 2, 2)
par(mai = c(2, 0, 1, 0))
plot(0, 0, type = "n", xlim = c(-0.5, 1.5), ylim = c(0, 1), axes = F, xlab = "")

mtext("Legend for\ntreatments", 3, line = 1)
axis(1, at = c(0, 0.5, 1), labels = c("n", "n + 1", "n + 2"),
     font = 3)
mtext("Key to\nabbreviations\n\nGerm = Germination\nLER = Leaf expansion rate\nSER = Stem elongation rate\nPhoto = Photosynthesis", 1, line = 12, adj = 0, at = -0.75)

# Temperature
polygon(c(0, 0.5, 1, 1, 0.5, 0), c(0.8, 1, 1, 0.55, 0.55, 0.35), border = NA, 
				col = alpha(primShade0, 0.25))
polygon(c(0, 0.5, 1, 1, 0.5, 0), c(0.6, 0.45, 0.45, 0, 0, 0.15), border = NA, 
				col = alpha(sec1shade0, 0.25))
points(c(0, 0.5, 1), c(0.575, 0.775, 0.775), lwd = 2, type = "l", col = primShade3)
points(c(0, 0.5, 1), c(0.375, 0.225, 0.225), lwd = 2, type = "l", col = sec1shade3)

# Drought
lines(c(0, 1), c(1, 1), lwd = 2, lty = 1)
lines(c(0, 1), c(1, 0), lwd = 2, lty = 2)
text(-0.25, 0.95, labels = "Water\nFreq", cex = 1.5)
text(0.5, 0.9, labels = "Hot", col = primShade3, cex = 1.5)
text(0.5, 0.1, labels = "Cool", col = sec1shade3, cex = 1.5)
text(1.25, 1, labels = "Max", col = alpha(primShade0, 0.25), cex = 1.5)
text(1.25, 0.55, labels = "Min", col = alpha(primShade0, 0.25), cex = 1.5)
text(1.25, 0.775, labels = "Mean", col = primShade3, cex = 1.5)

text(1.25, 0, labels = "Min", col = alpha(sec1shade0, 0.25), cex = 1.5)
text(1.25, 0.45, labels = "Max", col = alpha(sec1shade0, 0.25), cex = 1.5)
text(1.25, 0.225, labels = "Mean", col = sec1shade3, cex = 1.5)

dev.off()

@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure summarizing experimental design
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_ExptlDes.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Experimental Design]{Overview of experimental treatments and timing of key trait measurements. All plants germinated within 21 days of sowing. At that time, we began temperature treatments (left axis), simulating a typical June-August weather pattern at Hot (red) and Cool (blue) sites. The bold lines track the average daily temperatures. Within each day, there was a maximum daytime temperature (top of translucent polygons) and minimum nighttime temperature (bottom of translucent polygons). The drought treatment commenced later by ramping down the frequency of bottom-watering episodes (dashed black line; right axis), while watering frequency was maintained in the control treatment (solid black line). Grey boxes on the bottom of the plot outline the period of key measurements described in the Material and Methods.}
	\label{fig:Fig_ExptlDes}
\end{figure}

\subsection*{Trait measurements}

We measured five traits in response to temperature and watering treatments (Table~\ref{table:Table_traits}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Table of key traits
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{table}[ht]
   \centering
   \topcaption{Key traits measured in this study.}
   \begin{tabular}{@{} ll @{}}
      \toprule
  Trait & Units \\
      \midrule
  Days to germination  & day \\
  Leaf expansion rate  &  mm day$^{-1}$  \\
  Stem elongation rate  &  cm day$^{-1}$  \\
  Photosynthetic rate &  $\mu$mol CO$_2$ m$^{-2}$ s$^{-1}$\\
  Mortality & probability of death  \\
	    \bottomrule
   \end{tabular}
   \label{table:Table_traits}
\end{table}

\paragraph{Days to germination} We tested for population variation in germination rate, measured as Days to Germination, using a lognormal survival model fit using the survreg function in the R package \pkg{survival} version 2.38 \citep{Therneau_2015}. We treated Population as a fixed effect and Family as random effect using a $\Gamma$ frailty function. Statistical significance of the Population effect was determined using analysis of deviance. Note that, unlike other traits discussed below, we did not include Block, Treatment, or Population $\times$ Treatment interactions because during germination plants had not been placed into blocks and treatments had not yet been applied.

<<Germination rate, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  library(survival)

  master <- read.csv(paste(pathDat, "/MasterDatasheet_out.csv", sep = ""), 
    row.names = 1)

  # Create Surv object
	germ <- with(subset(master, !is.na(master$MinGermDay)), 
	             Surv(MinGermDay, MaxGermDay,	type = "interval2"))

	# lognormal produced best fit to data
	fitGerm <- survreg(germ ~ PopID + frailty(Family, sparse = F), 
		data = subset(master, !is.na(master$MinGermDay)), dist = "lognormal")
	suppressWarnings(anova(fitGerm)) # significant family and population effects
	
@

\paragraph{Growth rate: leaf expansion and stem elongation}

We measured growth rate during two phases: leaf expansion and stem elongation. Growth measurements were taken during the early vegetative stage. We censused leaf length twice per week shortly after the emergence of true leaves from 12 May -- 12 June (28--59 days after sowing), resulting in 10 measurements. We ceased measuring leaf length once it appeared to asymptote and growth shifted to stem elongation. We also censused plant height on 7 occasions (twice per week) between 29 May and 20 June (45 to 67 days after sowing) until plants began to initiate floral buds. Thus all growth measurements occured during the vegetative, prereproductive phase. Both leaf expansion and stem elongation were modelled separately as second-order polynomials. We used empirical Bayes' estimates of growth for each individual plant from linear mixed-effects models fit with the R package \pkg{lme4} version 1.1-12 \citep{Bates_etal_2015}.

<<Leaf expansion rate, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  library(lme4)
  library(lmerTest)

	# Note that growth data used in the mixed model are from the master datasheet.
	# The code below shows how growth rates were calculated.

  # New (blank) datasheet on which to add data
  newdata <- read.csv(paste(pathDat, "/MasterDatasheet_in.csv", sep = ""))

  # Read in LER data
  LERdata <- read.csv(paste(pathDat, "/LERdata.csv", sep = ""))

  #
	# Empirical Bayes Approach: get coefficients for every plant
	#

	# suppressWarnings(mmLER <- lmer(log(LLL) ~ poly(DayN, 2, raw = T) + 
    # (poly(DayN, 2, raw = T)|indiv), data = LERdata))
  # save(mmLER, file = paste(pathDat, "/mmLER.RData", sep = ""))
  load(file = paste(pathDat, "/mmLER.RData", sep = ""))

  # Extract coefficients and add to datasheet
	Y <- t(unlist(fixef(mmLER)) + t(ranef(mmLER)$indiv))[match(newdata$indiv, 
		rownames(ranef(mmLER)$indiv)), ]
	colnames(Y) <- c("b0_LER", "b1_LER", "b2_LER")
  newdata <- cbind(newdata, Y)

  # Model-based LER
	# Timeframe for Cool treatment: Day 0 - 31
	# Timeframe for Hot treatment: Day 0 - 24
	newdata$LERstart <- exp(newdata$b0_LER)
	newdata$LERend <- exp(newdata$b0_LER + 
					newdata$b1_LER * ifelse(newdata$TempTrt == "Cool", 31, 24) + 
					newdata$b2_LER * ifelse(newdata$TempTrt == "Cool", 31 ^ 2, 24 ^ 2))
	newdata$LER_AbsGrowth <- (newdata$LERend - newdata$LERstart) / 
		ifelse(newdata$TempTrt == "Cool", 31, 24)

  # Mixed model ANOVA with family as random factor fit using step down procedure
	mm <- lmer(LER_AbsGrowth ~ AvgGermDay + PopID * TempTrt * WaterTrt + 
						 	(1|Family) + (1|Block), data = master)
	fitLER <- step(mm, reduce.random = F) # includes pop, temp, h2o, population x water

	# Figure of reaction norms
  tab <- tapply(newdata$LER_AbsGrowth, 
                list(newdata$Population, newdata$WaterTrt, newdata$TempTrt), mean)
  pdf(paste(pathFig, "/FigureS_LERrxnNorm.pdf", sep = ""), 4, 4, useDingbats = F)
  par(mai = c(0.5, 1, 0.5, 0.25))
  plot(0, 0, type = "n", xlab = "", ylab = "", axes = F, frame.plot = T, 
       xlim = c(1, 4), ylim = c(1, 3.5))
  abline(v = 2.5)
  mtext(side = 3, c("Cool", "Hot"), at = c(1.625, 3.375))
  mtext(side = 1, c("Wet", "Dry", "Wet", "Dry"), at = c(1, 2.25, 2.75, 4))
  mtext(side = 1, "Water Treatment", font = 2, cex = 1.25, line = 1.5)
  mtext(side = 3, "Temperature Treatment", font = 2, cex = 1.25, line = 1.5)
  axis(2, lwd = 0, lwd.ticks = 1, las = 1)
  title(ylab = expression(paste("Leaf expansion rate (mm ", day ^-1, ")")), 
        cex.lab = 1.5)

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.75), c(tab[i, "Wet", "Cool"], tab[i, "Wet", "Hot"]), col = "grey")
    lines(c(2.25, 4), c(tab[i, "Dry", "Cool"], tab[i, "Dry", "Hot"]), col = "grey",
          lty = 2)
  }

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.25), c(tab[i, "Wet", "Cool"], tab[i, "Dry", "Cool"]), lty = 2)
    lines(c(2.75, 4), c(tab[i, "Wet", "Hot"], tab[i, "Dry", "Hot"]))
  }

  mtext(side = 3, "A", font = 2, cex = 2, at = grconvertX(-0.1, "npc", "user"))
  dev.off()
  
  # Confidence intervals on effect size (mean squared error)
  fn <- function(merMod) {
    x <- anova(merMod)
    ret <- x[c("PopID", "PopID:TempTrt", "PopID:WaterTrt"), "Mean Sq"]
    ret <- c(ret, x["PopID", "Mean Sq"] - x["PopID:TempTrt", "Mean Sq"],
             x["PopID", "Mean Sq"] - x["PopID:WaterTrt", "Mean Sq"])
    ret
  }

  ciMerModLER <- confint(mm, method = "boot", FUN = fn, nsim = 1e3, seed = 50717035,
                         parallel = "multicore", ncpus = 2L, level = 0.95)
  ciMerModLER <- as.data.frame(ciMerModLER)
  fit <- anova(mm)
  ciMerModLER$estimate <- c(fit[c("PopID", "PopID:TempTrt", "PopID:WaterTrt"), "Mean Sq"],
                            fit["PopID", "Mean Sq"] - fit["PopID:TempTrt", "Mean Sq"],
                            fit["PopID", "Mean Sq"] - fit["PopID:WaterTrt", "Mean Sq"])
  rownames(ciMerModLER) <- c("PopID", "PopID:TempTrt", "PopID:WaterTrt", 
                             "PopID - PopID:TempTrt", "PopID - PopID:WaterTrt")
  rm(fit)

@


<<Stem elongation rate, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

	# Note that growth data used in the mixed model are from the master datasheet.
	# The code below shows how growth rates were calculated.
	
  # Read in SER data
  SERdata <- read.csv(paste(pathDat, "/SERdata.csv", sep = ""))

  #
  # Empirical Bayes Approach: get coefficients for every plant
	#

	# suppressWarnings(mmSER <- lmer(log(height + 1) ~ poly(DayN, 2, raw = T) + 
    # (poly(DayN, 2, raw = T)|indiv), data = SERdata))
  # save(mmSER, file = paste(pathDat, "/mmSER.RData", sep = ""))
  load(file = paste(pathDat, "/mmSER.RData", sep = ""))

  # Extract coefficients and add to datasheet
  Y <- t(unlist(fixef(mmSER)) + t(ranef(mmSER)$indiv))[match(newdata$indiv, 
		rownames(ranef(mmSER)$indiv)), ]
	colnames(Y) <- c("b0_SER", "b1_SER", "b2_SER")
  newdata <- cbind(newdata, Y)

  # Model-based SER
	# Timeframe for Cool treatment: Day 0 - 31
	# Timeframe for Hot treatment: Day 0 - 24
	newdata$SERstart <- exp(newdata$b0_SER)
	newdata$SERend <- exp(newdata$b0_SER + 
					newdata$b1_SER * ifelse(newdata$TempTrt == "Cool", 31, 24) + 
					newdata$b2_SER * ifelse(newdata$TempTrt == "Cool", 31 ^ 2, 24 ^ 2))
	newdata$SER_AbsGrowth <- (newdata$SERend - newdata$SERstart) / 
		ifelse(newdata$TempTrt == "Cool", 31, 24)

	# Mixed model including LER_AbsGrowth as covariate
  # mm <- lmer(SER_AbsGrowth ~ AvgGermDay + LER_AbsGrowth + 
  #             PopID * TempTrt * WaterTrt + (1|Family) + (1|Block), 
  #           data = master)
	# fitSER <- step(mm, reduce.random = F) #includes LER_AbsGrowth, pop, temp, h2o, and temp x h2o

	# Mixed model without LER_AbsGrowth as covariate
  mm <- lmer(SER_AbsGrowth ~ AvgGermDay + 
               PopID * TempTrt * WaterTrt + (1|Family) + (1|Block), 
             data = master)
	fitSER <- step(mm, reduce.random = F) 

	# Figure of reaction norms
  tab <- tapply(newdata$SER_AbsGrowth, 
                list(newdata$Population, newdata$WaterTrt, newdata$TempTrt), mean, 
                na.rm = T)
  pdf(paste(pathFig, "/FigureS_SERrxnNorm.pdf", sep = ""), 4, 4, useDingbats = F)
  par(mai = c(0.5, 1, 0.5, 0.25))
  plot(0, 0, type = "n", xlab = "", ylab = "", axes = F, frame.plot = T, 
       xlim = c(1, 4), ylim = c(0.15, 0.50))
  abline(v = 2.5)
  mtext(side = 3, c("Cool", "Hot"), at = c(1.625, 3.375))
  mtext(side = 1, c("Wet", "Dry", "Wet", "Dry"), at = c(1, 2.25, 2.75, 4))
  mtext(side = 1, "Water Treatment", font = 2, cex = 1.25, line = 1.5)
  mtext(side = 3, "Temperature Treatment", font = 2, cex = 1.25, line = 1.5)
  axis(2, lwd = 0, lwd.ticks = 1, las = 1)
  title(ylab = expression(paste("Stem elongation rate (cm ", day ^-1, ")")), 
        cex.lab = 1.5)

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.75), c(tab[i, "Wet", "Cool"], tab[i, "Wet", "Hot"]), col = "grey")
    lines(c(2.25, 4), c(tab[i, "Dry", "Cool"], tab[i, "Dry", "Hot"]), col = "grey",
          lty = 2)
  }

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.25), c(tab[i, "Wet", "Cool"], tab[i, "Dry", "Cool"]), lty = 2)
    lines(c(2.75, 4), c(tab[i, "Wet", "Hot"], tab[i, "Dry", "Hot"]))
  }
 
  mtext(side = 3, "B", font = 2, cex = 2, at = grconvertX(-0.1, "npc", "user"))

  dev.off()

  # Confidence intervals on effect size (mean squared error)
  ciMerModSER <- confint(mm, method = "boot", FUN = fn, nsim = 1e3, seed = 46887099,
                         parallel = "multicore", ncpus = 2L)
  ciMerModSER <- as.data.frame(ciMerModSER)
  fit <- anova(mm)
  ciMerModSER$estimate <- c(fit[c("PopID", "PopID:TempTrt", "PopID:WaterTrt"), "Mean Sq"],
                            fit["PopID", "Mean Sq"] - fit["PopID:TempTrt", "Mean Sq"],
                            fit["PopID", "Mean Sq"] - fit["PopID:WaterTrt", "Mean Sq"])
  rownames(ciMerModSER) <- c("PopID", "PopID:TempTrt", "PopID:WaterTrt", 
                             "PopID - PopID:TempTrt", "PopID - PopID:WaterTrt")
  rm(fit)

@

\paragraph{Photosynthesis}
During the week of 10 to 16 June (57 to 63 days after sowing), we measured daytime photosynthetic rate on a subset of \Sexpr{length(which(!is.na(master$Photo)))} plants evenly spread between treatments and families within populations. The youngest, fully-expanded leaf acclimated for 3 minutes to reach steady state in a 6-cm$^2$ chamber of a LI-COR 6400XT Portable Photosynthesis System (LI-COR Biosciences, Lincoln, Nebraska). We made all measurements at ambient light (400 $\mu$mol m$^{-2}$ s$^{-1}$ of photosynthetically active radiation), atmospheric CO$_2$ (400 ppm), temperature, and moderate relative humidity. All measurements were taken between 9:00 AM and 5:00 PM (3 hours after lights turned on and 5 hours before lights turned off). During this period, we suspended normal day-to-day temperature fluctuations and set daytime temperatures to the average for that period (Cool: 26.5$\degree$; Hot: 36.1$\degree$) so that all plants within a temperature level could be measured under the same conditions. We were photosynthesis after dry down had progressed to assess differences in photosynthetic responses to drought.

<<Photosynthesis, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  # Photosynthetic rate (remove is satisfied w/ resPhoto analysis)
  mm <- lmer(Photo ~ PopID * TempTrt * WaterTrt + (1|Family) + (1|Block),
    data = subset(master, !is.na(master$Photo)))
    fitPhoto <- step(mm, reduce.random = F)

  # Figure of reaction norms
  tab <- tapply(master$Photo, 
                list(master$Population, master$WaterTrt, master$TempTrt), mean, 
                na.rm = T)
  pdf(paste(pathFig, "/FigureS_PhotoRxnNorm.pdf", sep = ""), 4, 4, useDingbats = F)
  par(mai = c(0.5, 1, 0.5, 0.25))
  plot(0, 0, type = "n", xlab = "", ylab = "", axes = F, frame.plot = T, xlim = c(1, 4), 
       ylim = c(5, 16))
  abline(v = 2.5)
  mtext(side = 3, c("Cool", "Hot"), at = c(1.625, 3.375))
  mtext(side = 1, c("Wet", "Dry", "Wet", "Dry"), at = c(1, 2.25, 2.75, 4))
  mtext(side = 1, "Water Treatment", font = 2, cex = 1.25, line = 1.5)
  mtext(side = 3, "Temperature Treatment", font = 2, cex = 1.25, line = 1.5)
  axis(2, lwd = 0, lwd.ticks = 1, las = 1)
  title(ylab = expression(paste("Photosynthesis (", mu, "mol ", CO[2]~m^-2~s^-1, ")")), 
        cex.lab = 1.5)

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.75), c(tab[i, "Wet", "Cool"], tab[i, "Wet", "Hot"]), col = "grey")
    lines(c(2.25, 4), c(tab[i, "Dry", "Cool"], tab[i, "Dry", "Hot"]), col = "grey", lty = 2)
  }

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.25), c(tab[i, "Wet", "Cool"], tab[i, "Dry", "Cool"]), lty = 2)
    lines(c(2.75, 4), c(tab[i, "Wet", "Hot"], tab[i, "Dry", "Hot"]))
  }

  mtext(side = 3, "C", font = 2, cex = 2, at = grconvertX(-0.1, "npc", "user"))

  dev.off()
  
  # Confidence intervals on effect size (mean squared error)
  ciMerModPhoto <- confint(mm, method = "boot", FUN = fn, nsim = 1e3, seed = 12657909,
                           parallel = "multicore", ncpus = 2L)
  ciMerModPhoto <- as.data.frame(ciMerModPhoto)
  fit <- anova(mm)
  ciMerModPhoto$estimate <- c(fit[c("PopID", "PopID:TempTrt", "PopID:WaterTrt"), "Mean Sq"],
                              fit["PopID", "Mean Sq"] - fit["PopID:TempTrt", "Mean Sq"],
                              fit["PopID", "Mean Sq"] - fit["PopID:WaterTrt", "Mean Sq"])
  rownames(ciMerModPhoto) <- c("PopID", "PopID:TempTrt", "PopID:WaterTrt", 
                               "PopID - PopID:TempTrt", "PopID - PopID:WaterTrt")
  rm(fit)
  
  
@

\paragraph{Mortality}
We assayed mortality during twice-weekly growth measurements. We analyzed the probability of surviving until the end of the experiment as a function of population, treatment, and their interactions using a Generalized Linear Mixed Model (GLMM) assuming binomially distributed errors. We included Family and Block as random effects. We assessed significance of fixed effects using Type-II Analysis of Deviance with Wald $\chi ^2$ tests in the R package \pkg{car} \citep{Fox_Weisberg_2011}. 

<<Mortality, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  library(car)

  fitMort <- glmer(DiedFromStress ~ PopID * TempTrt * WaterTrt + (1|Family) +
  								 (1|Block), control = glmerControl(optimizer = "bobyqa"),
  								 data = subset(master, !is.na(master$DiedFromStress)), 
  								 family = "binomial")
  # Anova(fitMort, type = 2)

  fn <- function(merMod) {
    x <- Anova(merMod, type = 2)
    ret <- x[c("PopID", "PopID:TempTrt", "PopID:WaterTrt"), "Chisq"]
    ret <- c(ret, x["PopID", "Chisq"] - x["PopID:TempTrt", "Chisq"],
             x["PopID", "Chisq"] - x["PopID:WaterTrt", "Chisq"])
    ret
  }

  system.time(
  ciMerModMort <- confint(fitMort, method = "boot", FUN = fn, nsim = 1e1, seed = 925200709,
                          parallel = "multicore", ncpus = 2L)
)
  # Figure of reaction norms
  d <- subset(master, !is.na(master$DiedFromStress))
  tab <- tapply(d$DiedFromStress, list(d$PopID, d$WaterTrt, d$TempTrt), 
                function(X) 1 - length(which(X)) / length(X))
  pdf(paste(pathFig, "/FigureS_MortRxnNorm.pdf", sep = ""), 4, 4, useDingbats = F)
  par(mai = c(0.5, 1, 0.5, 0.25))
  plot(0, 0, type = "n", xlab = "", ylab = "", axes = F, frame.plot = T, 
       xlim = c(1, 4), ylim = c(0, 1))
  abline(v = 2.5)
  mtext(side = 3, c("Cool", "Hot"), at = c(1.625, 3.375))
  mtext(side = 1, c("Wet", "Dry", "Wet", "Dry"), at = c(1, 2.25, 2.75, 4))
  mtext(side = 1, "Water Treatment", font = 2, cex = 1.25, line = 1.5)
  mtext(side = 3, "Temperature Treatment", font = 2, cex = 1.25, line = 1.5)
  axis(2, lwd = 0, lwd.ticks = 1, las = 1)
  title(ylab = "Survivorship", cex.lab = 1.5)

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.75), c(tab[i, "Wet", "Cool"], tab[i, "Wet", "Hot"]), col = "grey")
    lines(c(2.25, 4), c(tab[i, "Dry", "Cool"], tab[i, "Dry", "Hot"]), col = "grey",
          lty = 2)
  }

  for(i in 1:dim(tab)[1]) {
    lines(c(1, 2.25), c(tab[i, "Wet", "Cool"], tab[i, "Dry", "Cool"]), lty = 2)
    lines(c(2.75, 4), c(tab[i, "Wet", "Hot"], tab[i, "Dry", "Hot"]))
  }

  mtext(side = 3, "D", font = 2, cex = 2, at = grconvertX(-0.1, "npc", "user"))

  dev.off()
  
  ##### TEMPORARY: CODE TO EXPORT FIGURE FOR TALK #####
  # d <- subset(master, !is.na(master$DiedFromStress))
  # x <- tapply(d$DiedFromStress, list(d$PopID, d$WaterTrt, d$TempTrt), function(X) 1 - length(which(X)) / length(X))
  # 
  # pdf("~/Google Drive/Talks/JobTalk/3_Mimulus/DroughtResponse.pdf", w = 5.5, h = 5, useDingbats = F)
  # par(mai = c(1, 1.5, 0.25, 0.25))
  # 
  # plot(focClim$bio16_avg, x[, "Dry", "Hot"] / x[, "Wet", "Hot"], axes = F, frame.plot = T,
  #      pch = 21, col = "black", bg = primShade0, cex = 2, type = "n",
  #      xlab = "Precipitation of Wettest Quarter (mm)", 
  #      ylab = "Relative Survivorship\n(Hot/Dry : Hot/Wet)", cex.lab = 1.5)
  # points(focClim$bio16_avg, x[, "Dry", "Hot"] / x[, "Wet", "Hot"], pch = 21, col = "black", 
  #        bg = primShade0, cex = 1.5, lwd = 2)
  # axis(1, lwd = 0, lwd.ticks = 1)
  # axis(2, lwd = 0, lwd.ticks = 1, las = 1)
  # dev.off()
  # 

@


<<Effect size confidence intervals, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  pathFig <- "/Users/cdmuir/Google Drive/CardAdapt/ms/Figures"
  
pdf(paste0(pathFig, "/FigureS_EffectSizes.pdf"), 12, 4, useDingbats = F)
par(mai = c(1, 1, 0.5, 0.1), mfrow = c(1, 3))

for (mod in list(ciMerModLER, ciMerModSER, ciMerModPhoto)) {

#  assign("mod", mod)
  
  plot(0, 0, type = "n", axes = FALSE, frame.plot = TRUE,
       ylim = range(mod), xlim = c(1, 5),
       xlab = "", ylab = "Effect size (Mean squared error)", cex.lab = 1.5,
       main = )

  abline(h = 0, lty = 2, col = "grey50")
  for (i in 1:5) {
    lines(rep(i, 2), c(mod[i, "2.5 %"], mod[i, "97.5 %"]))
    points(i, mod[i, "estimate"], pch = 19)
  }

  axis(2, las = 1, lwd = 0, lwd.ticks = 1)
  clip(-100, 100, -100, 100)
  text(x = 1:5, y = grconvertY(-0.025, "npc", "user"), srt = 45, pos = 2, 
       labels = c("Pop", "Pop x Temp", "Pop x Water", "Pop - Pop x Temp",
                  "Pop - Pop x Water"))

}

dev.off()


@

\subsection*{Genetic variation in trait means and plasticity}

For all traits (Table~\ref{table:Table_traits}) except germination (see above), we tested for Population, Treatment (Temperature, Water, and Temperature $\times$ Water), and Population $\times$ Treatment interactions (Population $\times$ Temperature, Population $\times$ Water, and Population $\times$ Temperature $\times$ Water). We interpreted significant Population effects to indicate genetic variation in trait means and Population $\times$ Treatment interactions to indicate genetic variation in plasticity. As mentioned above, we used survival and GLMM models for germination rate and mortality, respectively. For all other traits, we used mixed model ANOVAs with Family and Block included as random factors. We fit models using restricted maximum likelihood in lmer, a function in the R package \pkg{lme4} \citep{Bates_etal_2015}. We determined significant fixed effect terms using a step-wise backward elimination procedure implemented with the step function in the R package \pkg{lmerTest} version 2.0-32 \citep{Kuznetsova_etal_2016}. We used Satterthwaite's approximation to calculate denominator degrees of freedom for $F$-tests. We also included days to germination as a covariate in growth analyses. To ensure that Population and Treatment effects were specific to a particular growth phase, we included germination day as a covariate in leaf expansion and stem elongation analyses.

Failure to detect a significant effect could be the result of Type-2 error, so we complemented the above analysis by comparing effect sizes in the full model. The full model contains all main effects, two-way interactions (Population $\times$ Temperature, Population $\times$ Water), a three-way interaction (Population $\times$ Temperature $\times$ Water), and random effects. We compared the effect sizes (mean squared error for LMM, ? for GLMM) of Population to Population $\times$ Treatment in 1000 parametric bootstrap samples simulated from fitted models. MARKER

\subsection*{Principal components of germination, growth, and photosynthesis}
For each single-trait model above, we extracted the Population coefficient (factoring out Treatment and other effects). The multivariate distribution of these coefficients was then summarized using principal components analysis. The first principal component of these traits (TraitPC1) loaded positively with germination, growth, and photostynthetic rate, therefore we define this as a phenotypic axis delineating fast to slow growth.

<<Principal Components, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  popMeans <- data.frame(
    germ = c(coef(fitGerm)[1], coef(fitGerm)[1] + coef(fitGerm)[2:16]),
    LER = fitLER$lsmeans.table[1:16, "Estimate"],
    SER = fitSER$lsmeans.table[1:16, "Estimate"],
    photo = fitPhoto$lsmeans.table[1:16, "Estimate"])

  traitPC <- prcomp(popMeans, center = T, scale. = T)
  focClim$PC1 <- traitPC$x[, "PC1"]

  # Percent variance explained by first principal component
  pve <- paste(round(100 * traitPC$sdev[1] ^ 2 / sum(traitPC$sdev ^ 2), 1), "%")
  
  # Plot latitude versus PC1
  pdf(paste(pathFig, "/Figure_PC1vLat.pdf", sep = ""), 5.5, 5)
  par(mai = c(1, 1.5, 0.25, 0.25))
  plot(focClim$Lat, focClim$PC1, xlim = c(32, 44), ylim = c(-4, 4), axes = F,
    frame.plot = T, type = "n", xlab = "", ylab = "")
  title(xlab = "Latitude", cex.lab = 1.5)
  title(ylab = "TraitPC1", cex.lab = 2, line = 5)
  title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
    list(x = pve)))
  axis(1, at = seq(32, 44, 4), labels = c(expression(32*degree), 
		expression(36*degree), expression(40*degree), expression(44*degree)), 
		lwd = 0, lwd.ticks = 1) 
  axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
  fitLat <- lm(PC1 ~ Latitude, data = focClim)
  
	# Label y-axis from slow to fast growth
	mtext(c("Slow\nGrowth", "Fast\nGrowth"), 2, at = c(-4, 4), cex = 1.25, 
				font = 3, las = 1, line = 4, adj = 0.5)

  # polygon of confidence intervals
  x <- seq(min(focClim$Latitude), max(focClim$Latitude), l = 1e3)
  u <- predict(fitLat, new = data.frame(Latitude = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitLat) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), 
  											upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "white", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  lines(range(focClim$Lat), rev(range(predict(fitLat))), lwd = 2)
  points(focClim$Lat, focClim$PC1, pch = 21, col = "black", bg = "white", 
  			 cex = 1.5, lwd = 2)
  
  # P-value
  text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1),
	     labels = bquote(italic(P) == .(round(summary(fitLat)[[4]][2, 4], 3))))

  dev.off()

  ##### TEMPORARY VERSION FOR TALK #####
  # Plot latitude versus PC1
#   pdf("~/Google Drive/Talks/JobTalk/3_Mimulus/Figure_PC1vLat.pdf", 5.5, 5, useDingbats = F)
#   par(mai = c(1, 1.5, 0.25, 0.25))
#   plot(focClim$Lat, focClim$PC1, xlim = c(32, 44), ylim = c(-4, 4), axes = F,
#     frame.plot = T, type = "n", xlab = "", ylab = "")
#   title(xlab = "Latitude", cex.lab = 1.5)
#   title(ylab = "TraitPC1", cex.lab = 2, line = 5)
#   title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
#     list(x = pve)))
#   axis(1, at = seq(32, 44, 4), labels = c(expression(32*degree), 
# 		expression(36*degree), expression(40*degree), expression(44*degree)), 
# 		lwd = 0, lwd.ticks = 1) 
#   axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
#   fitLat <- lm(PC1 ~ Latitude, data = focClim)
#   
# 	# Label y-axis from slow to fast growth
# 	mtext(c("Slow\nGrowth", "Fast\nGrowth"), 2, at = c(-4, 4), cex = 1.25, 
# 				font = 3, las = 1, line = 4, adj = 0.5)
# 
#   # polygon of confidence intervals
#   x <- seq(min(focClim$Latitude), max(focClim$Latitude), l = 1e3)
#   u <- predict(fitLat, new = data.frame(Latitude = x))
#   s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitLat) %*% c(1, X)))
#   confInt <- data.frame(lowerCI = qnorm(0.025, u, s), 
#   											upperCI = qnorm(0.975, u, s))
#   polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
#     col = "grey", border = NA)
#   lines(x, confInt$lowerCI, lwd = 2, lty = 2)
#   lines(x, confInt$upperCI, lwd = 2, lty = 2)
#   lines(range(focClim$Lat), rev(range(predict(fitLat))), lwd = 2)
#   points(focClim$Lat, focClim$PC1, pch = 21, col = "black", bg = primShade0, 
#   			 cex = 1.5, lwd = 2)
#   
#   # P-value
#   text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1),
# 	     labels = bquote(italic(P) == .(round(summary(fitLat)[[4]][2, 4], 3))))
# 
#   dev.off()

@

<<thinnedOccClimate, echo=FALSE, results=hide, warning=FALSE>>=
# R data.frame of 178 E. cardinalis occurrences, thinned to correct for sampling bias, and associated climate data
load(paste(pathDat, "/thinnedOccClimate.RData", sep = ""))

@

\subsection*{Identifying putative selective agents}

Latitudinal clines are common, but it is often difficult to ascribe this variation to a particular selective agent. To reiterate, we tested three non-mutually exclusive hypotheses about how such latitudinal clines emerge: 1) one or two climatic variables explain latitudinal trait variation; 2) latitude is a proxy for multiple climatic factors that together shape trait variation; and 3) latitude integrates selection in a broader climatic neighborhood. We found that a population's position along TraitPC1 correlated strongly with the latitude of origin (see Results) and next used Random Forest regression \citep{Liaw_Wiener_2002} to identify putative climatic factors underlying trait-latitude associations in \textit{E. cardinalis}. We reasoned that if we identified a single climatic factor that explained more trait variation than latitude, then this would suggest that factor is a key selective agent underlying the latitudinal cline (Hypothesis 1). On the other hand, if multiple climatic factors together are necessary to explain trait variation, then this would suggest that many climatic factors together have imposed selection for the latitudinal cline (Hypothesis 2). We hereafter refer to factors identified in this analysis as `Climate-TraitPC1' variables. To test Hypothesis 3 about climatic neighborhoods driving selection, we directly competed local with neighborhood climate. We used the immediate collection location for local climate. For climate neighborhoods, we sampled climate at 1000 random points (at 90-m resolution) within a 62-km radius buffer around the collection and took the average. We chose this buffer radius based on population genetic structure, as inferred from $\approx$25,000 restriction-site associated SNPs among 49 populations from across the range \citep{Paul_etal_2016}. Spatial autocorrelation in allele frequencies persists for 62 km. However radii of 10 km$^2$ and 100 km$^2$ resulted in similar outcomes (data not shown). Since \textit{E. cardinalis} is found exclusively in riparian areas, we only selected points along streams using the National Hydrogeoraphy Dataset \citep{NHD}. Climatic means and variances (see below) were weighted by their climatic suitability as determined using a multimodel ensemble average of ecological niche models \citep{Angert_ENM}. In addition to competing local and neighborhood climate, we compared the univariate correlation between local and neighborhood climate with TraitPC1 and Latitude using paired $t$-tests. We adjusted degrees of freedom to account for the fact that many climatic factors are highly correlated and not independent. Specifically, we calculated the effective number of independent climatic factors ($M_\text{eff}$) using the formula $M_\text{eff} = 1 + (M - 1) (1 - \text{Var}(\lambda) / M)$ \citep{Chevrud_2001}, where $M$ is the original number of climatic factors and $\lambda$ are the eigenvalues of the correlation matrix of all climatic factors. 

To help eliminate potentially spurious correlations between TraitPC1 and climate, we tested for overlap between climatic variables that best predict latitude of all \textit{E. cardinalis} occurrence records (see detail below), not just the 16 focal populations. We refer to these climatic factors as `Climate-Latitude' variables. The logic is that climatic factors associated with both TraitPC1 and latitude for all populations are more likely to be important selective agents than climatic factors that happen to correlate with TraitPC1 but do not covary with latitude throughout the \textit{E. cardinalis} range. Therefore, we did not consider Climate-TraitPC1 variables to be candidate selective agents unless the same or very similar variable was found in the Climate-Latitude analysis. However, we do interpret potential selective agents identified in Climate-Latiude analyses alone, because the goal was to explain the latitudinal clines in traits, not all aspects of climate that vary with latitude.

We selected Climate-Latitude and Climate-TraitPC1 variables independently using Variable Selection Using Random Forest (VSURF) algorithm in the R package \pkg{VSURF} version 1.0.3 \citep{Genuer_etal_2016}. Random Forest regression is useful for cases like ours when the number of potential predictors is similar to or greater than the number of observations (`high $p$, low $n$' problem). VSURF is a multistep algorithm that progressively retains or eliminates variables based on their importance over regression trees in the forest. Variable importance is defined as the average amount a climate variable reduces mean-squared error in the predicted response (TraitPC1 or Latitude), compared to a randomly permuted dataset, across all trees in the random forest (see \citeauthor{Genuer_etal_2015} [\citeyear{Genuer_etal_2015}] for further detail). Hence, VSURF automatically eliminates unimportant and redundant variables based on the data without having to arbitrarily choose among colinear climate variables before the analysis. We kept only variables selected for prediction, the most stringent criterion. We visually depict how we selected climatic variables in Fig~\ref{fig:Fig_VarSelect}. 

<<Fig_VarSelect, eval=FALSE, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  library(raster)
	library(rgeos)
  library(extrafont)
  
  # Get maps of California and Oregon
  usa <- getData('GADM', country = "USA", level = 1, path = pathDat)
  states <- gSimplify(usa[usa@data$NAME_1 %in% c("California", "Oregon",
                                                 "Nevada", "Idaho"), ], 
                      tol = 0.01, topologyPreserve = TRUE)

  # Load thinned occurrences used in Climate-Lat analyses and convert to Spatial class
  occ <- read.csv(paste(pathDat, "/thinnedOccLatLon.csv", sep = ""), 
                  row.names = NULL)
  coordinates(occ) <- ~Longitude+Latitude
  projection(occ) <- CRS('+proj=longlat');

  # Convert focal population coordinates to Spatial class
  foc <- focClim[, c("Longitude", "Latitude")]
  coordinates(foc) <- ~Longitude+Latitude
  projection(foc) <- CRS('+proj=longlat');

  # Clip and map
  clipExtent <- as(extent(floor(min(occ$Longitude) * 10) / 10, 
	   											ceiling(max(occ$Longitude) * 10) / 10, 
	  											floor(min(occ$Latitude) * 10) / 10, 
		  										ceiling(max(occ$Latitude) * 10) / 10), 
			  					 "SpatialPolygons")
  proj4string(clipExtent) <- CRS(proj4string(states))
  states <- gIntersection(states, clipExtent, byid = TRUE)

  # Transform to Albert's Equal Area Projection
  projAEA <- sprintf("+proj=aea +lat_1=%s +lat_2=%s +lat_0=%s +lon_0=%s +x_0=0 +y_0=0",
                     extent(clipExtent)[3], extent(clipExtent)[4], 
		  							 extent(clipExtent)[1], extent(clipExtent)[2])
  states <- spTransform(states, CRS(projAEA))
  occAEA <- spTransform(occ, CRS(projAEA))
  focAEA <- spTransform(foc, CRS(projAEA))

  # PLOT
  pdf(paste(pathFig, "/Figure_VarSelect.pdf", sep = ""), w = 8, h = 7)#, 
	  	#useDingbat = F)
  loadfonts()
  mat <- matrix(c(1, 4, 6, 9,
	  							1, 4, 6, 9,
		  						1, 4, 6, 9,
			  					2, 5, 7, 10,
				  				2, 5, 7, 10,
					  			3, 3, 8, 8,
						  		3, 3, 8, 8), ncol = 4, byrow = T)
  layout(mat)

  # A. Climate-Latitude variables

	  ## Panel 1: Map
	  par(mai = rep(0.5, 4), family = "sans")
	  plot(states)
	  mtext(text = "Climate-Latitude variables", side = 3, at = grconvertX(2, "in"),
	        font = 2, line = 2)
	  mtext(text = "local", side = 3)
    points(occAEA, pch = 19, cex = 0.5)
    
	  # Position of Cuyamaca Rancho population in device units
	  xCUR <- grconvertX(focAEA$Longitude["CUR"], from = "user", to = "device")
	  yCUR <- grconvertY(focAEA$Latitude["CUR"], from = "user", to = "device")
	  
	  ## Panel 2: Cartoon of point versus spatial average climate estimate
	  par(mai = rep(0, 4), family = "LM Roman 10")
	  
	    ### River 1
	    x1 <- seq(-1, 1, length.out = 1e2)
	    r1 <- function(x) exp(x) - exp(0) + sin(x * 10) / 10
	    y1 <- r1(x1)
	
	    ### River 2
	    x2 <- seq(0, 1, length.out = 1e2)
	    r2 <- function(x) exp(-x) - exp(0) + sin(x * 10) / 10
	    y2 <- r2(x2)
	
	  plot(0, 0, type = "n", xlim = c(-1, 1), ylim = c(-1, 1), axes = F)
	  clip(-1, 1, -1, 1)
	  points(x1, y1, type = "l", col = "blue", lwd = 5)
	  points(x2, y2, type = "l", col = "blue", lwd = 5)
	    
	  # Draw buffer area
	  x <- seq(-1, 0, length.out = 1e3)
	  clip(-2, 2, -2, 2)
	  polygon(c(x, rev(x)), c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  polygon(c(x + 1, rev(x + 1)), c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  polygon(c(x, rev(x)), -c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  polygon(c(x + 1, rev(x + 1)), -c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)

	  # Draw point and label
	  points(0, 0, pch = 19, cex = 3)
	  text(0, 0, labels = expression(italic(x[paste(i, ",", 0)])), adj = 1.5, 
	       cex = 1.5)
    text(0, 1, labels = expression(Climate[italic(i)] == 
           italic(x[paste(i, ",", 0)])), cex = 1.5, pos = 3, offset = 1)
    
	  ## Panel 3: Climate-Latitude relationship
	  set.seed(01152016)
  	y <- (occ$Latitude - mean(occ$Latitude)) / sd(occ$Latitude) + 
		      rnorm(length(occ), 0, sqrt(0.1))
	  par(mai = c(0.6, 0.6, 1/25, 1/25), family = "sans")
	  plot(y, occ$Latitude, las = 1, axes = F, frame.plot = T, 
		  	 xlim = c(-2.5, 2.5), ylim = c(32, 45), cex.lab = 1.5,
			   pch = 19, col = "grey", xlab = "", ylab = "Latitude of origin")
	  title(xlab = "Climate variable\n(local or neighborhood)", cex.lab = 1.5, line = 2)
	
  	for (i in seq(32, 44, 2)) {
		  axis(2, at = i, labels = bquote(.(i)*degree), las = 1)
	  	}
	  fit <- lm(occ$Latitude ~ y)
	  lines(range(y), range(predict(fit)), lwd = 2)
  
	 	## Panel 4: Map
	  par(mai = rep(0.5, 4), family = "sans")
	  plot(states)
  	mtext(text = "neighborhood", side = 3)
    points(occAEA, pch = 21, cex = 2, col = "black", bg = "white")
	  points(occAEA, pch = 19, cex = 0.5)
    
	  ## Panel 5: Cartoon of neighborhood climate calculation
	  par(mai = rep(0, 4), family = "LM Roman 10")

  		### Climate grid

  			#### Functions to find closest point along "stream"
				fn <- function(par, x, y) {
					if (x < 0) {
						d1 <- sqrt((par - x) ^ 2 + (r1(par) - y) ^ 2)
						return(d1)
					} else {
						d1 <- sqrt((par - x) ^ 2 + (r1(par) - y) ^ 2)
						d2 <- sqrt((par - x) ^ 2 + (r2(par) - y) ^ 2)
						return(min(c(d1, d2)))
					}
				}

				getStreamDist <- function(x, y) {
					fit <- optim(x, fn, x = x, y = y, method = "Brent", lower = -2, 
											 upper = 2)
					retval <- setNames(c(fit$par, fit$value), c("s", "dist"))
					return(retval)
				}

			  #### "Climate" is fxn of distance from stream and pos along x-axis
			  xy <- expand.grid(x = seq(-0.95, 0.95, 0.1), 
			                    y = seq(-0.95, 0.95, 0.1))
			  xyz <- cbind(xy, 
			               t(apply(xy, 1, function(X) getStreamDist(X[1], X[2]))))
			  xyz$z <- xyz$dist + xyz$x * 0.3
			  xyz$z <- xyz$z - min(xyz$z)
			  xyz$z <- xyz$z / max(xyz$z)

		  plot(0, 0, type = "n", xlim = c(-1, 1), ylim = c(-1, 1), axes = F)
		  clip(-1, 1, -1, 1)
		  points(x1, y1, type = "l", col = "blue", lwd = 5)
		  points(x2, y2, type = "l", col = "blue", lwd = 5)

		  apply(xyz, 1, function(X) rect(X["x"] - 0.05, X["y"] - 0.05, 
																	   X["x"] + 0.05, X["y"] + 0.05,
														 			   col = rgb(0, 0, 1, alpha = X["z"]), 
																	   border = NA))

			# Draw buffer area and circle
			x <- seq(-1, 0, length.out = 1e3)
			clip(-2, 2, -2, 2)
			polygon(c(x, rev(x)), c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			polygon(c(x + 1, rev(x + 1)), c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			polygon(c(x, rev(x)), -c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			polygon(c(x + 1, rev(x + 1)), -c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			
			x <- seq(-1, 1, length.out = 1e3)
			polygon(c(x, rev(x)), c(sqrt(1 - x ^ 2), -sqrt(1 - x ^ 2)),
			        border = "black", lwd = 2)
			
			# Draw points and label
	    points(0, 0, pch = 19, cex = 3)
	    text(0, 0, labels = expression(italic(x[paste(i, ",", 0)])), adj = 1.5, 
	         cex = 1.5)
      text(0, 1, labels = expression(Climate[italic(i)] == 
           sum(over(italic(x)[paste(italic(i), ",", italic(j))], italic(n)), 
               scriptstyle(italic(j) == 1), italic(n))), 
           cex = 1.5, pos = 3, offset = 1)

      points(c(-2/3, -1/3, 1/2, 1/3, 2/3),
             c(r1(-2/3), r1(-1/3), r1(1/2), r2(1/3), r2(2/3)),
             cex = 2, pch = 21, bg = "grey", col = "black")
      
      text(-2/3 + 0.15, r1(-2/3) - 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 1)])))
      text(-1/3, r1(-1/3) - 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 2)])))
      text(1/3 - 0.15, r2(1/3) - 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 3)])))
      text(1/2 - 0.15, r1(1/2) + 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 4)])))
      text(2/3, r2(2/3) + 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 5)])))

      par(family = "sans")
      str <- "62-km\nbuffer"
      rect(-0.4 - strwidth(str, cex = 1.75) / 2,
           0.4 - strheight(str, cex = 1.75) / 2,
           -0.4 + strwidth(str, cex = 1.75) / 2,
           0.4 + strheight(str, cex = 1.75) / 2, 
           col = rgb(1, 1, 1, alpha = 0.5), border = NA)
      text(-0.4, 0.4, labels = str, cex = 1.5)   
 
   # B. Example Climate-Trait variable, point estimated

  	## Panel 6: Map
	  par(mai = rep(0.5, 4), family = "sans")
	  plot(states)
	  mtext(text = "Climate-Trait variables", side = 3, at = grconvertX(6, "in"),
	        font = 2, line = 2)
    mtext(text = "local", side = 3)
	  points(focAEA, pch = 19, cex = 0.5)

	  ## Panel 7: Cartoon of local climate
	  par(mai = rep(0, 4), family = "LM Roman 10")
	  plot(0, 0, type = "n", xlim = c(-1, 1), ylim = c(-1, 1), axes = F)
	  clip(-1, 1, -1, 1)
	  points(x1, y1, type = "l", col = "blue", lwd = 5)
	  points(x2, y2, type = "l", col = "blue", lwd = 5)
	  
	  # Draw buffer area
	  x <- seq(-1, 0, length.out = 1e3)
	  clip(-2, 2, -2, 2)
	  polygon(c(x, rev(x)), c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  polygon(c(x + 1, rev(x + 1)), c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  polygon(c(x, rev(x)), -c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  polygon(c(x + 1, rev(x + 1)), -c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
	          col = "white", border = "white", lwd = 2)
	  
	  # Draw point and label
	  points(0, 0, pch = 19, cex = 3)
	  text(0, 0, labels = expression(italic(x[paste(i, ",", 0)])), adj = 1.5, 
	       cex = 1.5)
    text(0, 1, labels = expression(Climate[italic(i)] == 
           italic(x[paste(i, ",", 0)])), cex = 1.5, pos = 3, offset = 1)
	  
	  ## Panel 8: Climate-trait relationship
	  par(mai = c(0.6, 0.6, 1/25, 1/25), family = "sans")
	  y <- (foc$Latitude - mean(foc$Latitude)) / sd(foc$Latitude) + 
		      rnorm(length(foc), 0, sqrt(0.1))
	  plot(y, foc$Latitude, las = 1, axes = F, frame.plot = T, 
		  	 xlim = c(-2.5, 2.5), ylim = c(32, 45),
			   pch = 19, col = "grey", xlab = "", ylab = "")
	  title(ylab = "Trait", cex.lab = 1.5, line = 1)
	  title(xlab = "Climate variable\n(local or neighborhood)", cex.lab = 1.5, line = 2)
    fit <- lm(foc$Latitude ~ y)
	  lines(range(y), range(predict(fit)), lwd = 2)
	  
  	## Panel 9: Map
  	par(mai = rep(0.5, 4), family = "sans")
  	plot(states)
  	mtext(text = "neighborhood", side = 3)
  	points(focAEA, pch = 21, cex = 2, col = "black", bg = "white")
	  points(focAEA, pch = 19, cex = 0.5)

  	## Panel 10: Cartoon of point versus spatial average climate estimate
	  par(mai = rep(0, 4), family = "LM Roman 10")

  		### Climate grid

			  #### "Climate" is fxn of distance from stream and pos along x-axis
			  xy <- expand.grid(x = seq(-0.95, 0.95, 0.1), 
			                    y = seq(-0.95, 0.95, 0.1))
			  xyz <- cbind(xy, 
			               t(apply(xy, 1, function(X) getStreamDist(X[1], X[2]))))
			  xyz$z <- xyz$dist + xyz$x * 0.3
			  xyz$z <- xyz$z - min(xyz$z)
			  xyz$z <- xyz$z / max(xyz$z)

		  plot(0, 0, type = "n", xlim = c(-1, 1), ylim = c(-1, 1), axes = F)
		  clip(-1, 1, -1, 1)
		  points(x1, y1, type = "l", col = "blue", lwd = 5)
		  points(x2, y2, type = "l", col = "blue", lwd = 5)

		  apply(xyz, 1, function(X) rect(X["x"] - 0.05, X["y"] - 0.05, 
																	   X["x"] + 0.05, X["y"] + 0.05,
														 			   col = rgb(0, 0, 1, alpha = X["z"]), 
																	   border = NA))

			# Draw buffer area and circle
			x <- seq(-1, 0, length.out = 1e3)
			clip(-2, 2, -2, 2)
			polygon(c(x, rev(x)), c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			polygon(c(x + 1, rev(x + 1)), c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			polygon(c(x, rev(x)), -c(sqrt(1 - x ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			polygon(c(x + 1, rev(x + 1)), -c(sqrt(1 - (x + 1) ^ 2), rep(1.1, 1e3)), 
			        col = "white", border = "white", lwd = 2)
			
			x <- seq(-1, 1, length.out = 1e3)
			polygon(c(x, rev(x)), c(sqrt(1 - x ^ 2), -sqrt(1 - x ^ 2)),
			        border = "black", lwd = 2)
			
			# Draw points and label
	    points(0, 0, pch = 19, cex = 3)
	    text(0, 0, labels = expression(italic(x[paste(i, ",", 0)])), adj = 1.5, 
	         cex = 1.5)

      text(0, 1, labels = expression(Climate[italic(i)] == 
           sum(over(italic(x)[paste(italic(i), ",", italic(j))], italic(n)), 
               scriptstyle(italic(j) == 1), italic(n))), 
           cex = 1.5, pos = 3, offset = 1)

      points(c(-2/3, -1/3, 1/2, 1/3, 2/3),
             c(r1(-2/3), r1(-1/3), r1(1/2), r2(1/3), r2(2/3)),
             cex = 2, pch = 21, bg = "grey", col = "black")
      
      text(-2/3 + 0.15, r1(-2/3) - 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 1)])))
      text(-1/3, r1(-1/3) - 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 2)])))
      text(1/3 - 0.15, r2(1/3) - 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 3)])))
      text(1/2 - 0.15, r1(1/2) + 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 4)])))
      text(2/3, r2(2/3) + 0.15, cex = 1.5,
           labels = expression(italic(x[paste(i, ",", 5)])))

      par(family = "sans")
      str <- "62-km\nbuffer"
      rect(-0.4 - strwidth(str, cex = 1.75) / 2,
           0.4 - strheight(str, cex = 1.75) / 2,
           -0.4 + strwidth(str, cex = 1.75) / 2,
           0.4 + strheight(str, cex = 1.75) / 2, 
           col = rgb(1, 1, 1, alpha = 0.5), border = NA)
      text(-0.4, 0.4, labels = str, cex = 1.5)   
      
	dev.off()
		
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of variable selection algorithm
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_VarSelect.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Climatic variable selection]{Overview of method for identifying putative climatic selective agents underlying latitudinal cline. We looked for climate variables that explained both the latitude of \Sexpr{2 * nrow(thinnedOccClimate)} \textit{E. cardinalis} occurrences (`Climate-Latitude variables') and with traits (`Climate-Trait variables'). For Climate-Latitude variables we extracted climate data from recent occurrences located throughout California and Oregon, USA (shown in map). For Climate-Trait variables, we extracted climatic data for the 16 focal populations. For both analyses, we extracted local and neighborhood climate. Local climate refers to climate only from where a population was collected ($x_{i,0}$). Neighborhood climate was calculated as the average over 1000 points in a 62-km radius climatic neighborhood ($x_{i,1}, x_{i,2}, \dots$), but only along stream habitats as \textit{E. cardinalis} is riparian. We identified climatic factors that most strongly predicted latitude of occurrences (Climate-Latitude variables) and traits (Climate-Trait variables), as shown for hypothetical data in plots at the bottom of the figure.}
	\label{fig:Fig_VarSelect}
\end{figure}

<<VSURF, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

  library(VSURF)
  
	# Load climVars
	load(paste(pathDat, "/ExploratoryClimateVariables.RData", sep = ""))

  # Load VSURF results of climatic variables associated with latitude at cardinalis occurence points
  # Note that occurrences have been randomly thinned to account for variation in sampling effort
  vsurfLat <- get(load(paste(pathDat, "/clim2lat_fit_t1.RData", sep = "")))

  # Variable selection using random forests (VSURF)
	# Climate variables that predict PC1 (point and spatially averaged)
  set.seed(69648)
  vsurfPC1 <- VSURF(focClim[, climVars], focClim[, "PC1"], parallel = TRUE, ncores = 3, 
                    ntree = 1e4)

  imp1 <- climVars[vsurfLat$varselect.pred]
  nImpLat <- length(imp1)

  imp2 <- climVars[vsurfPC1$varselect.pred]
  nImpPC1 <- length(imp2)

	# Table of symbols for plotting climatic factor labels
	clim2sym <- read.csv(paste(pathMS, "/climVar2plot.csv", sep = ""), 
										   stringsAsFactors = F, row.names = 1)

#####	Figure_ClimVarImp.pdf ####

# Calculatins for plot dimensions
sw1a <- strwidth(s = clim2sym[imp1, "climFactor"], units = "inches", font = 4)
sw2a <- strwidth(s = clim2sym[imp1, "tempScale"], units = "inches", font = 4)

sw1b <- strwidth(s = clim2sym[imp2, "climFactor"], units = "inches", font = 4)
sw2b <- strwidth(s = clim2sym[imp2, "tempScale"], units = "inches", font = 4)

# Figure size in inches
panelWidth <- 3
panelHeight <- 4
panelGap <- 0.1
yLabWidth <- 4 * panelGap + max(c(sw1a, sw1b)) + max(c(sw2a, sw2b)) +
  (panelHeight / max(nImpLat + 1, nImpPC1 + 1) * 5)
xLabHeight <- 1
heightA <- (panelHeight + 2 * xLabHeight)
heightB <- (panelHeight * (nImpPC1) / (nImpLat) + xLabHeight)

pdf(paste(pathFig, "/Figure_ClimVarImp.pdf", sep = ""), 
       width = yLabWidth + panelWidth + 2 * panelGap, 
       height = heightA + heightB)
par(mai = c(xLabHeight, yLabWidth, xLabHeight, 2 * panelGap), ann = F, yaxs = "i")
layout(matrix(1:2, ncol = 1), widths = rep(1, 2), heights = lcm(2.54 * c(heightA, heightB)))

# Panel A: Climate-Latitude
plot(0, 0, type = "n", xlim = c(0, ceiling(vsurfLat$imp.mean.dec)[1]), 
     ylim = c(0.5, nImpLat + 0.5), axes = F, frame.plot = T)
abline(h = nImpLat:1, col = "grey")
points(vsurfLat$imp.mean.dec[1:length(vsurfLat$varselect.pred)], nImpLat:1, 
       pch = 21, col = "black", bg = "white", cex = 1.5)
text(grconvertX(0, "npc", "user") + strwidth("A", "user", cex = 2, font = 2), 
     grconvertY(1, "npc", "user"), labels = "A", font = 2, cex = 2, pos = 1)

# X axis & labels
axis(1, lwd = 0, lwd.ticks = 1)
title(xlab = "Climate-Latitude Importance", line = 3, cex.lab = 1.5)

# Y labels
clip(grconvertX(0, from = "nfc", to = "user"), 
     grconvertX(1, from = "nfc", to = "user"),
     grconvertY(0, from = "nfc", to = "user"),
     grconvertY(1, from = "nfc", to = "user"))

# Climatic factor
text(grconvertX(panelGap, from = "in", to = "user"), nImpLat:1, 
     pos = 4, labels = clim2sym[imp1, "climFactor"],
     font = ifelse(imp1 %in% intersect(imp1, imp2), 2, 1))
text(grconvertX(panelGap + max(c(sw1a, sw1b)) / 2, from = "in", to = "user"), 
     grconvertY(1, "ndc", "user"), pos = 1, labels = "Climatic\nfactor")

# Temporal scale
text(grconvertX(2 * panelGap + max(c(sw1a, sw1b)), from = "in", to = "user"), nImpLat:1,
     pos = 4, labels = clim2sym[imp1, "tempScale"],
     font = ifelse(imp1 %in% intersect(imp1, imp2), 2, 1))
text(grconvertX(2 * panelGap + max(c(sw1a, sw1b)) + max(c(sw2a, sw2b)) / 2, from = "in", to = "user"), 
     grconvertY(1, "ndc", "user"), pos = 1, labels = "Temporal\nScale")

# Climatic average or variation?
bx <- panelHeight / max(nImpLat + 1, nImpPC1 + 1)
xs <- 3 * panelGap + max(c(sw1a, sw1b)) + max(c(sw2a, sw2b))
for (i in nImpLat:0) {
  lines(x = c(grconvertX(xs, "in", "user"), grconvertX(xs + 2 * bx, "in", "user")), 
        rep(i + 0.5, 2), col = "grey")
}

for (i in c(grconvertX(xs, "in", "user"), 
            grconvertX(xs + bx, "in", "user"),
            grconvertX(xs + 2 * bx, "in", "user"))) {
  lines(rep(i, 2), c(max(nImpLat, nImpPC1) - nImpLat + 0.5, max(nImpLat, nImpPC1) + 0.5),
        col = "grey")
}

x <- ifelse(grepl("_avg", imp1), xs + bx / 2, xs + 3 * bx / 2)
points(grconvertX(x, "in", "user"), nImpLat:1, pch = 19)

text(grconvertX(xs, from = "in", to = "user"), 
     max(nImpLat, nImpPC1) + 1, pos = 4, labels = "Avg.", srt= 45)
text(grconvertX(xs + bx, from = "in", to = "user"), 
     max(nImpLat, nImpPC1) + 1, pos = 4, labels = "Var.", srt= 45)

text(grconvertX(xs + bx, from = "in", to = "user"), 
     grconvertY(1, "ndc", "user"), pos = 1, labels = "Type")

# Spatial scale
xs <- xs + 3 * bx
for (i in nImpLat:0) {
  lines(x = c(grconvertX(xs, "in", "user"), grconvertX(xs + 2 * bx, "in", "user")), 
        rep(i + 0.5, 2), col = "grey")
}

x <- ifelse(!grepl("_SA", imp1), xs + bx / 2, xs + 3 * bx / 2)
points(grconvertX(x, "in", "user"), nImpLat:1, pch = 19)

for (i in c(grconvertX(xs, "in", "user"), 
            grconvertX(xs + bx, "in", "user"),
            grconvertX(xs + 2 * bx, "in", "user"))) {
  lines(rep(i, 2), c(max(nImpLat, nImpPC1) - nImpLat + 0.5, max(nImpLat, nImpPC1) + 0.5),
        col = "grey")
}

text(grconvertX(xs, from = "in", to = "user"), 
     max(nImpLat, nImpPC1) + 1, pos = 4, labels = "Local", srt= 45)
text(grconvertX(xs + bx, from = "in", to = "user"), 
     max(nImpLat, nImpPC1) + 1, pos = 4, labels = "Neighborhood", srt= 45)

text(grconvertX(xs + bx, from = "in", to = "user"), 
     grconvertY(1, "ndc", "user"), pos = 1, labels = "Spatial\nscale")

# Panel B: Climate-Trait
par(mai = c(xLabHeight, yLabWidth, 0, 2 * panelGap))
plot(0, 0, type = "n", xlim = c(0, 0.1 * ceiling(10 * vsurfPC1$imp.mean.dec)[1]), 
     ylim = c(0.5, nImpPC1 + 0.5), axes = F, frame.plot = T)
abline(h = nImpPC1:1, col = "grey")
points(vsurfPC1$imp.mean.dec[1:length(vsurfPC1$varselect.pred)], nImpPC1:1, 
       pch = 21, col = "black", bg = "white", cex = 1.5)
text(grconvertX(0, "npc", "user") + strwidth("B", "user", cex = 2, font = 2), 
     grconvertY(1, "npc", "user"), labels = "B", font = 2, cex = 2, pos = 1)

# X axis & labels
axis(1, lwd = 0, lwd.ticks = 1)
title(xlab = "Climate-TraitPC1 Importance", line = 3, cex.lab = 1.5)

# Y labels
clip(grconvertX(0, from = "nfc", to = "user"), 
     grconvertX(1, from = "nfc", to = "user"),
     grconvertY(0, from = "nfc", to = "user"),
     grconvertY(1, from = "nfc", to = "user"))

# Climatic factor
text(grconvertX(panelGap, from = "in", to = "user"), nImpPC1:1, 
     pos = 4, labels = clim2sym[imp2, "climFactor"],
     font = ifelse(imp2 %in% intersect(imp1, imp2), 2, 1))

# Temporal scale
text(grconvertX(2 * panelGap + max(c(sw1a, sw1b)), from = "in", to = "user"), nImpPC1:1,
     pos = 4, labels = clim2sym[imp2, "tempScale"],
     font = ifelse(imp2 %in% intersect(imp1, imp2), 2, 1))

# Climatic average or variation?
xs <- 3 * panelGap + max(c(sw1a, sw1b)) + max(c(sw2a, sw2b))
for (i in nImpPC1:0) {
  lines(x = c(grconvertX(xs, "in", "user"), grconvertX(xs + 2 * bx, "in", "user")), 
        rep(i + 0.5, 2), col = "grey")
}

for (i in c(grconvertX(xs, "in", "user"), 
            grconvertX(xs + bx, "in", "user"),
            grconvertX(xs + 2 * bx, "in", "user"))) {
  lines(rep(i, 2), c(0.5, nImpPC1 + 0.5), col = "grey")
}

x <- ifelse(grepl("_avg", imp2), xs + bx / 2, xs + 3 * bx / 2)
points(grconvertX(x, "in", "user"), nImpPC1:1, pch = 19)

# Spatial scale
xs <- xs + 3 * bx
for (i in nImpPC1:0) {
  lines(x = c(grconvertX(xs, "in", "user"), grconvertX(xs + 2 * bx, "in", "user")), 
        rep(i + 0.5, 2), col = "grey")
}

x <- ifelse(!grepl("_SA", imp2), xs + bx / 2, xs + 3 * bx / 2)
points(grconvertX(x, "in", "user"), nImpPC1:1, pch = 19)

for (i in c(grconvertX(xs, "in", "user"), 
            grconvertX(xs + bx, "in", "user"),
            grconvertX(xs + 2 * bx, "in", "user"))) {
  lines(rep(i, 2), c(0.5, nImpPC1 + 0.5), col = "grey")
}

dev.off()

##### Plot most important climatic predictors versus PC1 #####
	
  pdf(paste(pathFig, "/Figure_ClimPC1.pdf", sep = ""), 8, 8)

	# Panel A. bio16_cva_SA
  par(mai = c(1, 1, 0.25, 0.25), mfrow = c(2, 2))
  plot(focClim$bio16_cva_SA, focClim$PC1, xlim = c(0.25, 0.55), ylim = c(-4, 4),
       axes = F, frame.plot = T, type = "n", xlab = "", ylab = "")
  text(grconvertX(0.075, "npc", "user"), grconvertY(0.925, "npc", "user"),
       labels = "A", font = 2, cex = 2)
  title(xlab = "Precipitation of Wettest Quarter", cex.lab = 1.5)
  title(xlab = "(Coefficient of variation)", cex.lab = 1, line = 4)
  title(ylab = "TraitPC1", cex.lab = 1.5, line = 4)
  title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
    list(x = pve)))
  axis(1, lwd = 0, lwd.ticks = 1) 
  axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
  fitBio16cva <- lm(PC1 ~ bio16_cva_SA, data = focClim)

  # polygon of confidence intervals
  x <- seq(min(focClim$bio16_cva_SA), max(focClim$bio16_cva_SA), l = 1e3)
  u <- predict(fitBio16cva, new = data.frame(bio16_cva_SA = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitBio16cva) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), 
                        upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "white", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  lines(range(focClim$bio16_cva_SA), range(predict(fitBio16cva)), lwd = 2)
  points(focClim$bio16_cva_SA, focClim$PC1, pch = 21, col = "black", 
         bg = "white", cex = 1.5, lwd = 2)

  # R2-value
  text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1), 
	     labels = bquote(italic(R)^2 == .(round(summary(fitBio16cva)$r.squared, 2))))

  # Panel B. bio15_cva_SA
  plot(focClim$bio15_cva_SA, focClim$PC1, xlim = c(0.14, 0.24), ylim = c(-4, 4),
       axes = F, frame.plot = T, type = "n", xlab = "", ylab = "")
  text(grconvertX(0.075, "npc", "user"), grconvertY(0.925, "npc", "user"),
       labels = "B", font = 2, cex = 2)
  title(xlab = "Precipitation Seasonality", cex.lab = 1.5)
  title(xlab = "(Coefficient of variation)", cex.lab = 1, line = 4)
  title(ylab = "TraitPC1", cex.lab = 1.5, line = 4)
  title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
    list(x = pve)))
  axis(1, lwd = 0, lwd.ticks = 1) 
  axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
  fitBio15cva <- lm(PC1 ~ bio15_cva_SA, data = focClim)

  # polygon of confidence intervals
  x <- seq(min(focClim$bio15_cva_SA), max(focClim$bio15_cva_SA), l = 1e3)
  u <- predict(fitBio15cva, new = data.frame(bio15_cva_SA = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitBio15cva) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), 
                        upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "white", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  lines(x, u, lwd = 2)
  points(focClim$bio15_cva_SA, focClim$PC1, pch = 21, col = "black", 
         bg = "white", cex = 1.5, lwd = 2)

  # R2-value
  text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1), 
	     labels = bquote(italic(R)^2 == .(round(summary(fitBio15cva)$r.squared, 2))))

  # Panel C. bio15_cva_SA vs bio16_cva_SA
  plot(focClim$bio16_cva_SA, focClim$bio15_cva_SA, xlim = c(0.25, 0.55), ylim = c(0.14, 0.24),
       axes = F, frame.plot = T, type = "n", xlab = "", ylab = "")
  text(grconvertX(0.075, "npc", "user"), grconvertY(0.925, "npc", "user"),
       labels = "C", font = 2, cex = 2)
  title(xlab = "Precipitation of Wettest Quarter", cex.lab = 1.5)
  title(xlab = "(Coefficient of variation)", cex.lab = 1, line = 4)
  title(ylab = "Precipitation Seasonality", cex.lab = 1.5, line = 4)
  title(ylab = "(Coefficient of variation)", cex.lab = 1)
  axis(1, lwd = 0, lwd.ticks = 1) 
  axis(2, las = 1, lwd = 0, lwd.ticks = 1)    
  fitBio16cva_Bio15cva <- lm(bio15_cva_SA ~ bio16_cva_SA, data = focClim)

  # polygon of confidence intervals
  x <- seq(min(focClim$bio16_cva_SA), max(focClim$bio16_cva_SA), l = 1e3)
  u <- predict(fitBio16cva_Bio15cva, new = data.frame(bio16_cva_SA = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitBio16cva_Bio15cva) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), 
                        upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "white", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  lines(x, u, lwd = 2)
  points(focClim$bio16_cva_SA, focClim$bio15_cva_SA, pch = 21, col = "black", 
         bg = "white", cex = 1.5, lwd = 2)

  # R2-value
  text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1), 
	     labels = bquote(italic(R)^2 == .(round(summary(fitBio16cva_Bio15cva)$r.squared, 2))))

  # Panel D. Latitude vs bio16_cva_SA
  plot(thinnedOccClimate$Latitude, thinnedOccClimate$bio16_cva_SA, 
       xlim = c(32, 44), ylim = c(0.25, 0.55), axes = F, frame.plot = T, type = "n", 
       xlab = "", ylab = "")
  text(grconvertX(0.075, "npc", "user"), grconvertY(0.925, "npc", "user"),
       labels = "D", font = 2, cex = 2)
  title(xlab = "Latitude", cex.lab = 1.5)
  axis(1, at = seq(32, 44, 4), labels = c(expression(32*degree), 
		expression(36*degree), expression(40*degree), expression(44*degree)), 
		lwd = 0, lwd.ticks = 1) 
  title(ylab = "Precipitation of Wettest Quarter", cex.lab = 1.5, line = 4)
  title(ylab = "(Coefficient of variation)", cex.lab = 1)
  axis(2, las = 1, lwd = 0, lwd.ticks = 1)    
  points(thinnedOccClimate$Latitude, thinnedOccClimate$bio16_cva_SA, 
         col = "grey", pch = 19)
  points(focClim$Latitude, focClim$bio16_cva_SA, pch = 21, col = "black", 
         bg = "white", cex = 1.5, lwd = 2)
  legend(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), pch = c(19, 21),
         col = c("grey", "black"), bg = "white", bty = "n", xjust = 1, yjust = 1,
         legend = c("All occurrences", "Focal populations"))
  
  dev.off()
  
##### Figure of bio11_cva_SA versus PC1 #####
  
  pdf(paste(pathFig, "/FigureS_bio11PC1.pdf", sep = ""), 5, 5)

  par(mar = c(5, 5, 1, 1))
  plot(focClim$bio11_cva_SA, focClim$PC1, xlim = c(0.002, 0.005), ylim = c(-4, 4),
       axes = F, frame.plot = T, type = "n", xlab = "", ylab = "")
  title(xlab = "Temperature of Coldest Quarter", cex.lab = 1.5)
  title(xlab = "(Coefficient of variation)", cex.lab = 1, line = 4)
  title(ylab = "TraitPC1", cex.lab = 1.5, line = 3)
  title(ylab = substitute(plain(Principal~Component~1)~group("(", x, ")"), 
    list(x = pve)), line = 2)
  axis(1, lwd = 0, lwd.ticks = 1) 
  axis(2, at = seq(-4, 4, 2), las = 1, lwd = 0, lwd.ticks = 1)    
  fitBio11cva <- lm(PC1 ~ bio11_cva_SA, data = focClim)
  
  # polygon of confidence intervals
  x <- seq(min(focClim$bio11_cva_SA), max(focClim$bio11_cva_SA), l = 1e3)
  u <- predict(fitBio11cva, new = data.frame(bio11_cva_SA = x))
  s <- sqrt(sapply(x, function(X) t(c(1, X)) %*% vcov(fitBio11cva) %*% c(1, X)))
  confInt <- data.frame(lowerCI = qnorm(0.025, u, s), 
                        upperCI = qnorm(0.975, u, s))
  polygon(c(x, rev(x)), c(confInt$lowerCI, rev(confInt$upperCI)), 
    col = "white", border = NA)
  lines(x, confInt$lowerCI, lwd = 2, lty = 2)
  lines(x, confInt$upperCI, lwd = 2, lty = 2)
  points(x, u, type = "l", lwd = 2)
  points(focClim$bio11_cva_SA, focClim$PC1, pch = 21, col = "black", 
         bg = "white", cex = 1.5, lwd = 2)

  # R2-value
  text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1), 
	     labels = bquote(italic(R)^2 == .(round(summary(fitBio11cva)$r.squared, 2))))

  dev.off()

##### Table of important climatic variables #####

clim2sym <- read.csv(paste(pathMS, "/climVar2plot.csv", sep = ""), 
										 stringsAsFactors = F, row.names = 1)

tableS <- file(paste(pathMS, "/Tables/TableS_ClimVarImp.txt", sep = ""), open = "w")
for (i in 1:max(c(nImpLat, nImpPC1))) {
clim2sym[imp1[i], "name"]
	# Climate-Latitude
	if (nImpLat >= i) {
		cat(sprintf('%s ($\\%s$, %s) & ', clim2sym[imp1[i], "name"], clim2sym[imp1[i], "what"], clim2sym[imp1[i], "spatialScale"]), file = tableS)
	}else{
		cat(' & ', file = tableS)
	}

	# Climate-TraitPC1
	if (nImpPC1 >= i) {
		cat(sprintf('%s ($\\%s$, %s) \\\\\n', clim2sym[imp2[i], "name"], clim2sym[imp2[i], "what"], clim2sym[imp2[i], "spatialScale"]), file = tableS)
	}else{
		cat(' \\\\\n', file = tableS)		
	}
}
close(tableS)

##### Two-factor model #####

  fit2var1 <- lm(PC1 ~ bio15_cva_SA + bio11_cva_SA, data = focClim)
  fit2var2 <- lm(PC1 ~ bio16_cva_SA + bio11_cva_SA, data = focClim)

##### Figure of local versus neighborhood Climate-TraitPC1 and Climate-Latitude correlations #####
  
  pdf(paste(pathFig, "/Figure_LocalvNeighCorr.pdf", sep = ""), w = 8, h = 4, useDingbats = F)
  par(mai = c(1, 1, 0.2, 0.2), mfrow = c(1, 2))

  # Panel A. Local versus neighborhood Climate-TraitPC1 correlations

    # Vector of correlations between climVars and TraitPC1
    r <- apply(focClim[, climVars], 2, cor, y = focClim$PC1)

    # Matrix comparing correlation coefficients of local and neighborhood climate
    corTab <- as.data.frame(cbind(r[!grepl("_SA", climVars)], r[grepl("_SA", climVars)]))
    corTab <- abs(corTab)
    colnames(corTab) <- c("local", "neighborhood")
    corTab$type <- ifelse(grepl("_avg", rownames(corTab)), "avg", "cva")

    plot(corTab[, "local"], corTab[, "neighborhood"], xlim = c(0, 1), ylim = c(0, 1),
         xlab = "", ylab = "", type = "n", axes = F, frame.plot = T)
    abline(0, 1, lty = 2)
    axis(1, lwd = 0, lwd.ticks = 1)
    axis(2, lwd = 0, lwd.ticks = 1, las = 1)
    title(xlab = expression(paste("local |", italic(r)[paste(climate-TraitPC1)], "|")),
          ylab = expression(paste("neighborhood |", italic(r)[paste(climate-TraitPC1)], "|")),
          cex.lab = 1.5)
    points(corTab[, "local"], corTab[, "neighborhood"], pch = 21, col = "black",
           bg = ifelse(corTab$type == "avg", "black", "white"))

    legend(1, 0, xjust = 1, yjust = 0, legend = c("Avg.", "Var."), pch = 21, col = "black", 
           pt.bg = c("black", "white"))
    text(0.975, 0.95, labels = "1:1", pos = 1)
    text(0 + strwidth("A", cex = 2, font = 2) / 2, 1 - strheight("A", cex = 2, font = 2) / 2, 
         labels = "A", font = 2, cex = 2)
    
    # Is neighborhood greater than local |r|?  Paired t-test
    # Effective number of variables (Meff) from Chevrud et al 1983, 2001:
    # Meff = 1 + (M - 1) * ((1 - Var(lambda)) / M) 

    mat <- cor(focClim[, climVars]) # Correlation matrix
    M <- nrow(mat) # Correlation matrix rank
    lambda <- eigen(mat)$values # Eigenvalues
    Meff1 <- 1 + (M - 1) * (1 - var(lambda) / M) # Effective number of variables
    tt1 <- t.test(corTab$local, corTab$neighborhood, paired = T) # paired t-test
    pt1 <- pt(tt1$statistic, Meff1 / 2 - 1) * 2 # Corrected p-value

  # Panel B. Local versus neighborhood Climate-Latitude correlations

    # Vector of correlations between climVars and TraitPC1
    r <- apply(thinnedOccClimate[, climVars], 2, cor, y = thinnedOccClimate$Latitude)

    # Matrix comparing correlation coefficients of local and neighborhood climate
    corTab <- as.data.frame(cbind(r[!grepl("_SA", climVars)], r[grepl("_SA", climVars)]))
    corTab <- abs(corTab)
    colnames(corTab) <- c("local", "neighborhood")
    corTab$type <- ifelse(grepl("_avg", rownames(corTab)), "avg", "cva")

    plot(corTab[, "local"], corTab[, "neighborhood"], xlim = c(0, 1), ylim = c(0, 1),
         xlab = "", ylab = "", type = "n", axes = F, frame.plot = T)
    abline(0, 1, lty = 2)
    axis(1, lwd = 0, lwd.ticks = 1)
    axis(2, lwd = 0, lwd.ticks = 1, las = 1)
    title(xlab = expression(paste("local |", italic(r)[paste(climate-latitude)], "|")),
          ylab = expression(paste("neighborhood |", italic(r)[paste(climate-latitude)], "|")),
          cex.lab = 1.5)
    points(corTab[, "local"], corTab[, "neighborhood"], pch = 21, col = "black",
           bg = ifelse(corTab$type == "avg", "black", "white"))

    legend(1, 0, xjust = 1, yjust = 0, legend = c("Avg.", "Var."), pch = 21, col = "black", 
           pt.bg = c("black", "white"))
    text(0.975, 0.95, labels = "1:1", pos = 1)
    text(0 + strwidth("B", cex = 2, font = 2) / 2, 1 - strheight("B", cex = 2, font = 2) / 2, 
         labels = "B", font = 2, cex = 2)

    # Is neighborhood greater than local |r|?  Paired t-test
    # Effective number of variables (Meff) from Chevrud et al 1983, 2001:
    # Meff = 1 + (M - 1) * ((1 - Var(lambda)) / M) 

    mat <- cor(thinnedOccClimate[, climVars]) # Correlation matrix
    M <- nrow(mat) # Correlation matrix rank
    lambda <- eigen(mat)$values # Eigenvalues
    Meff2 <- 1 + (M - 1) * (1 - var(lambda) / M) # Effective number of variables
    tt2 <- t.test(corTab$local, corTab$neighborhood, paired = T) # paired t-test
    pt2 <- pt(tt2$statistic, Meff2 / 2 - 1) * 2 # Corrected p-value

    dev.off()

@

For Climate-Latitude analyses, we compiled a representative set of \Sexpr{2 * nrow(thinnedOccClimate)} recent (since 2000) known \textit{E. cardinalis} occurrences from a comprehensive set of herbarium records and an exhaustive field survey in 2010-11 \citep{Angert_ENM}. These occurrences were thinned by 50\% to correct for uneven sampling. For both Climate-TraitPC1 analyses (16 focal populations) and Climate-Latitude (many populations), we used a 90-m digital elevation model from HydroSHEDS \citep{Lehner_etal_2006} to extract elevation. Monthly interpolated climate layers were calculated using ClimateWNA version 5.30 \citep{Wang_etal_2012}, which accurately downscales climate data specifically for the rugged topography of western North America. For each occurence, we calculated bioclimatic variables using the biovars function in the R package \pkg{dismo} version 1.1-1 \citep{Hijmans_etal_2016}. We included \Sexpr{length(climVars) / 4} climatic factors, 9 from ClimateWNA and 15 bioclimatic variables (Table~\ref{table:TableS_ClimVars}). The bioclimatic variables included all permutations of two climatic factors, temperature and precipitation, and six temporal scales (annual average, coldest quarter, warmest quarter, wettest quarter, driest quarter, or seasonality) as well as mean diurnal range, isothermality, and annual temperature range. For each variable, we calculated both a 30-year normal by averaging annual values between 1981 and 2010 and 30-year coefficient of variation, a standardized metric of interannual climatic variation. Temperatures were converted to Kelvin to be on a ratio scale appropriate for calculating the coefficient of variation (CV). In total, the VSURF algorithm selected among \Sexpr{length(climVars)} climate variables: \Sexpr{length(climVars) / 4} climatic factors $\times$ 2 types (30-year average and CV) $\times$ 2 spatial scales (local and neighborhood).

%--------------------------------------------------
% Results
%--------------------------------------------------

\section*{Results}

\subsection*{A coordinated latitudinal cline in germination, growth, and photosynthesis}

There are strong genetically-based trait differences in time to germination, growth, and photosynthetic rate among populations of \textit{E. cardinalis}, as evidenced by large and significant population effects for these traits (Table~\ref{table:Table_AnovaSummary}). A single principal component captured \Sexpr{gsub("%", "", pve)}\% of the trait variation among populations, defining an axis of variation from fast to slow growth. A population's position along this axis strongly covaried with its latitude of origin; southern populations grew faster than northern populations (Fig~\ref{fig:Fig_PC1vLat}). There were similar latitudinal clines for individual traits underlying PC1 (\Cref{fig:FigS_GermLat,fig:FigS_LERlat,fig:FigS_SERlat,fig:FigS_Photolat}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Table of Population, Treatment, Population x Treatment interactions
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{table}[ht]
   \centering
   \topcaption[ANOVA summary]{Summary of Population, Treatment, and Population $\times$ Treatment effects. We used different statistical modeling for the diverse traits assayed -- glmer: generalized linear mixed model using the R package \pkg{lme4} \citep{Bates_etal_2015}; lmer: linear mixed model using the R package \pkg{lme4} \citep{Bates_etal_2015}; survreg: survival regression using the R package \pkg{survival} \citep{Therneau_2015}. Note that temperature and water treatments were imposed after germination, hence are not applicable to this trait. Complete analysis of variance/deviance tables for each trait are available in the Supporting Information. Key to statistical significance: *$P < 0.05$; ** $P < 0.01$; *** $P < 0.001$}
   \resizebox{6.5in}{!}{
   \begin{tabular}{@{} lllllll @{}}
      \toprule
    {\raggedleft Trait}                          & Germination & Leaf expansion & Stem elongation & Photosynthesis & Mortality \\
    Statistical model              & survreg     & lmer           & lmer            & lmer           & glmer \\
      \midrule
  Population                       &
    \Sexpr{sigStar(anova(fitGerm)["Pr(>Chi)"]["PopID",])}  &
    \Sexpr{sigStar(fitLER$anova.table["PopID", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID", "Pr(>Chisq)"])} & \\
  Temperature                      &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["TempTrt", "Pr(>F)"])}    &
    \Sexpr{sigStar(fitSER$anova.table["TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["TempTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["TempTrt", "Pr(>Chisq)"])} & \\
  Water                            &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["WaterTrt", "Pr(>F)"])}   &
    \Sexpr{sigStar(fitSER$anova.table["WaterTrt", "Pr(>F)"])}   &
    \Sexpr{sigStar(fitPhoto$anova.table["WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["WaterTrt", "Pr(>Chisq)"])} & \\
  Pop $\times$ Temp                &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["PopID:TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID:TempTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID:TempTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID:TempTrt", "Pr(>Chisq)"])} & \\
  Pop $\times$ Water               &
     NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["PopID:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID:WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID:WaterTrt", "Pr(>Chisq)"])} & \\
  Temp $\times$ Water              &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["TempTrt:WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["TempTrt:WaterTrt", "Pr(>Chisq)"])} & \\
  Pop $\times$ Temp $\times$ Water &
    NA                                                          &
    \Sexpr{sigStar(fitLER$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitSER$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} &
    \Sexpr{sigStar(fitPhoto$anova.table["PopID:TempTrt:WaterTrt", "Pr(>F)"])} & 
    \Sexpr{sigStar(Anova(fitMort, type = 2)["PopID:TempTrt:WaterTrt", "Pr(>Chisq)"])} & \\
      \bottomrule
   \end{tabular} }
   \label{table:Table_AnovaSummary}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of latitude versus PC1 (growth axis)
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_PC1vLat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations grow faster]{Trait variation, from fast to slow growth, is closely associated with latitude. Each point is a population's latitude of origin (x-axis) and position along the slow to fast growth axis (y-axis), defined as Principal Component 1 of four traits (see Material and Methods). The line and 95\% confidence intervals were estimated using linear regression.}
	\label{fig:Fig_PC1vLat}
\end{figure}

\subsection*{Little evidence for variation in plasticity}

In contrast to the genetic variation in trait means described above, we found little evidence of G$\times$E in \textit{E. cardinalis}. There were only two statistically significant Population $\times$ Treatment interactions (Table~\ref{table:Table_AnovaSummary}, Fig.~\ref{fig:FigS_RxnNorms}), but these were not strong compared to Population and Temperature effects. Otherwise, populations responded similarly to treatments: faster growth in the hot treatment, slower growth in the dry treatment, and high mortality in the hot, dry treatment (Table~\ref{table:Table_AnovaSummary}). Complete ANOVA tables are available in the Supporting Information (\Cref{table:TableS_fitLER,table:TableS_fitSER,table:TableS_fitPhoto,table:TableS_fitMort})

\subsection*{Neighborhood climatic variability best explains latitudinal cline}

Interannual variation in climate averaged over each populations's climatic neighborhood correlated most strongly with trait variation and latitude of \textit{E. cardinalis} occurrences (Fig.~\ref{fig:Fig_ClimVarImp}, Table~\ref{table:TableS_ClimVarImp}). All \Sexpr{nImpLat} Climate-Latitude and \Sexpr{nImpPC1} Climate-TraitPC1 variables were neighborhood rather than local variables (Fig.~\ref{fig:Fig_ClimVarImp}). In fact, neighborhood climate almost always correlated better with TraitPC1 and Latitude than local climate (Fig.~\ref{fig:Fig_LocalvNeighCorr}). On average, neighborhood Climate-TraitPC1 correlation coefficients were \Sexpr{-round(tt1$estimate, 2)} higher than correlations with local-scale climate variables (paired \textit{t}-test, $t = \Sexpr{-round(tt1$statistic, 2)}$, d.f. = \Sexpr{round(Meff1 / 2 - 1, 1)}, $P = \Sexpr{round(pt1 * 10 ^ -floor(log10(pt1)), 2)} \times 10 ^ {\Sexpr{floor(log10(pt1))}}$). Likewise, neighborhood Climate-Latitude correlation coefficients were \Sexpr{-round(tt2$estimate, 2)} higher than those for local-scale climate (paired \textit{t}-test, $t = \Sexpr{-round(tt2$statistic, 2)}$, d.f. = \Sexpr{round(Meff2 / 2 - 1, 1)}, $P = \Sexpr{round(pt2 * 10 ^ -floor(log10(pt2)), 2)} \times 10 ^ {\Sexpr{floor(log10(pt2))}}$). 
Among Climate-Latitude and Climate-TraitPC1 variables, neighborhood climatic variability over 30 years (1981--2010) in either winter precipitation (bio16$_\sigma$) and/or temperature (bio11$_\sigma$) are the strongest candidates to explain the latitudinal cline in \textit{E. cardinalis} (see Table~\ref{table:TableS_ClimVars} for a key to climate variable abbreviations). Note that the coefficient of variation of a climatic factor is subscripted with $\sigma$ whereas the mean is subscripted with $\mu$. More specifically, greater winter precipitation variability and lower winter temperature variability are associated with Southern latitudes and higher TraitPC1 values (Fig.~\ref{fig:Fig_ClimPC1}A,B). Neighborhood interannual variation in winter precipitation (bio16$_\sigma$) was the most important Climate-Latitude variable (Fig.~\ref{fig:Fig_ClimVarImp}A). However, neighborhood bio16$_\sigma$ did not overlap with Climate-TraitPC1 variables (Fig.~\ref{fig:Fig_ClimVarImp}B). We nevertheless consider it a plausible candidate for two reasons. First, neighborhood bio16$_\sigma$ correlated strongly with TraitPC1 (Fig.~\ref{fig:Fig_ClimPC1}A). Second, one of the most important Climate-TraitPC1 variables (neighborhood bio15$_\sigma$; Fig.~\ref{fig:Fig_ClimPC1}B,C) is very similar to bio16$_\sigma$. In Mediterranean climates like California, most precipitation occurs in the wettest quarter (winter), so years with low winter precipitation also have low precipitation seasonality. Hence, highly variable year-to-year winter precipitation at lower latitude (Fig.~\ref{fig:Fig_ClimPC1}D) is closely associated with large swings in precipitation seasonality (Fig.~\ref{fig:Fig_ClimPC1}C). 

Interannual variation in temperature of the coldest quarter (neighborhood bio11$_\sigma$) is another plausible candidate because it was the only variable in both Climate-Latitude and Climate-TraitPC1 analyses (Fig.~\ref{fig:Fig_ClimVarImp}). Neighborhood bio11$_\sigma$ explained more variation in TraitPC1 than latitude (latitude $r^2=\Sexpr{round(summary(fitLat)["r.squared"]$r.squared, 2)}$ vs. bio11$_\sigma$ $r^2=\Sexpr{round(summary(fitBio11cva)["r.squared"]$r.squared, 2)}$; Fig.~\ref{fig:FigS_bio11PC1}), whereas neighborhood bio16$_\sigma$ did slightly worse (bio16$_\sigma$ $r^2=\Sexpr{round(summary(fitBio16cva)["r.squared"]$r.squared, 2)}$). Models using bio15$_\sigma$ or bio11$_\sigma$ to predict TraitPC1 also had significantly lower Akaike Information Criteria (AIC) than the latitude model (AIC of different models -- bio15$_\sigma$: \Sexpr{round(AIC(fitBio15cva), 1)}; bio11$_\sigma$: \Sexpr{round(AIC(fitBio11cva), 1)}; latitude: \Sexpr{round(AIC(fitLat), 1)}). The best two-factor model including both neighborhood bio15$_\sigma$ and bio11$_\sigma$ did not significantly improve explanatory power ($r^2=\Sexpr{round(summary(fit2var1)["r.squared"]$r.squared, 2)}$, AIC$=\Sexpr{round(AIC(fit2var1), 1)}$). In summary, either variation in precipitation or temperature seasonality may be important selective agents, but there is no strong evidence that they are both important. The most important Climate-TraitPC1 variable, neighborhood variation in mean diurnal range (bio2$_\sigma$; Fig.~\ref{fig:Fig_ClimVarImp}B) did not have any obvious similarity to Climate-Latitude variables. Given the large number of potential associations, we therefore think this may be a spuriously strong relationship.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of climatic variable importance
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_ClimVarImp.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Climatic variation integrated over climatic neighborhood is closely correlated with latitude of \textit{E. cardinalis} and trait variation]{Climatic variation integrated over climatic neighborhood is closely correlated with latitude of \textit{E. cardinalis} and trait variation.  A. Using Random Forest regression, we identified \Sexpr{nImpLat} climatic variables significantly (high importance) associated with latitude of \textit{E. cardinalis} occurrences. B. Only one of of the most important Climate-Latitude variables (in bold) was among the most important Climate-TraitPC1 variables. Variable importance is defined as the average amount a climate variable reduces mean-squared error in the predicted response (TraitPC1 or Latitude), compared to a randomly permuted dataset, across all trees in the random forest (see \citeauthor{Genuer_etal_2015} [\citeyear{Genuer_etal_2015}] for further detail). Note that the Importance values in A and B are not comparable because the dependent variables (Latitude and Trait PC1, respectively) are on different scales. Climatic variables (left of A; right of B) are defined by four qualities: Climatic factor -- Temperature (Temp), Precipitation (Prec), Heating degree-days (Heating), Snow (precipitation as snow); Temporal scale -- Annual, Coldest quarter (Cold Quar), Warmest Quarter (Warm Quar), Wettest quarter (Wet Quar), Driest Quarter (Dry Quar), or Seasonality; Type -- 30-year average (Avg.) or coefficient of variation (Var.); Spatial scale -- local or 62-km radius climatic neighborhood.}
	\label{fig:Fig_ClimVarImp}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of local versus neighborhood Climate-TraitPC1 and Climate-Latitude correlations
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_LocalvNeighCorr.pdf}}
\usefont{T1}{phv}{m}{n}
\fontsize{10}{12}
\selectfont
\caption[Local versus neighborhood correlations]{Neighborhood climate predicts TraitPC1 (`Climate-trait', panel A) and Latitude of occurences (`Climate-latitude', panel B) better than local climate. Each point is the absolute value of the Pearson correlation coefficient (|$r$|) between TraitPC1 (A) or latitude (B) for \Sexpr{length(climVars)/4} climatic factors, for which we used both the 30-year mean (closed circles) and coefficient of variation (open circles). Most points lie above the 1:1 line, indicating stronger correlations with neighborhood compared to local climate. Neighborhood climate was integrated over a 62-km radius around focal populations (see text for further detail).}
\label{fig:Fig_LocalvNeighCorr}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Figure of key Climate-TraitPC1 relationships
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{figure}[h!]
	\centerline{\includegraphics[width=1\textwidth]{Figures/Figure_ClimPC1.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Variation in precipitation is correlated with TraitPC1 and latitude]{Variation in precipitation is correlated with TraitPC1 and latitude. A. Greater values of TraitPC1 are associated with greater interannual variation in precipitation of the wettest quarter. This was the most important Climate-Latitude variable, but not among the most important Climate-TraitPC1 variables. B. However, a closely related parameter, interannaul variation in precipitation seasonality, was among the most important Climate-TraitPC1 variables. C. Across focal populations, variation in precipitation of the wettest quarter and seasonality are closely correlated. D. Southern populations of \textit{E. cardinalis} experience much greater interannual variationi in precipitation. In all panels, we report climatic neighborhood values (see Material and Methods). Regression lines, 95\% confidence intervals, and coefficients of determination ($R^2$) were calculated using linear regression.}
	\label{fig:Fig_ClimPC1}
\end{figure}

%--------------------------------------------------
% Discussion
%--------------------------------------------------

\section*{Discussion}

We found evidence for one of two common signatures of local adaptation in the perennial herb \textit{Erythranthe cardinalis}. Latitudinal clines in germination rate, photosynthesis, and growth suggest adaptive differentiation in important physiological traits of the species. However, we found little evidence for variation in plasticity to temperature or drought. Due to low replication within families, we did not have power to assess within-population genotype-by-environment interactions, which may be present. As we discuss below, low variation in plasticity among populations may indicate that some dimensions of the fundamental abiotic niche are relatively conserved. Note that statistical power to detect significant plasticity is lower than that for differences in trait means. However, the fact that the Population and Temperature effects were often highly significant ($P \ll 0.001$ in most cases) suggests that statistical power alone cannot explain low variation in plasticity. Finally, our results suggest that neighborhood-scale climate and interannual variation are more important selective agents than local averages. In the paragraphs that follow, we tie these results into the broader threads of evolutionary theory that might help explain why variation in physiological trait means changes clinally, whereas plastic responses to temperature and drought are relatively static. One caveat to bear in mind is that we are limited by the size of the climate grid ($\approx 90~\text{m}^2$) and therefore unable to detect very fine-scale local adaptation.

Evolutionary theory indicates that the shape of fitness tradeoffs, demography, and gene flow can constrain adaptation \citep{Levins_1968, Ronce_Kirkpatrick_2001, Lenormand_2002} and hence the type of variation maintained within species. Specifically, adaptive variation can be maintained by spatially varying selection if tradeoffs are not too strong, demography is symmetric, and/or maladaptive gene flow is low. Strong tradeoffs can prevent local adaptation in spatially variable environments because selection favors habitat specialists that track a specific habitat regardless of its frequency in the environment \citep{Levins_1968}. For example, a riparian specialist may experience similar selection in rivers of high rainfall regions and deserts, even though the habitat is much rarer in the latter. In \textit{E. cardinalis} we found substantial genetically based variation among populations along a phenotypic axis from fast to slow growth that varied over a large spatial scale (Fig.~\ref{fig:Fig_PC1vLat}). If this variation is adaptive, it suggests one of several possibilities to investigate in the future: the fitness tradeoff between low versus high latitude environments is not too strong nor swamped by demographic asymmetry or maladaptive gene flow. That is, alleles favoured at one latitude are not strongly selected against when they flow to another population, allowing locally adaptive genetic variation to be maintained by spatially heterogenous selection. We also know from previous work that population size does not vary strongly with latitude (Angert, unpub. data). Gene flow appears to be high, but attenuates at broad spatial scales, especially between Southern ($<35\degree$N) and Northern portions of the range \citep{Paul_etal_2016}.  

Nevertheless, local gene flow from similar environments may shape how selection varies with latitude. Theory predicts that populations will not be perfectly adapted to their immediate habitat when there is gene flow from surrounding populations with different optima \citep{Lenormand_2002}. With spatial heterogeneity and gene flow, traits will not covary perfectly with the local optimum \citep{Slatkin_1978, Paul_etal_2011, Hadfield_2016}, but should instead better match the average environment experienced by nearby populations connected through gene flow, which we refer to as the climatic neighborhood. Gene flow and spatial heterogeneity may therefore be important in maintaining genetic variation \citep{Yeaman_Jarvis_2006}. As this hypothesis predicts, climatic neighborhoods (62-km buffer around populations) correlated with traits and latitude of occurrences better than local climate (Fig.~\ref{fig:Fig_ClimVarImp}). We interpret this as suggestive evidence that gene flow between neighboring \textit{E. cardinalis} populations shapes selection -- populations are locally adapted to prevailing climate in their neighborhood, but perhaps not perfectly adapted to their local climate. This may not greatly constrain local adaptation because local and neighborhood climate values were generally similar in \textit{E. cardinalis} populations (Fig.~\ref{fig:Fig_LocalvNeighCorr}), at least at the resolution of ClimateWNA (90 m$^2$). Therefore, we would predict in reciprocal transplants that populations whose local climate is farther from their neighborhood average would be less well adapted than those close to their neighborhood average.

It is reasonable to predict that southern populations, which appear to experience more frequent drought years (see below), might have physiological adaptations to respond to drought stress to survive and grow in drier soil. We found no evidence for this type of drought tolerance; all populations responded to drought and temperature similarly (Table~\ref{table:Table_AnovaSummary}). Plants grew faster in the Hot treatment, but there was little effect of drought on growth. Rather, the effects of drought took longer to materialize but resulted in high mortality, especially in the Hot treatment. However, there was no differential mortality among populations in this treatment. Although our results indicate that this axis of the species niche may be constrained, plants have multiple ways to resist drought through both tolerance and escape \citep{Ludlow_1989, Kooyers_2015}. Next, we consider why drought tolerance may less important in local adaptation than a form of escape for this species.

We hypothesize that tolerance to dry soil may be constrained by a combination of strong fitness tradeoffs, demographic asymmetry, and gene flow. Soil moisture in riparian habitats where \textit{E. cardinalis} lives is highly heterogeneous at very small spatial scales (several meters). Plants in the stream never have to tolerate drought whereas plants only a few meters away may experience extreme drought since there is little direct precipitation during the growing season in Mediterranean climates of western North America. We hypothesize that alleles confering greater drought tolerance may be quite costly in well-watered soils, and \textit{vice versa}, leading to strong fitness tradeoffs. Such tradeoffs would promote specialization to one soil moisture or another, thereby inhibiting the evolution of broad environmental tolerance within a population. Demography and gene flow may reinforce niche conservatism. A new mutant with increased drought tolerance that could survive at the resource-poor margin of a population would likely be demographically overwhelmed by the larger census populations that can be maintained in higher-resource environments. Infrequent wet years may also produce most seeds, so selection is weighted towards alleles that have high fitness in the wet environment, even if dry years are more frequent \citep{Templeton_Levin_1979, Brown_Venable_1986}. However, demographic asymmetry should equally hinder the evolution of both drought tolerance and escape, so it should not explain why one mechanism evolves but not the other. Finally, gene flow, which is generally high among \textit{E. cardinalis} populations within the same ecoregion \citep{Paul_etal_2016}, will thwart local adaptation and reinforce specialization. Thus, the spatial grain of the environment, demographic asymmetry, and gene flow may conspire to constrain local adaptation along this environmental axis. Consistent with this hypothesis, recent record-setting droughts have caused the decline or even local extinction of some natural populations of \textit{E. cardinalis} \citep{Sheth_Angert_2017}.

In sum, these results indicate that genetic differences in physiology and growth are better candidates than plastic responses to temperature and drought as mediators of local adaptation to climate in \textit{E. cardinalis}. Next, we would like to understand why variation in these particular traits may be adaptive. We argue that temporally more variable environments, as experienced by southern populations, select for a more `annualized' life-history strategy, a form of drought escape. Demographic observations in natural populations of \textit{E. cardinalis} reveal that southern populations tend to flower earlier at a smaller size, while northern populations invest more in vegetative growth \citep{Sheth_Angert_2017}. The association between position along the `fast-slow' continuum and associated traits in \textit{E. cardinalis} is similar to interspecific relationships between growth, functional traits, and life history \citep{Adler_etal_2014, Salguero-Gomez_etal_2016}. However, we cannot exclude unexplored factors (e.g. edaphic conditions, competitors, pollinators, etc.) which may also contribute to the latitudinal cline. 

Greater investment in aboveground growth, as opposed to belowground storage for future seasons, may be favoured in climates with more frequent drought years, but maladaptive in climates with more consistent precipitation. This is a form of drought escape in that plants are investing more reproduction in the present to avoid possible drought in subsequent years. Suppose plants that grow quickly and allocate new resources to continued growth rather than storage have higher fitness over a single growing season. However, by not allocating resources to storage, these fast-growing plants begin future seasons at a deficit. Therefore, in a stable environment where winter survivorship is assured in most years, failure to store resources may reduce lifetime fitness. But for perennial herbs in Mediterranean climates, a dry winter (rainy season) can kill the rhizomes (underground stems that store nutrients for future growth) before emergence or aboveground stems before flowering. If drought years occur frequently enough, selection may favour the fast-growing strategy because there is no advantage to storage if drought kills plants before flowering. Considering life-history strategy as a continuum from no storage (annual) to lots of storage (perennial), we hypothesize that the optimal allocation to aboveground growth is more `annualized' in southern climates that have greater interannual variation in precipitation. This scenario differs from classic drought escape syndromes in which plants speed up development early in the season before the onset of drought.

The hypothesis that greater precipitation variability selects for an annualized life history is tentative, but consistent with theory and data from other species. Life history theory shows that less variable environments are one factor that favours the evolution of perenniality \citep{Stearns_1976, Iwasa_Cohen_1989, Friedman_Rubin_2015}. Populations of the perennial \textit{Plantago asiatica} show a similar latitudinal cline in growth and allocation to storage \citep{Sawada_etal_1994}, though these authors attribute the cline to variation in growing season length. There are also life history clines in the closely related species \textit{E. guttata}, but the underlying traits and climatic drivers are quite different. Annual \textit{E. guttata} flower sooner and produce fewer stolons in response to climates with shorter seasons and more intense summer drought \citep{Lowry_Willis_2010, Friedman_etal_2015, Kooyers_etal_2015}. In contrast, there are no truly annual (monocarpic and semelparous) populations of \textit{E. cardinalis}. Rather, our hypothesis states that climatic variability selects on quantitative variation in allocation to growth versus storage. This hypothesis makes several independent, testable predictions. The allocation tradeoff predicts that northern populations will provision more photosynthetic assimilate to rhizomes compared with southern populations. If southern populations are indeed more `annualized' because more frequent droughts cause mortality, then we predict that species distribution models using recent climate would best predict occurrences in the south, whereas longer term climate would be a better predictor in the north. Finally, we predict that southern populations would show greater variation in the size of recruits and higher maximum population growth rates.

In summary, we found evidence for a coordinated latitudinal cline in germination rate, photosynthesis, and growth, suggesting local adaptation. We therefore predict to find different optima for  these traits in different climates. We did not find evidence that the relative performance of populations shifts with temperature or watering regime, suggesting relatively little variation in plasticity. Exploratory analysis implicate that more variable precipitation regimes at lower latitude drive much of the latitudinal cline, though other climatic factors could also contribute. Interestingly, the climatic neighborhood may shape selective pressures more than local climate. In the future, we will use field experiments to test whether greater variation in precipitation selects for faster growth and if selection on temperature/drought responses does not vary among populations. By doing so, we aim to understand why certain physiological and developmental mechanisms, but not others, contribute to local adaptation.

%--------------------------------------------------
% References
%--------------------------------------------------

\setlength{\bibsep}{6pt}
\bigskip

\bibliography{refs}
\bibliographystyle{evolution}

\clearpage

%--------------------------------------------------
% Supporting Information
%--------------------------------------------------

\section*{Supporting Information}

% Modify and restart table/figure numbering for appendixes
\renewcommand\thefigure{S\arabic{figure}}    
\renewcommand\thetable{S\arabic{table}}    
\renewcommand\theequation{S\arabic{equation}}    
\setcounter{table}{0}    
\setcounter{equation}{0}
\setcounter{figure}{0}

%--------------------------------------------------
% Supporting Tables
%--------------------------------------------------

\subsection*{Supporting Tables}

% DIC table for initial size (LLL on 5/12)
\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[DIC table for initial size of seedlings]{Initial size of seedlings did not vary among Populations, Families, or Treatments. We used a censored Gaussian model of initial size at the outset of the experiment (longest leaf length of the first true leaves). The model was censored because we could not accurately measure leaves less than 0.25 mm with digital callipers (217 of 702, 30.9\%, were too small). We fit models using a Bayesian MCMC method implemented using the MCMCglmm function with default priors in the R package \pkg{MCMCglmm} version 2.17 \citep{Hadfield_2010}. We estimated the posterior distribution from 1000 samples of an MCMC chain run for $10 ^ 5$ steps after a $10^4$ step burn-in. We used step-wise backward elimination procedure to find the best-supported model according to Deviance Information Criterion (DIC).}
	\begin{center}
	\begin{tabular}{>{\everypar{\hangindent1cm}{}\raggedright}p{6cm}lc}
	\toprule

	\input{Tables/Table_InitialSize.txt}

	\bottomrule
	\end{tabular}
	\end{center}
	\label{table:TableS_InitialSize}
\end{table}

\begin{table}[ht]
	\caption[Climate variables]{Climatic variables used }
   \centering
   \begin{tabular}{@{} lp{12cm} @{}}
      \toprule
   Abbreviation & Climate variable \\
      \midrule
  % climateWNA VARIABLES:
	DD\_0	& degree-days below 0\celsius (chilling degree-days)  \\
	DD5	  & degree-days above 5\celsius (growing degree-days)   \\
	DD\_18	& degree-days below 18\celsius (heating degree-days)  \\
	DD18	& degree-days above 18\celsius (cooling degree-days)  \\
	NFFD	& number of frost-free days                           \\
	PAS	  & precipitation as snow (mm) between August in previous year and July in current  \\
	Eref	& Hargreaves reference evaporation (mm) \\
	CMD	  & Hargreaves climatic moisture deficit (mm) \\
	RH    &	mean annual relative humidity \\
  % BIOCLIM VARIABLES:
  bio1	&	annual mean temperature	\\
	bio2	&	mean diurnal range (mean of monthly (max temp - min temp))	\\
	bio3	&	isothermality (bio2/bio7) (* 100)	\\
	bio4	&	temperature seasonality (standard deviation *100)	\\
	bio5	&	max temperature of warmest month	\\
	bio6	&	min temperature of coldest month	\\
	bio7	&	temperature annual range (bio5-bio6)	\\
	bio8	&	mean temperature of wettest quarter	\\
	bio9	&	mean temperature of driest quarter	\\
	bio10	&	mean temperature of warmest quarter	\\
	bio11	&	mean temperature of coldest quarter	\\
	bio12	&	annual precipitation	\\
	bio15	&	precipitation seasonality (coefficient of variation)	\\
	bio16	&	precipitation of wettest quarter	\\
	bio17	&	precipitation of driest quarter	\\
	bio18	&	precipitation of warmest quarter	\\
	bio19	&	precipitation of coldest quarter	\\
	    \bottomrule
   \end{tabular}
   \label{table:TableS_ClimVars}
\end{table}

% ANOVA table for LER

<<TableS_fitLER, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=
	
  # this is sourced above. remove from here once testing is over  
  source(paste(pathMS, "/functions.R", sep = ""))

  # Convert numbers from R to LaTeX characters
  lerTab <- data.frame(ss = sigDig(fitLER$anova.table$`Sum Sq`),
                       ms = sigDig(fitLER$anova.table$`Mean Sq`),
                       df1 = sigDig(fitLER$anova.table$NumDF),
                       df2 = sigDig(fitLER$anova.table$DenDF),
                       fval = sigDig(fitLER$anova.table$F.value),
                       pval = pval2latex(fitLER$anova.table$`Pr(>F)`),
                       stringsAsFactors = F)
  rownames(lerTab) <- rownames(fitLER$anova.table)

@

\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[ANOVA table, leaf expansion rate]{Analysis of varianace (ANOVA) table on leaf expansion rate (LER) using \pkg{lmerTest} \citep{Kuznetsova_etal_2016}. Family and Block were included as random effects. Abbreviations: SS = sum of squares; MS = mean sum of squares (SS / df1); df1 = numerator degrees of freedom; df2 = denominator degrees of freedom.}
	\begin{center}
	\begin{tabular}{lcccccc}
	\toprule

  & SS & MS & df1 & df2 & \em{F}-value & $P$-value \\ \midrule
  Day to Germination & \Sexpr{lerTab["AvgGermDay", "ss"]} & 
    \Sexpr{lerTab["AvgGermDay", "ms"]} & \Sexpr{lerTab["AvgGermDay", "df1"]} & 
    \Sexpr{lerTab["AvgGermDay", "df2"]} & \Sexpr{lerTab["AvgGermDay", "fval"]} &
    \Sexpr{lerTab["AvgGermDay", "pval"]} \\
  Population & \Sexpr{lerTab["PopID", "ss"]} & \Sexpr{lerTab["PopID", "ms"]} & 
    \Sexpr{lerTab["PopID", "df1"]} & \Sexpr{lerTab["PopID", "df2"]} & 
    \Sexpr{lerTab["PopID", "fval"]} & \Sexpr{lerTab["PopID", "pval"]} \\
  Temperature & \Sexpr{lerTab["TempTrt", "ss"]} & 
    \Sexpr{lerTab["TempTrt", "ms"]} & \Sexpr{lerTab["TempTrt", "df1"]} &
    \Sexpr{lerTab["TempTrt", "df2"]} & \Sexpr{lerTab["TempTrt", "fval"]} &
    \Sexpr{lerTab["TempTrt", "pval"]} \\
  Water & \Sexpr{lerTab["WaterTrt", "ss"]} & \Sexpr{lerTab["WaterTrt", "ms"]} & 
    \Sexpr{lerTab["WaterTrt", "df1"]} & \Sexpr{lerTab["WaterTrt", "df2"]} & 
    \Sexpr{lerTab["WaterTrt", "fval"]} & \Sexpr{lerTab["WaterTrt", "pval"]} \\
  Temperature $\times$ Water & \Sexpr{lerTab["TempTrt:WaterTrt", "ss"]} & 
    \Sexpr{lerTab["TempTrt:WaterTrt", "ms"]} & 
    \Sexpr{lerTab["TempTrt:WaterTrt", "df1"]} & 
    \Sexpr{lerTab["TempTrt:WaterTrt", "df2"]} & 
    \Sexpr{lerTab["TempTrt:WaterTrt", "fval"]} &
    \Sexpr{lerTab["TempTrt:WaterTrt", "pval"]} \\
  Population $\times$ Temperature & \Sexpr{lerTab["PopID:TempTrt", "ss"]} & 
    \Sexpr{lerTab["PopID:TempTrt", "ms"]} & 
    \Sexpr{lerTab["PopID:TempTrt", "df1"]} & 
    \Sexpr{lerTab["PopID:TempTrt", "df2"]} & 
    \Sexpr{lerTab["PopID:TempTrt", "fval"]} &
    \Sexpr{lerTab["PopID:TempTrt", "pval"]} \\
  Population $\times$ Water & \Sexpr{lerTab["PopID:WaterTrt", "ss"]} & 
    \Sexpr{lerTab["PopID:WaterTrt", "ms"]} & 
    \Sexpr{lerTab["PopID:WaterTrt", "df1"]} &
    \Sexpr{lerTab["PopID:WaterTrt", "df2"]} &
    \Sexpr{lerTab["PopID:WaterTrt", "fval"]} & 
    \Sexpr{lerTab["PopID:WaterTrt", "pval"]} \\
  Population $\times$ Temperature $\times$ Water & 
    \Sexpr{lerTab["PopID:TempTrt:WaterTrt", "ss"]} & 
    \Sexpr{lerTab["PopID:TempTrt:WaterTrt", "ms"]} & 
    \Sexpr{lerTab["PopID:TempTrt:WaterTrt", "df1"]} & 
    \Sexpr{lerTab["PopID:TempTrt:WaterTrt", "df2"]} & 
    \Sexpr{lerTab["PopID:TempTrt:WaterTrt", "fval"]} & 
    \Sexpr{lerTab["PopID:TempTrt:WaterTrt", "pval"]} \\
	\bottomrule
	\end{tabular}
	\label{table:TableS_fitLER}
	\end{center}
\end{table}

% ANOVA table for SER

<<TableS_fitSER, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=
	
  # Convert numbers from R to LaTeX characters
  serTab <- data.frame(ss = sigDig(fitSER$anova.table$`Sum Sq`),
                       ms = sigDig(fitSER$anova.table$`Mean Sq`),
                       df1 = sigDig(fitSER$anova.table$NumDF),
                       df2 = sigDig(fitSER$anova.table$DenDF),
                       fval = sigDig(fitSER$anova.table$F.value),
                       pval = pval2latex(fitSER$anova.table$`Pr(>F)`),
                       stringsAsFactors = F)
  rownames(serTab) <- rownames(fitSER$anova.table)

@

\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[ANOVA table, stem elongation rate]{Analysis of varianace (ANOVA) table on stem elongation rate (SER) using \pkg{lmerTest} \citep{Kuznetsova_etal_2016}. Family and Block were included as random effects. Abbreviations: SS = sum of squares; MS = mean sum of squares (SS / df1); df1 = numerator degrees of freedom; df2 = denominator degrees of freedom.}
	\begin{center}
	\begin{tabular}{lcccccc}
	\toprule

  & SS & MS & df1 & df2 & \em{F}-value & $P$-value \\ \midrule
  Day to Germination & \Sexpr{serTab["AvgGermDay", "ss"]} & 
    \Sexpr{serTab["AvgGermDay", "ms"]} & \Sexpr{serTab["AvgGermDay", "df1"]} & 
    \Sexpr{serTab["AvgGermDay", "df2"]} & \Sexpr{serTab["AvgGermDay", "fval"]} &
    \Sexpr{serTab["AvgGermDay", "pval"]} \\
  Population & \Sexpr{serTab["PopID", "ss"]} & \Sexpr{serTab["PopID", "ms"]} & 
    \Sexpr{serTab["PopID", "df1"]} & \Sexpr{serTab["PopID", "df2"]} & 
    \Sexpr{serTab["PopID", "fval"]} & \Sexpr{serTab["PopID", "pval"]} \\
  Temperature & \Sexpr{serTab["TempTrt", "ss"]} & 
    \Sexpr{serTab["TempTrt", "ms"]} & \Sexpr{serTab["TempTrt", "df1"]} &
    \Sexpr{serTab["TempTrt", "df2"]} & \Sexpr{serTab["TempTrt", "fval"]} &
    \Sexpr{serTab["TempTrt", "pval"]} \\
  Water & \Sexpr{serTab["WaterTrt", "ss"]} & \Sexpr{serTab["WaterTrt", "ms"]} & 
    \Sexpr{serTab["WaterTrt", "df1"]} & \Sexpr{serTab["WaterTrt", "df2"]} & 
    \Sexpr{serTab["WaterTrt", "fval"]} & \Sexpr{serTab["WaterTrt", "pval"]} \\
  Temperature $\times$ Water & \Sexpr{serTab["TempTrt:WaterTrt", "ss"]} & 
    \Sexpr{serTab["TempTrt:WaterTrt", "ms"]} & 
    \Sexpr{serTab["TempTrt:WaterTrt", "df1"]} & 
    \Sexpr{serTab["TempTrt:WaterTrt", "df2"]} & 
    \Sexpr{serTab["TempTrt:WaterTrt", "fval"]} &
    \Sexpr{serTab["TempTrt:WaterTrt", "pval"]} \\
  Population $\times$ Temperature & \Sexpr{serTab["PopID:TempTrt", "ss"]} & 
    \Sexpr{serTab["PopID:TempTrt", "ms"]} & 
    \Sexpr{serTab["PopID:TempTrt", "df1"]} & 
    \Sexpr{serTab["PopID:TempTrt", "df2"]} & 
    \Sexpr{serTab["PopID:TempTrt", "fval"]} &
    \Sexpr{serTab["PopID:TempTrt", "pval"]} \\
  Population $\times$ Water & \Sexpr{serTab["PopID:WaterTrt", "ss"]} & 
    \Sexpr{serTab["PopID:WaterTrt", "ms"]} & 
    \Sexpr{serTab["PopID:WaterTrt", "df1"]} &
    \Sexpr{serTab["PopID:WaterTrt", "df2"]} &
    \Sexpr{serTab["PopID:WaterTrt", "fval"]} & 
    \Sexpr{serTab["PopID:WaterTrt", "pval"]} \\
  Population $\times$ Temperature $\times$ Water & 
    \Sexpr{serTab["PopID:TempTrt:WaterTrt", "ss"]} & 
    \Sexpr{serTab["PopID:TempTrt:WaterTrt", "ms"]} & 
    \Sexpr{serTab["PopID:TempTrt:WaterTrt", "df1"]} & 
    \Sexpr{serTab["PopID:TempTrt:WaterTrt", "df2"]} & 
    \Sexpr{serTab["PopID:TempTrt:WaterTrt", "fval"]} & 
    \Sexpr{serTab["PopID:TempTrt:WaterTrt", "pval"]} \\
	\bottomrule
	\end{tabular}
	\label{table:TableS_fitSER}
	\end{center}
\end{table}

% ANOVA table for photosynthesis

<<TableS_fitPhoto, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=
	
  # Convert numbers from R to LaTeX characters
  photoTab <- data.frame(ss = sigDig(fitPhoto$anova.table$`Sum Sq`),
                         ms = sigDig(fitPhoto$anova.table$`Mean Sq`),
                         df1 = sigDig(fitPhoto$anova.table$NumDF),
                         df2 = sigDig(fitPhoto$anova.table$DenDF),
                         fval = sigDig(fitPhoto$anova.table$F.value),
                         pval = pval2latex(fitPhoto$anova.table$`Pr(>F)`),
                         stringsAsFactors = F)
  rownames(photoTab) <- rownames(fitPhoto$anova.table)

@

\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[ANOVA table, photosynthesis]{Analysis of varianace (ANOVA) table on photosynthetic rate using \pkg{lmerTest} \citep{Kuznetsova_etal_2016}. Family and Block were included as random effects. Abbreviations: SS = sum of squares; MS = mean sum of squares (SS / df1); df1 = numerator degrees of freedom; df2 = denominator degrees of freedom.}
	\begin{center}
	\begin{tabular}{lcccccc}
	\toprule

  & SS & MS & df1 & df2 & \em{F}-value & $P$-value \\ \midrule
  Population & \Sexpr{photoTab["PopID", "ss"]} & 
    \Sexpr{photoTab["PopID", "ms"]} & \Sexpr{photoTab["PopID", "df1"]} &
    \Sexpr{photoTab["PopID", "df2"]} & \Sexpr{photoTab["PopID", "fval"]} &
    \Sexpr{photoTab["PopID", "pval"]} \\
  Temperature & \Sexpr{photoTab["TempTrt", "ss"]} & 
    \Sexpr{photoTab["TempTrt", "ms"]} & \Sexpr{photoTab["TempTrt", "df1"]} &
    \Sexpr{photoTab["TempTrt", "df2"]} & \Sexpr{photoTab["TempTrt", "fval"]} &
    \Sexpr{photoTab["TempTrt", "pval"]} \\
  Water & \Sexpr{photoTab["WaterTrt", "ss"]} & 
    \Sexpr{photoTab["WaterTrt", "ms"]} & \Sexpr{photoTab["WaterTrt", "df1"]} &
    \Sexpr{photoTab["WaterTrt", "df2"]} & \Sexpr{photoTab["WaterTrt", "fval"]} &
    \Sexpr{photoTab["WaterTrt", "pval"]} \\
  Temperature $\times$ Water & \Sexpr{photoTab["TempTrt:WaterTrt", "ss"]} & 
    \Sexpr{photoTab["TempTrt:WaterTrt", "ms"]} & 
    \Sexpr{photoTab["TempTrt:WaterTrt", "df1"]} & 
    \Sexpr{photoTab["TempTrt:WaterTrt", "df2"]} & 
    \Sexpr{photoTab["TempTrt:WaterTrt", "fval"]} &
    \Sexpr{photoTab["TempTrt:WaterTrt", "pval"]} \\
  Population $\times$ Temperature & \Sexpr{photoTab["PopID:TempTrt", "ss"]} & 
    \Sexpr{photoTab["PopID:TempTrt", "ms"]} & 
    \Sexpr{photoTab["PopID:TempTrt", "df1"]} & 
    \Sexpr{photoTab["PopID:TempTrt", "df2"]} & 
    \Sexpr{photoTab["PopID:TempTrt", "fval"]} &
    \Sexpr{photoTab["PopID:TempTrt", "pval"]} \\
  Population $\times$ Water & \Sexpr{photoTab["PopID:WaterTrt", "ss"]} & 
    \Sexpr{photoTab["PopID:WaterTrt", "ms"]} & 
    \Sexpr{photoTab["PopID:WaterTrt", "df1"]} &
    \Sexpr{photoTab["PopID:WaterTrt", "df2"]} &
    \Sexpr{photoTab["PopID:WaterTrt", "fval"]} & 
    \Sexpr{photoTab["PopID:WaterTrt", "pval"]} \\
  Population $\times$ Temperature $\times$ Water & 
    \Sexpr{photoTab["PopID:TempTrt:WaterTrt", "ss"]} & 
    \Sexpr{photoTab["PopID:TempTrt:WaterTrt", "ms"]} & 
    \Sexpr{photoTab["PopID:TempTrt:WaterTrt", "df1"]} & 
    \Sexpr{photoTab["PopID:TempTrt:WaterTrt", "df2"]} & 
    \Sexpr{photoTab["PopID:TempTrt:WaterTrt", "fval"]} & 
    \Sexpr{photoTab["PopID:TempTrt:WaterTrt", "pval"]} \\
	\bottomrule
	\end{tabular}
	\label{table:TableS_fitPhoto}
	\end{center}
\end{table}

% Analysis of deviance table for mortality

<<TableS_fitPhoto, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=
	
  # Convert numbers from R to LaTeX characters
  tab <- Anova(fitMort, type = 2)
  mortTab <- data.frame(chisq = sigDig(tab$Chisq), df = tab$Df,
                         pval = pval2latex(tab$`Pr(>Chisq)`),
                         stringsAsFactors = F)
  rownames(mortTab) <- rownames(tab)

@

\begin{table}[htbp]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Analysis of deviance table, mortality]{Analysis of deviance table on the probability of mortality by the end of the experiment using Type-II Wald $\chi ^2$ tests in the R package \pkg{car} \citep{Fox_Weisberg_2011}. Family and Block were included as random effects. Abbreviations: df = degrees of freedom}
	\begin{center}
	\begin{tabular}{lccc}
	\toprule

  & $\chi ^2$ & df & $P$-value \\ \midrule
  Population & \Sexpr{mortTab["PopID", "chisq"]} & 
    \Sexpr{mortTab["PopID", "df"]} & \Sexpr{mortTab["PopID", "pval"]} \\
  Temperature & \Sexpr{mortTab["TempTrt", "chisq"]} & 
    \Sexpr{mortTab["TempTrt", "df"]} & \Sexpr{mortTab["TempTrt", "pval"]} \\
  Water & \Sexpr{mortTab["WaterTrt", "chisq"]} & 
    \Sexpr{mortTab["WaterTrt", "df"]} & \Sexpr{mortTab["WaterTrt", "pval"]} \\
  Temperature $\times$ Water & \Sexpr{mortTab["TempTrt:WaterTrt", "chisq"]} & 
    \Sexpr{mortTab["TempTrt:WaterTrt", "df"]} & 
    \Sexpr{mortTab["TempTrt:WaterTrt", "pval"]} \\
  Population $\times$ Temperature & \Sexpr{mortTab["PopID:TempTrt", "chisq"]} & 
    \Sexpr{mortTab["PopID:TempTrt", "df"]} & 
    \Sexpr{mortTab["PopID:TempTrt", "pval"]} \\
  Population $\times$ Water & \Sexpr{mortTab["PopID:WaterTrt", "chisq"]} & 
    \Sexpr{mortTab["PopID:WaterTrt", "df"]} & 
    \Sexpr{mortTab["PopID:WaterTrt", "pval"]} \\
  Population $\times$ Temperature $\times$ Water & 
    \Sexpr{mortTab["PopID:TempTrt:WaterTrt", "chisq"]} & 
    \Sexpr{mortTab["PopID:TempTrt:WaterTrt", "df"]} & 
    \Sexpr{mortTab["PopID:TempTrt:WaterTrt", "pval"]} \\	\bottomrule
	\end{tabular}
	\label{table:TableS_fitMort}
	\end{center}
\end{table}

% Table of Climate-Latitude and Climate-TraitPC1 variables

\begin{table}[ht]
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
  \caption[Important climatic variables]{Important climatic variables predicting latitude of \textit{E. cardinalis} populations (`Climate-Latitude') and the first principal component of traits measured in a common garden (`Climate-TraitPC1'). Local climatic variables were measured from the exact location of collection; neighborhood climatic variables were averaged from a 62-km neighborhood around population (see Material and Methods). Importance and significance were determined using the variable selection using random forests (VSURF) algorithm (see Material and Methods). Climatic variables are described in Table~\ref{table:TableS_ClimVars}. $\mu$ signifies the mean of the climate variables from 1981--2010; $\sigma$ indicates coeffiecient of variation among years. }
   \begin{center}
   \begin{tabular}{@{} ll @{}}
      \toprule
    	Climate-Latitude variables & Climate-TraitPC1 variables \\
      \midrule
			\input{Tables/TableS_ClimVarImp.txt}
			\bottomrule
   \end{tabular}
   \end{center}
   \label{table:TableS_ClimVarImp}
\end{table}

\clearpage

%--------------------------------------------------
% Supporting Figures
%--------------------------------------------------

\subsection*{Supporting Figures}

% Figures showing latitudinal clines with specific traits

<<FigS_GermLat, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

	pdf(paste(pathFig, "/FigureS_GermLat.pdf", sep = ""), 5, 5)
	par(mai = c(1, 1, 0.1, 0.1))
	plot(focClim$Latitude, 
	     c(coef(fitGerm)[1], coef(fitGerm)[1] + coef(fitGerm)[2:16]), 
	     xlim = c(30, 45), ylim = c(1.65, 2.65), xlab = "Latitude of Origin", 
	     ylab = "Days to Germination [log scale]", cex.lab = 1.5, type = "n", 
	     axes = F, frame.plot = T)
	lowCI <- c(confint(fitGerm)[1, 1], 
	           coef(fitGerm)[1] + confint(fitGerm)[2:16, 1])
	uppCI <- c(confint(fitGerm)[1, 2], 
	           coef(fitGerm)[1] + confint(fitGerm)[2:16, 2])
	arrows(focClim$Latitude, lowCI, focClim$Lat, uppCI, length = 0.05, angle = 90,
	       code = 3)
	points(focClim$Latitude, c(coef(fitGerm)[1], 
	                           coef(fitGerm)[1] + coef(fitGerm)[2:16]), 
	       col = "white", bg = "black", pch = 21, cex = 1.5)
	
	axis(1, at = seq(30, 45, 5), lwd = 0, lwd.ticks = 1,
	     labels = expression(30*degree, 35*degree, 40*degree,	45*degree))	
	axis(2, lwd = 0, lwd.ticks = 1, las = 1, at = log(seq(6, 14, 2)),
	     labels = seq(6, 14, 2))	
	
	fit <- lm(c(coef(fitGerm)[1], coef(fitGerm)[1] + coef(fitGerm)[2:16]) ~
	            Latitude, data = focClim)
	points(range(focClim$Latitude), 
	       predict(fit, new = data.frame(Latitude = range(focClim$Latitude))),
	       type = "l", lwd = 2)
	
	text(grconvertX(0.05, "npc"), grconvertY(0.95, "npc"), adj = c(0, 1),
	     labels = bquote(italic(P) == .(round(summary(fit)[[4]][2, 4], 3))))
	
	dev.off()
	
@
	
\begin{figure}[h!]
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/FigureS_GermLat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations germinate faster]{Southern populations germinate faster. Each point is a population of \textit{E. cardinalis} showing its latitude of origin (x-axis) and model-predicted days to germination in days under growth chamber conditions (see Material and Methods). Bars around each point are 95\% confidence intervals. Predicted time to germination and confidence intervals are based on survival regression (see Materials and Materials). The line is the linear regression of log(model-predicated days to germination) $\sim$ latitude. The $P$-value of the regression is given in the upper left corner.}
	\label{fig:FigS_GermLat}
\end{figure}

<<FigS_LERlat, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

	pdf(paste(pathFig, "/FigureS_LERlat.pdf", sep = ""), 5, 5)
	par(mai = c(1, 1, 0.1, 0.1))
	plot(focClim$Latitude, fitLER$lsmeans.table[1:16, "Estimate"], 
	     xlim = c(30, 45), ylim = c(1.5, 3.1), xlab = "Latitude of Origin", 
	     ylab = expression(paste("Leaf expansion rate (", mm~day^{-1}, ")")), 
	     las = 1, cex.lab = 1.5, type = "n", axes = F, frame.plot = T)
	arrows(focClim$Lat, fitLER$lsmeans.table[1:16, "Lower CI"], 
	       focClim$Lat, fitLER$lsmeans.table[1:16, "Upper CI"], 
	       angle = 90, length = 0.05, code = 3)
	points(focClim$Lat, fitLER$lsmeans.table[1:16, "Estimate"], 
	       col = "white", bg = "black", pch = 21, cex = 1.5)
	
	axis(1, at = seq(30, 45, 5), lwd = 0, lwd.ticks = 1,
	     labels = expression(30*degree, 35*degree, 40*degree, 45*degree))	
	axis(2, lwd = 0, lwd.ticks = 1, las = 1)	
	
	fit <- lm(fitLER$lsmeans.table[1:16, "Estimate"] ~ Latitude, data = focClim)

	points(range(focClim$Latitude), 
	       predict(fit, new = data.frame(Latitude = range(focClim$Latitude))),
	       type = "l", lwd = 2)

	text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1),
	     labels = bquote(italic(P) == .(round(summary(fit)[[4]][2, 4], 3))))
	
	dev.off()
	
@

\begin{figure}[h!]
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/FigureS_LERlat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations grow faster (leaf expansion rate).]{Southern populations grow faster. Each point is a population of \textit{E. cardinalis} showing its latitude of origin (x-axis) and model-predicted leaf expansion rate during the rosette phase. Bars around each point are 95\% confidence intervals. Predicted leaf expansion rate based least-square mean estimates and confidence intervals were calculated from linear mixed-effects models (see Materials and Materials). The line is the linear regression of model-predicated leaf expansion rate $\sim$ latitude. The $P$-value of the regression is given in the upper right corner.}
	\label{fig:FigS_LERlat}
\end{figure}

<<FigS_SERlat, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

	pdf(paste(pathFig, "/FigureS_SERlat.pdf", sep = ""), 5, 5)
	par(mai = c(1, 1, 0.1, 0.1))
	plot(focClim$Latitude, fitSER$lsmeans.table[1:16, "Estimate"], 
	     xlim = c(30, 45), ylim = c(0.75, 1.65), xlab = "Latitude of Origin", 
	     ylab = expression(paste("Stem elongation rate (", mm~day^{-1}, ")")), 
	     las = 1, cex.lab = 1.5, type = "n", axes = F, frame.plot = T)
	arrows(focClim$Lat, fitSER$lsmeans.table[1:16, "Lower CI"], 
	       focClim$Lat, fitSER$lsmeans.table[1:16, "Upper CI"], 
	       angle = 90, length = 0.05, code = 3)
	points(focClim$Lat, fitSER$lsmeans.table[1:16, "Estimate"], 
	       col = "white", bg = "black", pch = 21, cex = 1.5)
	
	axis(1, at = seq(30, 45, 5), lwd = 0, lwd.ticks = 1,
	     labels = expression(30*degree, 35*degree, 40*degree, 45*degree))	
	axis(2, lwd = 0, lwd.ticks = 1, las = 1)	
	
	fit <- lm(fitSER$lsmeans.table[1:16, "Estimate"] ~ Latitude, data = focClim)

	points(range(focClim$Latitude), 
	       predict(fit, new = data.frame(Latitude = range(focClim$Latitude))),
	       type = "l", lwd = 2)

	text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1),
	     labels = bquote(italic(P) == .(round(summary(fit)[[4]][2, 4], 3))))
	
	dev.off()
	
@

\begin{figure}[h!]
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/FigureS_SERlat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations grow faster (stem elongation rate).]{Southern populations grow faster. Each point is a population of \textit{E. cardinalis} showing its latitude of origin (x-axis) and model-predicted stem elongation rate. Bars around each point are 95\% confidence intervals. Predicted stem elongation rate based least-square mean estimates and confidence intervals were calculated from linear mixed-effects models (see Materials and Materials). The line is the linear regression of model-predicated stem elongation rate $\sim$ latitude. The $P$-value of the regression is given in the upper right corner.}
	\label{fig:FigS_SERlat}
\end{figure}

<<FigS_PhotoLat, echo=FALSE, results=hide, warning=FALSE, message=FALSE>>=

	pdf(paste(pathFig, "/FigureS_PhotoLat.pdf", sep = ""), 5.5, 5)
	par(mai = c(1, 1.5, 0.1, 0.1))
	plot(focClim$Latitude, fitPhoto$lsmeans.table[1:16, "Estimate"], 
	     xlim = c(30, 45), ylim = c(8, 16), xlab = "Latitude of Origin", 
	     ylab = expression(paste(atop("Photosynthetic rate", paste("(", mu, 
	                             mol~CO[2]~s^{-1}~m^{-2}, ")")))), 
	     las = 1, cex.lab = 1.5, type = "n", axes = F, frame.plot = T)
	arrows(focClim$Lat, fitPhoto$lsmeans.table[1:16, "Lower CI"], 
	       focClim$Lat, fitPhoto$lsmeans.table[1:16, "Upper CI"], 
	       angle = 90, length = 0.05, code = 3)
	points(focClim$Lat, fitPhoto$lsmeans.table[1:16, "Estimate"], 
	       col = "white", bg = "black", pch = 21, cex = 1.5)
	
	axis(1, at = seq(30, 45, 5), lwd = 0, lwd.ticks = 1,
	     labels = expression(30*degree, 35*degree, 40*degree, 45*degree))	
	axis(2, lwd = 0, lwd.ticks = 1, las = 1)	
	
	fit <- lm(fitPhoto$lsmeans.table[1:16, "Estimate"] ~ Latitude, data = focClim)

	points(range(focClim$Latitude), 
	       predict(fit, new = data.frame(Latitude = range(focClim$Latitude))),
	       type = "l", lwd = 2)

	text(grconvertX(0.95, "npc"), grconvertY(0.95, "npc"), adj = c(1, 1),
	     labels = bquote(italic(P) == .(round(summary(fit)[[4]][2, 4], 3))))
	
	dev.off()
	
@

\begin{figure}[h!]
	\centerline{\includegraphics[width=0.5\textwidth]{Figures/FigureS_Photolat.pdf}}
	\usefont{T1}{phv}{m}{n}
	\fontsize{10}{12}
	\selectfont
	\caption[Southern populations photosynthesize faster]{Southern populations photosynthesize faster. Each point is a population of \textit{E. cardinalis} showing its latitude of origin (x-axis) and model-predicted instantaneous photosynthetic rate. Bars around each point are 95\% confidence intervals. Predicted photosynthetic rates based least-square mean estimates and confidence intervals were calculated from linear mixed-effects models (see Materials and Materials). The line is the linear regression of model-predicated photosynthetic rate $\sim$ latitude. The $P$-value of the regression is given in the upper right corner.}
	\label{fig:FigS_Photolat}
\end{figure}

% Reaction norms of LER, SER, Photo, and Mort

\begin{figure}
\centering
    \begin{subfigure}{.4\textwidth}
        \centering
        \includegraphics[width=\linewidth]{Figures/FigureS_LERrxnNorm.pdf}
        \phantomcaption{}\label{fig:FigS_LERrxnNorm}
    \end{subfigure} %
    \begin{subfigure}{.4\textwidth}
        \centering
        \includegraphics[width=\linewidth]{Figures/FigureS_SERrxnNorm.pdf}
        \phantomcaption{}\label{fig:FigS_SERrxnNorm}
    \end{subfigure} %
    \begin{subfigure}{0.4\textwidth}
        \centering
        \includegraphics[width=\linewidth]{Figures/FigureS_PhotoRxnNorm.pdf}
        \phantomcaption{}\label{fig:FigS_PhotoRxnNorm}
    \end{subfigure}
    \begin{subfigure}{0.4\textwidth}
        \centering
        \includegraphics[width=\linewidth]{Figures/FigureS_MortRxnNorm.pdf}
        \phantomcaption{}\label{fig:FigS_MortRxnNorm}
    \end{subfigure}
\caption[Reaction norms]{Reaction norms signify little Population $\times$ Treatment interactions. For all panels, black lines represent population-level reaction norms from Wet to Dry in the Cool temperature treatment (dashed black lines) and Hot temperature treatment (solid black lines); gray lines represent reaction norms from Cool to Hot in the Wet treatment (solid gray lines) and Dry treatment (dashed gray lines). The responses shown are (A) leaf expansion rate, (B) stem elongation rate, (C) photosynthesis, and (D) survivorship (= 1 - mortality).}
	\label{fig:FigS_RxnNorms}
\end{figure}

% Figure of bio11_cva_SA versus TraitPC1

\begin{figure}[h!]
\centerline{\includegraphics[width=0.5\textwidth]{Figures/FigureS_bio11PC1.pdf}}
\usefont{T1}{phv}{m}{n}
\fontsize{10}{12}
\selectfont
\caption[Trait PC1 versus bio11_cva_SA]{Trait variation, from fast to slow growth, is closely associated with neighborhood variation in temperature of the coldest quarter (bio11$_\sigma$) Each point is a population coefficient of variation in bio11 averaged over a 62-km climatic neighborhood  (x-axis) and position along the slow to fast growth axis (y-axis), defined as Principal Component 1 of four traits (see Material and Methods). The line and 95\% confidence intervals were estimated using linear regression.}
\label{fig:FigS_bio11PC1}
\end{figure}

\clearpage


\subsection*{Supporting Material and Methods}

\subsubsection*{Temperature treatments}

We simulated typical growing season (June 1 - August 15) air temperatures at the two most thermally divergent focal sites in our study, Whitewater Canyon (WWC, Hot) and Little Jameson (LIJ, Cool). We downloaded daily interpolated mean, minimum, and maximum air temperature from 13 years (2000-2012) at both sites from ClimateWNA \citep{Wang_etal_2012}. This range was chosen because seeds used in the experiment were collected around 2012, thus their presence in that location at that time suggests that populations were able to persist there for at least some years before collection. Monthly temperatures from ClimateWNA are highly correlated with the air temperature recorded from data loggers in the field at these sites (A. Angert, unpub. data). Hence, the ClimateWNA temperature profiles are similar to actual thermal regimes experienced by \textit{E. cardinalis} in nature. We simulated realistic temperature regimes by calculating the mean temperature trend from June to August using LOESS \citep{Cleveland_etal_1992}. The residuals were highly autocorrelated at both sites (warmer than average days are typically followed by more warm days) and there was strong correlation ($r = 0.65$) between sites (warm days in WWC were also warm in LIJ). The `VARselect' function in the \pkg{vars} package for R \citep{Pfaff_2008} indicated that a lag two Vector Autoregression (VAR(2)) model best captured the within-site autocorrelation as well as between-site correlation in residuals. We fit and simulated from the VAR(2) model using the package \pkg{dse} \citep{Gilbert_2014} in R. Simulated data closely resembled the autocorrelation and between-site correlation of the actual data. From simulated mean temperature, we next selected minimum and maximum daily temperatures. Mean, min, and max temperature were highly correlated at both sites. We chose min and max temperatures using site-specific fitted linear models between mean, max, and min temperature, with additional variation given by normally distributed random deviates with variance equal to the residual variance of the linear models. For each day, the nighttime (22:00 - 6:00) chamber temperature was set to the simulated minimum temperature. During the middle of the day, temperature was set to the simulated maximum temperature, with a variable period of transition between min and max so that the average temperature was equal the simulated mean temperature.

\subsubsection*{Watering treatments}

For watering treatments, we simulated two extreme types of streams where \textit{E. cardinalis} grows. In the well-watered treatment, we simulated a large stream that never goes dry during the summer growing season. In the drought treatment, we simulated a small stream that has ample flow at the beginning of the season due to rain and snow melt, but gradually dries down through the summer. In both treatments, plants were bottom-watered using  water chilled to 7.5\celsius. Plants in the well-watered treatment were fully saturated every two hours during the day. Watering in the drought treatment gradually declined from every two hours to every day between May 20 (36 days after sowing) and 10 June (57 days after sowing). Simultaneously, the amount of bottom-watering per flood decreased, such that only the bottom of the cone-tainers were wetted by the end of the experiment.

\end{document}